{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "US-9OQ8XgZQF"
   },
   "source": [
    "\n",
    "# GATK Germline Copy Number Variation Tutorial <a class=\"tocSkip\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**March 2023**  \n",
    "\n",
    "<img src=\"https://storage.googleapis.com/broad-dsde-methods-gatk-workshop-public/images/germline-cnv-tutorial/gatk-gcnv-pipeline-overview.png\" alt=\"drawing\" width=\"100%\" align=\"center\"/>\n",
    "\n",
    "\n",
    "The tutorial outlines steps in detecting germline copy number variants using GATK-gCNVs and illustrates two workflow modes: **cohort mode** and the **case mode**. The cohort mode simultaneously generates a cohort model and calls CNVs for the cohort samples. The case mode analyzes a single sample against an already constructed cohort model. The same workflow steps apply to both targeted exome and whole genome sequencing (WGS) data. The workflow is able to call both rare and common events and intelligently handles allosomal ploidies, i.e. cohorts of mixed male and female samples.\n",
    "\n",
    "For the cohort mode, the general recommendation is at least a hundred samples to start. Researchers should expect to tune workflow parameters from the provided defaults. In particular, GermlineCNVCaller's default inference parameters are conservatively set for efficient run times.\n",
    "\n",
    "The figure above diagrams the workflow tools. **Section 1** sets up the Notebook environment and downloads the data and resource files needed to complete this tutorial. **Section 2** creates an intervals list and counts read alignments overlapping the intervals. **Section 3** shows optional but recommended cohort mode steps to annotate intervals with covariates for use in filtering intervals as well as for use in explicit modeling. The section also removes outlier counts intervals. **Section 4** generates global baseline observations for the data and models and calls the ploidy of each contig (ploidy means the overall baseline copy number of the contig). **Section 5** is at the heart of the workflow and models per-interval copy number. Because of the high intensity of compute model fitting requires, the section shows how to analyze data in parts. Finally, **Section 6** calls per-sample copy number events per interval and per segment. Results are in VCF format.\n",
    "\n",
    "➤ This guided Notebook demo is largely based on [Tutorial #11682](https://gatk.broadinstitute.org/hc/en-us/articles/360035531152--How-to-Call-common-and-rare-germline-copy-number-variants)   \n",
    "➤ A highly recommended [paper](https://www.nature.com/articles/s41588-023-01449-0) detailing the methods was published in _Nature Genetics_ in 2023.  \n",
    "➤ For pipelined workflows, see the WDL scripts in this Terra workspace.  \n",
    "➤ This workflow is **not** appropriate for bulk tumor samples, as it infers absolute copy numbers. For somatic copy number alteration calling, see [Tutorial #11682](https://gatk.zendesk.com/hc/en-us/articles/360035531092).\n",
    "\n",
    "_This tutorial was last tested with the GATK v4.2.4.0 and IGV v2.8.0._\n",
    " See [GATK Tool Documentation](https://gatk.broadinstitute.org/hc/en-us/articles/360037224712) for further information on the tools we use below."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "OgzJFRYkgZQH"
   },
   "source": [
    "# Set up your Notebook\n",
    "\n",
    "\n",
    "## Set cloud environment values\n",
    "If you are opening a notebook for the first time today and you didn't adjust any runtime values, now's the time to edit them. Click on the gear icon in the upper right to edit your Notebook Runtime. Set the values as specified below:\n",
    "\n",
    "| Option | Value |\n",
    "| ------ | ------ |\n",
    "| Environment | [Custom](https://github.com/broadinstitute/gatk-workshop-terra-jupyter-image/wiki/Using-the-gatk%E2%80%90workshop%E2%80%90terra%E2%80%90jupyter%E2%80%90image-in-the-Terra-Jupyter-environment#6-type-the-command-setup_gatk_env-in-the-terminal-and-hit-enter) |\n",
    "| Profile | Custom |\n",
    "| CPU | 4 |\n",
    "| Disk size | 100 GB |\n",
    "| Memory | 15 GB |\n",
    "\n",
    "**Please Note:** This notebook currently requires the use of a custom environment, as described [here](https://github.com/broadinstitute/gatk-workshop-terra-jupyter-image/wiki/Using-the-gatk%E2%80%90workshop%E2%80%90terra%E2%80%90jupyter%E2%80%90image-in-the-Terra-Jupyter-environment#6-type-the-command-setup_gatk_env-in-the-terminal-and-hit-enter).\n",
    "\n",
    "Click the \"create\"/\"Update\" button when you are done, and Terra will begin to create a new runtime with your settings. When it is finished, it will pop up asking you to apply the new settings. In the meantime, you can continue with the setup instructions below. \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Setup the GATK conda environment\n",
    "\n",
    "Run the following commands in the Terra terminal (available from the right panel):\n",
    "\n",
    "```\n",
    "setup_gatk_env\n",
    "```\n",
    "\n",
    "When you are done, you should close this notebook and reopen.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check kernel type selected in the Jupyter notebook\n",
    "For this particular notebook, we will be using a specialized kernel that we've added to our **Kernel** menu by following the previous two steps for creating the custom envrionemnt. If you've successfully done this, you should be able to select a kernel called `Python [conda env:gatkconda]`under the **Kernel** > **Change Kernel** menu."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set up your files\n",
    "Your notebook has a temporary folder that exists so long as your cluster is running. To see what files are in your notebook environment at any time, you can click on the Jupyter logo in the upper left corner. \n",
    "\n",
    "For this tutorial, we need to copy some files from this temporary folder to and from our workspace bucket. Run the two commands below to set up the workspace bucket variable and the file paths inside your notebook.\n",
    "\n",
    "<font color = \"green\"> **Tool Tip:** To run a cell in a notebook, press `SHIFT + ENTER`</font>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "aZc1n9fxgZQI"
   },
   "outputs": [],
   "source": [
    "# Set your workspace bucket variable for this notebook.\n",
    "import os\n",
    "\n",
    "# Get the workspace bucket from Terra Bio\n",
    "os.environ['WORKSPACE_BUCKET'] = \"gs://fc-b426f92d-adf2-45bc-a714-69ee3db2ba38\"\n",
    "\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "WORKSHOP_BUCKET = 'gs://broad-dsde-methods-gatk-workshop-public'\n",
    "REFERENCE_BUCKET = 'gs://genomics-public-data/resources/broad/hg38/v0'\n",
    "WORKSPACE_LOCAL = '/home/jupyter/notebooks/germline-cnv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "M2vQ99YsgZQL"
   },
   "outputs": [],
   "source": [
    "# Create sandbox and file directories\n",
    "! mkdir -p $WORKSPACE_LOCAL/sandbox/\n",
    "! mkdir -p $WORKSPACE_LOCAL/ref/\n",
    "! mkdir -p $WORKSPACE_LOCAL/data/\n",
    "\n",
    "# Removes any old symbolic linked sandbox directory and adds a new link, in case you've run this before\n",
    "! rm -rf sandbox\n",
    "! ln -s $WORKSPACE_LOCAL/sandbox sandbox"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "lG2m2totgZQN"
   },
   "source": [
    "## Check data permissions\n",
    "For this tutorial, we have hosted the starting files in a public Google bucket. We will first check that the data is available to your user account, and if it is not, we simply need to install Google Cloud Storage."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "pH5Yhgh6gZQO",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://broad-dsde-methods-gatk-workshop-public/germline-cnv/\r\n",
      "gs://broad-dsde-methods-gatk-workshop-public/germline-cnv/tutorial_11684.tar.gz\r\n"
     ]
    }
   ],
   "source": [
    "# Check if data is accessible. The command should list several gs:// URLs.\n",
    "! gsutil ls $WORKSHOP_BUCKET/germline-cnv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# If you do not see gs:// URLs listed above, run this cell to install Google Cloud Storage. \n",
    "# Afterwards, restart the kernel with Kernel > Restart.\n",
    "#! pip install google-cloud-storage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "_QGHTOCvgZQS"
   },
   "source": [
    "## Download data to local disk"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The tutorial provides example small WGS data sourced from the _1000 Genomes Project_. Cohort mode illustrations use 24 samples, while case mode illustrations analyze one sample against a cohort model made from the remaining 23 samples. The tutorial uses a fraction of the workflow's recommended hundred samples for ease of illustration. Furthermore, commands in each step use one of three differently sized intervals lists for efficiency. Coverage data are from the entirety of chr20, chrX and chrY. So although a step may analyze a subset of regions, it is possible to instead analyze all three contigs in case or cohort modes.\n",
    "\n",
    "Download tutorial_11684.tar.gz either from the Workshop Google bucket. The bundle includes data for [Notebook #11685](https://gatk.zendesk.com/hc/en-us/articles/360035890031) and [Notebook #11686](https://gatk.zendesk.com/hc/en-us/articles/360035889891). The tutorial also requires the GRCh38 reference FASTA, dictionary and index. These are available from the GATK Resource Bundle. The example data is from the [1000 Genomes project](http://www.internationalgenome.org/) Phase 3 aligned to GRCh38."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "-TkCGW77gZQS",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://broad-dsde-methods-gatk-workshop-public/germline-cnv/tutorial_11684.tar.gz...\n",
      "==> NOTE: You are downloading one or more large file(s), which would            \n",
      "run significantly faster if you enabled sliced object downloads. This\n",
      "feature is enabled by default but requires that compiled crcmod be\n",
      "installed (see \"gsutil help crcmod\").\n",
      "\n",
      "| [1 files][543.0 MiB/543.0 MiB]   13.5 MiB/s                                   \n",
      "Operation completed over 1 objects/543.0 MiB.                                    \n"
     ]
    }
   ],
   "source": [
    "! gsutil cp $WORKSHOP_BUCKET/germline-cnv/tutorial_11684.tar.gz /home/jupyter/notebooks/germline-cnv/data/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "! tar -xvf $WORKSPACE_LOCAL/data/tutorial_11684.tar.gz -C $WORKSPACE_LOCAL/data > /dev/null 2>&1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# let's make a shortcut variable\n",
    "TUTORIAL_DATA_PATH = '/home/jupyter/notebooks/germline-cnv/data/Tutorial11684'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "cs1hPg-IgZQX"
   },
   "source": [
    "---\n",
    "\n",
    "# Collect raw counts data with _PreprocessIntervals_ and _CollectReadCounts_\n",
    "PreprocessIntervals pads exome targets and bins WGS intervals. Binning refers to creating equally sized intervals across the reference. For example, 1000 base binning would define chr1:1-1000 as the first bin. Because counts of reads on reference N bases are not meaningful, the tool automatically excludes bins with all Ns. For GRCh38 chr1, non-N sequences start at base 10,001, so the first few bin become:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "6SId3hYJgZQY"
   },
   "source": [
    "## For WGS data, bin entirety of reference, e.g. with 1000 base intervals.\n",
    "(We skip this step in this tutorial, but the command-line is shown below for your reference)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "jP5klLTAgZQZ",
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#  ! gatk PreprocessIntervals \\\n",
    "#         -R $REFERENCE_BUCKET/Homo_sapiens_assembly38.fasta \\\n",
    "#         --padding 0 \\\n",
    "#         -imr OVERLAPPING_ONLY \\\n",
    "#         -O $WORKSPACE_LOCAL/sandbox/grch38.preprocessed.interval_list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This produces a Picard-style intervals list of 1000 base bins."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## For exome data, pad target regions, e.g. with 250 bases.\n",
    "(We skip this step in this tutorial since the data is WGS, but the command-line is shown below for your reference)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ! gatk PreprocessIntervals \\\n",
    "#         -R $REFERENCE_BUCKET/Homo_sapiens_assembly38.fasta \\\n",
    "#         -L $TUTORIAL_DATA_PATH/targets.interval_list \\\n",
    "#         --bin-length 0 \\\n",
    "#         -imr OVERLAPPING_ONLY \\\n",
    "#         -O $WORKSPACE_LOCAL/sandbox/targets.preprocessed.interval_list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## For the tutorial, we only bin two contigs\n",
    "Lets subset the reference to just chr20 and chrY."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /gatk/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /gatk/gatk-package-4.5.0.0-local.jar PreprocessIntervals -R gs://genomics-public-data/resources/broad/hg38/v0/Homo_sapiens_assembly38.fasta --bin-length 1000 --padding 0 -L chr20 -L chrY -imr OVERLAPPING_ONLY -O /home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.interval_list\n",
      "16:38:07.812 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "16:38:07.993 INFO  PreprocessIntervals - ------------------------------------------------------------\n",
      "16:38:07.998 INFO  PreprocessIntervals - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "16:38:07.998 INFO  PreprocessIntervals - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "16:38:07.998 INFO  PreprocessIntervals - Executing as root@6fffd5f5c26a on Linux v6.5.11-linuxkit amd64\n",
      "16:38:07.998 INFO  PreprocessIntervals - Java runtime: OpenJDK 64-Bit Server VM v17.0.9+9-Ubuntu-122.04\n",
      "16:38:07.998 INFO  PreprocessIntervals - Start Date/Time: January 19, 2024 at 4:38:07 PM GMT\n",
      "16:38:07.998 INFO  PreprocessIntervals - ------------------------------------------------------------\n",
      "16:38:07.998 INFO  PreprocessIntervals - ------------------------------------------------------------\n",
      "16:38:08.001 INFO  PreprocessIntervals - HTSJDK Version: 4.1.0\n",
      "16:38:08.001 INFO  PreprocessIntervals - Picard Version: 3.1.1\n",
      "16:38:08.001 INFO  PreprocessIntervals - Built for Spark Version: 3.5.0\n",
      "16:38:08.002 INFO  PreprocessIntervals - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "16:38:08.002 INFO  PreprocessIntervals - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "16:38:08.002 INFO  PreprocessIntervals - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "16:38:08.003 INFO  PreprocessIntervals - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "16:38:08.003 INFO  PreprocessIntervals - Deflater: IntelDeflater\n",
      "16:38:08.003 INFO  PreprocessIntervals - Inflater: IntelInflater\n",
      "16:38:08.003 INFO  PreprocessIntervals - GCS max retries/reopens: 20\n",
      "16:38:08.003 INFO  PreprocessIntervals - Requester pays: disabled\n",
      "16:38:08.004 INFO  PreprocessIntervals - Initializing engine\n",
      "16:38:16.583 INFO  IntervalArgumentCollection - Processing 121671582 bp from intervals\n",
      "16:38:16.588 INFO  PreprocessIntervals - Done initializing engine\n",
      "16:38:16.591 INFO  ProgressMeter - Starting traversal\n",
      "16:38:16.593 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Records Processed   Records/Minute\n",
      "16:38:16.595 INFO  ProgressMeter -             unmapped              0.0                     0              NaN\n",
      "16:38:16.595 INFO  ProgressMeter - Traversal complete. Processed 0 total records in 0.0 minutes.\n",
      "16:38:16.595 INFO  PreprocessIntervals - Padding intervals...\n",
      "16:38:16.598 INFO  PreprocessIntervals - Generating bins...\n",
      "16:38:16.620 INFO  PreprocessIntervals - Filtering bins containing only Ns...\n",
      "16:54:11.266 INFO  PreprocessIntervals - Writing bins to /home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.interval_list...\n",
      "16:54:11.439 INFO  PreprocessIntervals - PreprocessIntervals complete.\n",
      "16:54:11.440 INFO  PreprocessIntervals - Shutting down engine\n",
      "[January 19, 2024 at 4:54:11 PM GMT] org.broadinstitute.hellbender.tools.copynumber.PreprocessIntervals done. Elapsed time: 16.06 minutes.\n",
      "Runtime.totalMemory()=143654912\n"
     ]
    }
   ],
   "source": [
    "! gatk PreprocessIntervals \\\n",
    "        -R $REFERENCE_BUCKET/Homo_sapiens_assembly38.fasta \\\n",
    "        --bin-length 1000 \\\n",
    "        --padding 0 \\\n",
    "        -L chr20 -L chrY \\\n",
    "        -imr OVERLAPPING_ONLY \\\n",
    "        -O $WORKSPACE_LOCAL/sandbox/chr20Y.interval_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr20\t60001\t61000\t+\t.\r\n",
      "chr20\t61001\t62000\t+\t.\r\n",
      "chr20\t62001\t63000\t+\t.\r\n",
      "chr20\t63001\t64000\t+\t.\r\n",
      "chr20\t64001\t65000\t+\t.\r\n",
      "chr20\t65001\t66000\t+\t.\r\n",
      "chr20\t66001\t67000\t+\t.\r\n",
      "chr20\t67001\t68000\t+\t.\r\n",
      "chr20\t68001\t69000\t+\t.\r\n",
      "chr20\t69001\t70000\t+\t.\r\n"
     ]
    }
   ],
   "source": [
    "# let us take a look\n",
    "! grep -v '^@' $WORKSPACE_LOCAL/sandbox/chr20Y.interval_list | head -n 10"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Comments on select parameters of _PreprocessIntervals_\n",
    "\n",
    "- For WGS, the default 1000 `--bin-length` is the recommended starting point for typical 30x data. Be sure to set `--padding 0` to disable padding outside of given genomic regions. Bin size should correlate with depth of coverage, e.g. lower coverage data should use larger bin size while higher coverage data can support smaller bin size. The size of the bin defines the resolution of CNV calls. The factors to consider in sizing include how noisy the data is, average coverage depth and how even coverage is across the reference.\n",
    "\n",
    "- For targeted exomes, provide the exome capture kit's target intervals with `-L`, set `--bin-length 0` to disable binning and pad the intervals with `--padding 250` or other desired length.\n",
    "\n",
    "- Provide intervals to exclude from analysis with `--exclude-intervals` or `-XL`, e.g. centromeric regions. Consider using this option especially if data is aligned to a reference other than GRCh38. The workflow enables excluding regions later again using `-XL`. A frugal strategy is to collect read counts using the entirety of intervals and then to exclude undesirable regions later at the _FilterIntervals_ step (section 3), the _DetermineGermlineContigPloidy_ step (section 4), at the _GermlineCNVCaller_ step (section 6) and/or post-calling."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Count reads per bin using _CollectReadCounts_\n",
    "_CollectReadCounts_ tabulates the raw integer counts of reads overlapping an interval. The tutorial has already collected read counts ahead of time for the three contigs: chr20 and chrY. Here, we collect read counts on small data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /gatk/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /gatk/gatk-package-4.5.0.0-local.jar CollectReadCounts -L /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/chr20sub.interval_list -R gs://genomics-public-data/resources/broad/hg38/v0/Homo_sapiens_assembly38.fasta -imr OVERLAPPING_ONLY -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/NA19017.chr20sub.bam --format TSV -O /home/jupyter/notebooks/germline-cnv/sandbox/NA19017.tsv\n",
      "00:51:42.065 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "00:51:42.487 INFO  CollectReadCounts - ------------------------------------------------------------\n",
      "00:51:42.490 INFO  CollectReadCounts - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "00:51:42.490 INFO  CollectReadCounts - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "00:51:42.490 INFO  CollectReadCounts - Executing as root@6fffd5f5c26a on Linux v6.5.11-linuxkit amd64\n",
      "00:51:42.490 INFO  CollectReadCounts - Java runtime: OpenJDK 64-Bit Server VM v17.0.9+9-Ubuntu-122.04\n",
      "00:51:42.490 INFO  CollectReadCounts - Start Date/Time: January 20, 2024 at 12:51:42 AM GMT\n",
      "00:51:42.490 INFO  CollectReadCounts - ------------------------------------------------------------\n",
      "00:51:42.491 INFO  CollectReadCounts - ------------------------------------------------------------\n",
      "00:51:42.493 INFO  CollectReadCounts - HTSJDK Version: 4.1.0\n",
      "00:51:42.493 INFO  CollectReadCounts - Picard Version: 3.1.1\n",
      "00:51:42.493 INFO  CollectReadCounts - Built for Spark Version: 3.5.0\n",
      "00:51:42.494 INFO  CollectReadCounts - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "00:51:42.495 INFO  CollectReadCounts - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "00:51:42.495 INFO  CollectReadCounts - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "00:51:42.495 INFO  CollectReadCounts - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "00:51:42.495 INFO  CollectReadCounts - Deflater: IntelDeflater\n",
      "00:51:42.495 INFO  CollectReadCounts - Inflater: IntelInflater\n",
      "00:51:42.495 INFO  CollectReadCounts - GCS max retries/reopens: 20\n",
      "00:51:42.495 INFO  CollectReadCounts - Requester pays: disabled\n",
      "00:51:42.496 INFO  CollectReadCounts - Initializing engine\n",
      "00:52:01.706 INFO  FeatureManager - Using codec IntervalListCodec to read file file:///home/jupyter/notebooks/germline-cnv/data/Tutorial11684/chr20sub.interval_list\n",
      "00:52:02.106 INFO  IntervalArgumentCollection - Processing 15103000 bp from intervals\n",
      "00:52:02.144 INFO  CollectReadCounts - Done initializing engine\n",
      "00:52:02.168 INFO  CollectReadCounts - Collecting read counts...\n",
      "00:52:02.168 INFO  ProgressMeter - Starting traversal\n",
      "00:52:02.170 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
      "00:52:10.066 INFO  CollectReadCounts - 0 read(s) filtered by: WellformedReadFilter \n",
      "110022 read(s) filtered by: MappedReadFilter \n",
      "0 read(s) filtered by: NonZeroReferenceLengthAlignmentReadFilter \n",
      "45195 read(s) filtered by: NotDuplicateReadFilter \n",
      "158151 read(s) filtered by: MappingQualityReadFilter \n",
      "313368 total reads filtered out of 3566880 reads processed\n",
      "00:52:10.072 INFO  ProgressMeter -       chr20:64332513              0.1               3253512       24716475.1\n",
      "00:52:10.072 INFO  ProgressMeter - Traversal complete. Processed 3253512 total reads in 0.1 minutes.\n",
      "00:52:10.072 INFO  CollectReadCounts - Writing read counts to /home/jupyter/notebooks/germline-cnv/sandbox/NA19017.tsv...\n",
      "00:52:10.287 INFO  CollectReadCounts - CollectReadCounts complete.\n",
      "00:52:10.288 INFO  CollectReadCounts - Shutting down engine\n",
      "[January 20, 2024 at 12:52:10 AM GMT] org.broadinstitute.hellbender.tools.copynumber.CollectReadCounts done. Elapsed time: 0.47 minutes.\n",
      "Runtime.totalMemory()=157286400\n"
     ]
    }
   ],
   "source": [
    "! gatk CollectReadCounts \\\n",
    "        -L $TUTORIAL_DATA_PATH/chr20sub.interval_list \\\n",
    "        -R $REFERENCE_BUCKET/Homo_sapiens_assembly38.fasta \\\n",
    "        -imr OVERLAPPING_ONLY \\\n",
    "        -I $TUTORIAL_DATA_PATH/NA19017.chr20sub.bam \\\n",
    "        --format TSV \\\n",
    "        -O $WORKSPACE_LOCAL/sandbox/NA19017.tsv "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This generates a TSV format table of read counts. Let's take a look. After the SAM format header section, denoted by lines starting with `@`, the body of the data has a column header line followed by read counts for every interval:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CONTIG\tSTART\tEND\tCOUNT\r\n",
      "chr20\t60001\t61000\t253\r\n",
      "chr20\t61001\t62000\t560\r\n",
      "chr20\t62001\t63000\t465\r\n",
      "chr20\t63001\t64000\t108\r\n",
      "chr20\t64001\t65000\t261\r\n",
      "chr20\t65001\t66000\t370\r\n",
      "chr20\t66001\t67000\t308\r\n",
      "chr20\t67001\t68000\t286\r\n",
      "chr20\t68001\t69000\t247\r\n"
     ]
    }
   ],
   "source": [
    "! grep -v '^@' $WORKSPACE_LOCAL/sandbox/NA19017.tsv | head -n 10"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Comments on select parameters of _CollectReadCounts_\n",
    "\n",
    "- The tutorial generates text-based TSV (tab-separated-value) format data instead of the default HDF5 format by adding `--format TSV` to the command. Omit this option to generate the default HDF5 format. Downstream tools process HDF5 format more efficiently.\n",
    "\n",
    "- Here and elsewhere in the workflow, set `--interval-merging-rule` (`-imr`) to `OVERLAPPING_ONLY`, to prevent the tool from merging abutting intervals.\n",
    "\n",
    "- The tool employs a number of engine-level read filters. Of note are _NotDuplicateReadFilter_ and _MappingQualityReadFilter_. This means the tool excludes reads marked as duplicate and excludes reads with mapping quality less than `10`. Change the mapping quality threshold with the `--minimum-mapping-quality` option."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotate intervals with features using _AnnotateIntervals_ and subset regions of interest using _FilterIntervals_\n",
    "\n",
    "The steps in this section pertain to the **cohort mode**.\n",
    "\n",
    "Researchers may desire to subset the intervals that _GermlineCNVCaller_ will analyze, either to exclude potentially problematic regions or to retain only regions of interest. For example one may wish to exclude regions where all samples in a large cohort have copy number zero, or regions around segmental duplications (low-copy repeats) that often harbor common CNVs (e.g. in case the researcher is primarily interested in _de novo_ CNVs). Filtering intervals can be especially impactful for analyses that utilize references other than GRCh38 or that are based on sequencing technologies affected by sequence context, e.g. targeted exomes. The tutorial data is WGS data aligned to GRCh38, and the gCNV workflow can process the entirety of the data, without the need for any interval filtering.\n",
    "\n",
    "Towards deciding which regions to exclude, _AnnotateIntervals_ labels the given intervals with GC content and additionally with mappability and segmental duplication content if given the respective optional resource files. _FilterIntervals_ then subsets the intervals list based on the annotations and other tunable thresholds. Later, _GermlineCNVCaller_ also takes in the annotated intervals to use as covariates towards analysis.\n",
    "\n",
    "Explicit GC-correction, although optional, is recommended. The default v4.1.0.0 `cnv_germline_cohort_workflow.wdl` pipeline workflow omits explicit gc-correction and we activate it in the pipeline by setting `do_explicit_gc_correction\":\"True\"`. The tutorial illustrates the optional _AnnotateIntervals_ step by performing the recommended explicit GC-content-based filtering."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## _AnnotateIntervals_ with GC content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /gatk/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /gatk/gatk-package-4.5.0.0-local.jar AnnotateIntervals -L /home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.interval_list -R gs://genomics-public-data/resources/broad/hg38/v0/Homo_sapiens_assembly38.fasta -imr OVERLAPPING_ONLY -O /home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.annotated.tsv\n",
      "00:52:23.017 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "00:52:23.194 INFO  AnnotateIntervals - ------------------------------------------------------------\n",
      "00:52:23.198 INFO  AnnotateIntervals - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "00:52:23.198 INFO  AnnotateIntervals - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "00:52:23.199 INFO  AnnotateIntervals - Executing as root@6fffd5f5c26a on Linux v6.5.11-linuxkit amd64\n",
      "00:52:23.199 INFO  AnnotateIntervals - Java runtime: OpenJDK 64-Bit Server VM v17.0.9+9-Ubuntu-122.04\n",
      "00:52:23.199 INFO  AnnotateIntervals - Start Date/Time: January 20, 2024 at 12:52:22 AM GMT\n",
      "00:52:23.200 INFO  AnnotateIntervals - ------------------------------------------------------------\n",
      "00:52:23.200 INFO  AnnotateIntervals - ------------------------------------------------------------\n",
      "00:52:23.203 INFO  AnnotateIntervals - HTSJDK Version: 4.1.0\n",
      "00:52:23.203 INFO  AnnotateIntervals - Picard Version: 3.1.1\n",
      "00:52:23.203 INFO  AnnotateIntervals - Built for Spark Version: 3.5.0\n",
      "00:52:23.205 INFO  AnnotateIntervals - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "00:52:23.205 INFO  AnnotateIntervals - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "00:52:23.205 INFO  AnnotateIntervals - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "00:52:23.205 INFO  AnnotateIntervals - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "00:52:23.205 INFO  AnnotateIntervals - Deflater: IntelDeflater\n",
      "00:52:23.205 INFO  AnnotateIntervals - Inflater: IntelInflater\n",
      "00:52:23.206 INFO  AnnotateIntervals - GCS max retries/reopens: 20\n",
      "00:52:23.206 INFO  AnnotateIntervals - Requester pays: disabled\n",
      "00:52:23.207 INFO  AnnotateIntervals - Initializing engine\n",
      "00:52:34.055 INFO  FeatureManager - Using codec IntervalListCodec to read file file:///home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.interval_list\n",
      "00:52:35.075 INFO  IntervalArgumentCollection - Processing 87641000 bp from intervals\n",
      "00:52:35.112 INFO  AnnotateIntervals - Done initializing engine\n",
      "00:52:35.114 INFO  AnnotateIntervals - Loading intervals for annotation...\n",
      "00:52:35.114 INFO  AnnotateIntervals - Loading resources for annotation...\n",
      "00:52:45.403 INFO  AnnotateIntervals - Adding GC-content annotator...\n",
      "00:52:45.406 INFO  AnnotateIntervals - Annotating intervals...\n",
      "00:52:45.407 INFO  ProgressMeter - Starting traversal\n",
      "00:52:45.417 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Records Processed   Records/Minute\n",
      "00:53:01.755 INFO  ProgressMeter -        chr20:2059001              0.3                  2000           7352.5\n",
      "00:53:17.881 INFO  ProgressMeter -        chr20:4059001              0.5                  4000           7393.5\n",
      "00:53:36.241 INFO  ProgressMeter -        chr20:6059001              0.8                  6000           7084.1\n",
      "00:53:56.108 INFO  ProgressMeter -        chr20:8059001              1.2                  8000           6790.5\n",
      "00:54:12.312 INFO  ProgressMeter -       chr20:10059001              1.4                 10000           6905.2\n",
      "00:54:28.484 INFO  ProgressMeter -       chr20:12059001              1.7                 12000           6985.9\n",
      "00:54:44.546 INFO  ProgressMeter -       chr20:14059001              2.0                 14000           7051.3\n",
      "00:55:00.415 INFO  ProgressMeter -       chr20:16059001              2.2                 16000           7111.3\n",
      "00:55:16.349 INFO  ProgressMeter -       chr20:18059001              2.5                 18000           7155.6\n",
      "00:55:32.364 INFO  ProgressMeter -       chr20:20059001              2.8                 20000           7188.0\n",
      "00:55:48.212 INFO  ProgressMeter -       chr20:22059001              3.0                 22000           7221.3\n",
      "00:56:04.020 INFO  ProgressMeter -       chr20:24059001              3.3                 24000           7250.7\n",
      "00:56:19.883 INFO  ProgressMeter -       chr20:26059001              3.6                 26000           7273.9\n",
      "00:56:35.169 INFO  ProgressMeter -       chr20:27113001              3.8                 27000           7051.1\n",
      "00:56:50.157 INFO  ProgressMeter -       chr20:29137001              4.1                 29000           7109.6\n",
      "00:57:05.745 INFO  ProgressMeter -       chr20:31364001              4.3                 31000           7144.9\n",
      "00:57:21.659 INFO  ProgressMeter -       chr20:33364001              4.6                 33000           7167.8\n",
      "00:57:37.486 INFO  ProgressMeter -       chr20:35364001              4.9                 35000           7190.1\n",
      "00:57:53.214 INFO  ProgressMeter -       chr20:37364001              5.1                 37000           7212.6\n",
      "00:58:08.998 INFO  ProgressMeter -       chr20:39364001              5.4                 39000           7231.7\n",
      "00:58:24.939 INFO  ProgressMeter -       chr20:41364001              5.7                 41000           7245.6\n",
      "00:58:41.706 INFO  ProgressMeter -       chr20:43364001              5.9                 43000           7241.4\n",
      "00:58:57.642 INFO  ProgressMeter -       chr20:45364001              6.2                 45000           7253.7\n",
      "00:59:13.598 INFO  ProgressMeter -       chr20:47364001              6.5                 47000           7264.7\n",
      "00:59:29.754 INFO  ProgressMeter -       chr20:49364001              6.7                 49000           7271.2\n",
      "00:59:45.739 INFO  ProgressMeter -       chr20:51364001              7.0                 51000           7280.2\n",
      "01:00:01.795 INFO  ProgressMeter -       chr20:53364001              7.3                 53000           7287.3\n",
      "01:00:17.749 INFO  ProgressMeter -       chr20:55364001              7.5                 55000           7295.6\n",
      "01:00:33.741 INFO  ProgressMeter -       chr20:57364001              7.8                 57000           7302.7\n",
      "01:00:49.714 INFO  ProgressMeter -       chr20:59364001              8.1                 59000           7309.6\n",
      "01:01:05.529 INFO  ProgressMeter -       chr20:61364001              8.3                 61000           7318.4\n",
      "01:01:21.132 INFO  ProgressMeter -       chr20:63364001              8.6                 63000           7329.7\n",
      "01:01:31.909 INFO  ProgressMeter -         chrY:2810001              8.8                 64000           7293.6\n",
      "01:01:47.890 INFO  ProgressMeter -         chrY:4810001              9.0                 66000           7299.9\n",
      "01:02:03.698 INFO  ProgressMeter -         chrY:6810001              9.3                 68000           7308.2\n",
      "01:02:20.358 INFO  ProgressMeter -         chrY:8810001              9.6                 70000           7305.1\n",
      "01:02:36.478 INFO  ProgressMeter -        chrY:11083001              9.9                 72000           7308.9\n",
      "01:02:52.486 INFO  ProgressMeter -        chrY:13133001             10.1                 74000           7313.9\n",
      "01:03:09.748 INFO  ProgressMeter -        chrY:15133001             10.4                 76000           7304.0\n",
      "01:03:25.769 INFO  ProgressMeter -        chrY:17133001             10.7                 78000           7308.5\n",
      "01:03:41.660 INFO  ProgressMeter -        chrY:19133001             10.9                 80000           7314.4\n",
      "01:03:58.538 INFO  ProgressMeter -        chrY:21182001             11.2                 82000           7309.2\n",
      "01:04:14.495 INFO  ProgressMeter -        chrY:23198001             11.5                 84000           7314.1\n",
      "01:04:31.776 INFO  ProgressMeter -        chrY:25198001             11.8                 86000           7305.1\n",
      "01:21:04.119 INFO  ProgressMeter -        chrY:26198001             28.3                 87641           3095.6\n",
      "01:21:04.120 INFO  ProgressMeter - Traversal complete. Processed 87641 total records in 28.3 minutes.\n",
      "01:21:04.120 INFO  AnnotateIntervals - Writing annotated intervals to /home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.annotated.tsv...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:21:04.757 INFO  AnnotateIntervals - AnnotateIntervals complete.\n",
      "01:21:04.757 INFO  AnnotateIntervals - Shutting down engine\n",
      "[January 20, 2024 at 1:21:04 AM GMT] org.broadinstitute.hellbender.tools.copynumber.AnnotateIntervals done. Elapsed time: 28.70 minutes.\n",
      "Runtime.totalMemory()=169869312\n"
     ]
    }
   ],
   "source": [
    "! gatk AnnotateIntervals \\\n",
    "        -L $WORKSPACE_LOCAL/sandbox/chr20Y.interval_list \\\n",
    "        -R $REFERENCE_BUCKET/Homo_sapiens_assembly38.fasta \\\n",
    "        -imr OVERLAPPING_ONLY \\\n",
    "        -O $WORKSPACE_LOCAL/sandbox/chr20Y.annotated.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This produces a four-column table where the fourth column gives the fraction of GC content. Let's take a look:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CONTIG\tSTART\tEND\tGC_CONTENT\r\n",
      "chr20\t60001\t61000\t0.470000\r\n",
      "chr20\t61001\t62000\t0.473000\r\n",
      "chr20\t62001\t63000\t0.482000\r\n",
      "chr20\t63001\t64000\t0.469333\r\n",
      "chr20\t64001\t65000\t0.476000\r\n",
      "chr20\t65001\t66000\t0.477000\r\n",
      "chr20\t66001\t67000\t0.483333\r\n",
      "chr20\t67001\t68000\t0.399000\r\n",
      "chr20\t68001\t69000\t0.293000\r\n"
     ]
    }
   ],
   "source": [
    "! grep -v '^@' $WORKSPACE_LOCAL/sandbox/chr20Y.annotated.tsv | head -n 10"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Comments on select parameters of _AnnotateIntervals_\n",
    "\n",
    "- The tool requires the `-R` reference and the `-L` intervals. The tool calculates GC-content for the intervals using the reference. Although optional for the tool, we recommend annotating mappability by providing a `--mappability-track` regions file in either .bed or .bed.gz format. Be sure to merge any overlapping intervals beforehand. The tutorial omits use of this resource.\n",
    "\n",
    "- GATK recommends use of the the single-read mappability track, as the multi-read track requires much longer times to process. For example, the Hoffman lab at the University of Toronto provides human and mouse mappability BED files for various kmer lengths at https://bismap.hoffmanlab.org. The accompanying publication is titled [Umap and Bismap: quantifying genome and methylome mappability](https://doi.org/10.1093/nar/gky677).\n",
    "\n",
    "- Optionally and additionally, annotate segmental duplication content by providing a `--segmental-duplication-track` regions file in either .bed or .bed.gz format. \n",
    "\n",
    "- Exclude undesirable intervals with the `-XL` parameter, e.g. intervals corresponding to centromeric regions."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## _FilterIntervals_ based on GC-content and cohort extreme counts\n",
    "\n",
    "_FilterIntervals_ takes preprocessed intervals and either annotated intervals or read counts or both. It can also exclude intervals given with `-XL`. When given both types of data, the tool retains the intervals that intersect from filtering on each data type. The v4.1.0.0 `cnv_germline_cohort_workflow.wdl` pipeline script requires read counts files, and so by default the pipeline script always performs the _FilterIntervals_ step on read counts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /gatk/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /gatk/gatk-package-4.5.0.0-local.jar FilterIntervals -L /home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.interval_list --annotated-intervals /home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.annotated.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG00096.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG00268.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG00419.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG00759.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01051.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01112.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01500.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01565.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01583.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01595.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01879.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG02568.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG02922.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG03006.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG03052.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG03642.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG03742.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA18525.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA18939.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA19017.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA19625.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA19648.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA20502.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA20845.tsv -imr OVERLAPPING_ONLY -O /home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.cohort.gc.filtered.interval_list\n",
      "01:27:03.421 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "01:27:03.701 INFO  FilterIntervals - ------------------------------------------------------------\n",
      "01:27:03.704 INFO  FilterIntervals - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "01:27:03.704 INFO  FilterIntervals - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "01:27:03.704 INFO  FilterIntervals - Executing as root@6fffd5f5c26a on Linux v6.5.11-linuxkit amd64\n",
      "01:27:03.704 INFO  FilterIntervals - Java runtime: OpenJDK 64-Bit Server VM v17.0.9+9-Ubuntu-122.04\n",
      "01:27:03.705 INFO  FilterIntervals - Start Date/Time: January 20, 2024 at 1:27:03 AM GMT\n",
      "01:27:03.705 INFO  FilterIntervals - ------------------------------------------------------------\n",
      "01:27:03.705 INFO  FilterIntervals - ------------------------------------------------------------\n",
      "01:27:03.707 INFO  FilterIntervals - HTSJDK Version: 4.1.0\n",
      "01:27:03.707 INFO  FilterIntervals - Picard Version: 3.1.1\n",
      "01:27:03.707 INFO  FilterIntervals - Built for Spark Version: 3.5.0\n",
      "01:27:03.708 INFO  FilterIntervals - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "01:27:03.708 INFO  FilterIntervals - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "01:27:03.709 INFO  FilterIntervals - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "01:27:03.709 INFO  FilterIntervals - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "01:27:03.709 INFO  FilterIntervals - Deflater: IntelDeflater\n",
      "01:27:03.709 INFO  FilterIntervals - Inflater: IntelInflater\n",
      "01:27:03.709 INFO  FilterIntervals - GCS max retries/reopens: 20\n",
      "01:27:03.709 INFO  FilterIntervals - Requester pays: disabled\n",
      "01:27:03.710 INFO  FilterIntervals - Initializing engine\n",
      "01:27:03.710 INFO  FilterIntervals - Done initializing engine\n",
      "01:27:04.763 INFO  FeatureManager - Using codec IntervalListCodec to read file file:///home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.interval_list\n",
      "01:27:05.513 INFO  IntervalArgumentCollection - Processing 87641000 bp from intervals\n",
      "01:27:05.650 INFO  FilterIntervals - After interval resolution, 87641 intervals remain...\n",
      "01:27:05.650 INFO  FilterIntervals - After interval intersection, 87641 intervals remain...\n",
      "01:27:05.762 INFO  FilterIntervals - Applying annotation-based filters...\n",
      "01:27:05.798 INFO  FilterIntervals - After applying GC-content filter (intervals with values outside of [0.1, 0.9] fail), 87636 / 87641 intervals remain...\n",
      "01:27:05.799 INFO  FilterIntervals - Validating and aggregating input read-counts files...\n",
      "01:27:05.828 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG00096.tsv (1 / 24)\n",
      "01:27:06.471 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG00268.tsv (2 / 24)\n",
      "01:27:06.802 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG00419.tsv (3 / 24)\n",
      "01:27:07.075 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG00759.tsv (4 / 24)\n",
      "01:27:07.538 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01051.tsv (5 / 24)\n",
      "01:27:07.922 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01112.tsv (6 / 24)\n",
      "01:27:08.167 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01500.tsv (7 / 24)\n",
      "01:27:08.412 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01565.tsv (8 / 24)\n",
      "01:27:08.700 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01583.tsv (9 / 24)\n",
      "01:27:09.162 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01595.tsv (10 / 24)\n",
      "01:27:09.398 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG01879.tsv (11 / 24)\n",
      "01:27:09.663 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG02568.tsv (12 / 24)\n",
      "01:27:09.953 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG02922.tsv (13 / 24)\n",
      "01:27:10.214 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG03006.tsv (14 / 24)\n",
      "01:27:10.462 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG03052.tsv (15 / 24)\n",
      "01:27:10.805 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG03642.tsv (16 / 24)\n",
      "01:27:11.083 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/HG03742.tsv (17 / 24)\n",
      "01:27:11.320 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA18525.tsv (18 / 24)\n",
      "01:27:11.563 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA18939.tsv (19 / 24)\n",
      "01:27:11.817 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA19017.tsv (20 / 24)\n",
      "01:27:12.104 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA19625.tsv (21 / 24)\n",
      "01:27:12.369 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA19648.tsv (22 / 24)\n",
      "01:27:12.626 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA20502.tsv (23 / 24)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:27:12.902 INFO  FilterIntervals - Aggregating read-counts file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA20845.tsv (24 / 24)\n",
      "01:27:13.189 INFO  FilterIntervals - Applying count-based filters...\n",
      "01:27:13.278 INFO  FilterIntervals - After applying low-count filter (intervals with a count < 10 in > 50.0% of samples fail), 62824 / 87641 intervals remain...\n",
      "01:27:13.450 INFO  FilterIntervals - After applying extreme-count filter (intervals with a count percentile outside of [1.0, 99.0] in > 90.0% of samples fail), 62398 / 87641 intervals remain...\n",
      "01:27:13.487 INFO  FilterIntervals - 62398 / 87641 intervals passed all filters...\n",
      "01:27:13.513 INFO  FilterIntervals - Writing filtered intervals to /home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.cohort.gc.filtered.interval_list...\n",
      "01:27:13.652 INFO  FilterIntervals - FilterIntervals complete.\n",
      "01:27:13.653 INFO  FilterIntervals - Shutting down engine\n",
      "[January 20, 2024 at 1:27:13 AM GMT] org.broadinstitute.hellbender.tools.copynumber.FilterIntervals done. Elapsed time: 0.17 minutes.\n",
      "Runtime.totalMemory()=829423616\n"
     ]
    }
   ],
   "source": [
    "! gatk FilterIntervals \\\n",
    "        -L $WORKSPACE_LOCAL/sandbox/chr20Y.interval_list \\\n",
    "        --annotated-intervals $WORKSPACE_LOCAL/sandbox/chr20Y.annotated.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG00096.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG00268.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG00419.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG00759.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG01051.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG01112.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG01500.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG01565.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG01583.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG01595.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG01879.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG02568.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG02922.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG03006.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG03052.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG03642.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/HG03742.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18525.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18939.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19017.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19625.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19648.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20502.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20845.tsv \\\n",
    "        -imr OVERLAPPING_ONLY \\\n",
    "        -O $WORKSPACE_LOCAL/sandbox/chr20Y.cohort.gc.filtered.interval_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "87641\n",
      "62398\n"
     ]
    }
   ],
   "source": [
    "! grep -v '^@' $WORKSPACE_LOCAL/sandbox/chr20Y.interval_list | wc -l\n",
    "! grep -v '^@' $WORKSPACE_LOCAL/sandbox/chr20Y.cohort.gc.filtered.interval_list | wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This produces a Picard-style intervals list containing a subset of the starting intervals (80,534 of 87,641)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Comments on select parameters of _FilterIntervals_\n",
    "\n",
    "- The tool requires the preprocessed intervals, provided with `-L`, from the previous section. Given annotated intervals with `--annotated-intervals`, the tool filters intervals on the given annotation(s).\n",
    "\n",
    "- GC-content thresholds are set by `--minimum-gc-content` and `--maximum-gc-content`, where defaults are `0.1` and `0.9`, respectively.\n",
    "\n",
    "- Mappability thresholds are set by `--minimum-mappability` and `--maximum-mappability`. Defaults are `0.9` and `1.0`, respectively.\n",
    "\n",
    "- Segmental duplication content thresholds are set by `--minimum-segmental-duplication-content` and `--maximum-segmental-duplication-content`. Defaults are 0.0 and 0.5, respectively.\n",
    "\n",
    "- Given read counts files, each with `-I` and in either HDF5 or TSV format, the tool filters intervals on low and extreme read counts with the following tunable thresholds.\n",
    "    - `--low-count-filter-count-threshold` default is `5`\n",
    "    - `--low-count-filter-percentage-of-samples` default is `90.0`\n",
    "    - `--extreme-count-filter-minimum-percentile` default is `1.0`\n",
    "    - `--extreme-count-filter-maximum-percentile` default is `99.0`\n",
    "    - `--extreme-count-filter-percentage-of-samples` default is `90.0`    \n",
    "The read counts data must match each other in intervals. For the default parameters, the tool first filters intervals with a count less than `5` in greater than `90%` of the samples. The tool then filters the remaining intervals with a count percentile less than `1` or greater than `99` in a percentage of samples greater than `90%`. These parameters effectively exclude intervals where all samples have extreme outlier counts, e.g. are deleted.\n",
    "\n",
    "- To disable counts based filtering, omit the read counts or, e.g. when using the v4.1.0.0 _cnv_germline_cohort_workflow.wdl_ pipeline script, set the two percentage-of-samples parameters as follows.\n",
    "    - `--low-count-filter-percentage-of-samples 100`\n",
    "    - `--extreme-count-filter-percentage-of-samples 100`    \n",
    "\n",
    "- Provide intervals to exclude from analysis with `--exclude-intervals` or `-XL`, e.g. centromeric regions. A frugal strategy is to collect read counts using the entirety of intervals and then to exclude undesirable regions later at the _FilterIntervals_ step (section 3), the _DetermineGermlineContigPloidy_ step (section 4), at the _GermlineCNVCaller_ step (section 6) and/or post-calling."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Call autosomal and allosomal contig ploidy with _DetermineGermlineContigPloidy_\n",
    "\n",
    "_DetermineGermlineContigPloidy_ calls contig level ploidies (i.e. copy number) for both autosomal, e.g. human chr20, and allosomal contigs, e.g. human chrX. The tool determines baseline contig ploidies using sample coverages and contig ploidy priors that give the prior probabilities for each ploidy state for each contig. In this process, the tool generates global baseline coverage and noise data _GermlineCNVCaller_ will use in section 6.\n",
    "\n",
    "The tool determines baseline contig ploidies using the total read count per contig. Researchers should consider the impact of this for their data. For example, for the tutorial WGS data, the contribution of the PAR regions to total coverage counts on chrX is small and the tool correctly calls allosomal ploidies. However, consider blacklisting PAR regions for data where the contribution is disporportionate, e.g. targeted panels."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## _DetermineGermlineContigPloidy_ in cohort mode\n",
    "\n",
    "The cohort mode requires a `--contig-ploidy-priors` table and produces a ploidy model. The ploidy prior table specifies the prior probability of various ploidy states is formatted as follows:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cat $TUTORIAL_DATA_PATH/chr20XY_contig_ploidy_priors.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "According to this table, we put 98% prior probability to call chr20 as diploid (PLOIDY_PRIOR_2), 0% prior probability to call it homozygous deletion (would you be alive without chr20?), and a small probability 1% each for heterozygous deletion (PLOIDY_PRIOR_1) and trisomy (PLOIDY_PRIOR_3).\n",
    "\n",
    "  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /gatk/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /gatk/gatk-package-4.5.0.0-local.jar DetermineGermlineContigPloidy -L /home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.cohort.gc.filtered.interval_list --interval-merging-rule OVERLAPPING_ONLY -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA18525.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA18939.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA19017.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA19625.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA19648.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA20502.tsv -I /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA20845.tsv --contig-ploidy-priors /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/chr20XY_contig_ploidy_priors.tsv --max-training-epochs 20 --num-thermal-advi-iters 1000 --max-advi-iter-subsequent-epochs 500 --output /home/jupyter/notebooks/germline-cnv/sandbox --output-prefix ploidy-cohort7 --verbosity DEBUG\n",
      "01:27:32.913 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "01:27:32.942 DEBUG NativeLibraryLoader - Extracting libgkl_compression.so to /tmp/libgkl_compression2714745363503502976.so\n",
      "01:27:33.082 INFO  DetermineGermlineContigPloidy - ------------------------------------------------------------\n",
      "01:27:33.086 INFO  DetermineGermlineContigPloidy - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "01:27:33.086 INFO  DetermineGermlineContigPloidy - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "01:27:33.086 INFO  DetermineGermlineContigPloidy - Executing as root@6fffd5f5c26a on Linux v6.5.11-linuxkit amd64\n",
      "01:27:33.086 INFO  DetermineGermlineContigPloidy - Java runtime: OpenJDK 64-Bit Server VM v17.0.9+9-Ubuntu-122.04\n",
      "01:27:33.087 INFO  DetermineGermlineContigPloidy - Start Date/Time: January 20, 2024 at 1:27:32 AM GMT\n",
      "01:27:33.087 INFO  DetermineGermlineContigPloidy - ------------------------------------------------------------\n",
      "01:27:33.087 INFO  DetermineGermlineContigPloidy - ------------------------------------------------------------\n",
      "01:27:33.089 INFO  DetermineGermlineContigPloidy - HTSJDK Version: 4.1.0\n",
      "01:27:33.089 INFO  DetermineGermlineContigPloidy - Picard Version: 3.1.1\n",
      "01:27:33.089 INFO  DetermineGermlineContigPloidy - Built for Spark Version: 3.5.0\n",
      "01:27:33.095 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.BUFFER_SIZE : 131072\n",
      "01:27:33.095 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "01:27:33.095 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.CREATE_INDEX : false\n",
      "01:27:33.095 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.CREATE_MD5 : false\n",
      "01:27:33.095 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.CUSTOM_READER_FACTORY : \n",
      "01:27:33.095 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.DISABLE_SNAPPY_COMPRESSOR : false\n",
      "01:27:33.095 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.EBI_REFERENCE_SERVICE_URL_MASK : https://www.ebi.ac.uk/ena/cram/md5/%s\n",
      "01:27:33.095 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.NON_ZERO_BUFFER_SIZE : 131072\n",
      "01:27:33.095 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.REFERENCE_FASTA : null\n",
      "01:27:33.095 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.SAM_FLAG_FIELD_FORMAT : DECIMAL\n",
      "01:27:33.096 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "01:27:33.096 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "01:27:33.096 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "01:27:33.096 INFO  DetermineGermlineContigPloidy - HTSJDK Defaults.USE_CRAM_REF_DOWNLOAD : false\n",
      "01:27:33.096 DEBUG ConfigFactory - Configuration file values: \n",
      "01:27:33.104 DEBUG ConfigFactory - \tgcsMaxRetries = 20\n",
      "01:27:33.104 DEBUG ConfigFactory - \tgcsProjectForRequesterPays = \n",
      "01:27:33.104 DEBUG ConfigFactory - \tgatk_stacktrace_on_user_exception = false\n",
      "01:27:33.104 DEBUG ConfigFactory - \tsamjdk.use_async_io_read_samtools = false\n",
      "01:27:33.104 DEBUG ConfigFactory - \tsamjdk.use_async_io_write_samtools = true\n",
      "01:27:33.105 DEBUG ConfigFactory - \tsamjdk.use_async_io_write_tribble = false\n",
      "01:27:33.105 DEBUG ConfigFactory - \tsamjdk.compression_level = 2\n",
      "01:27:33.105 DEBUG ConfigFactory - \tspark.kryoserializer.buffer.max = 512m\n",
      "01:27:33.105 DEBUG ConfigFactory - \tspark.driver.maxResultSize = 0\n",
      "01:27:33.105 DEBUG ConfigFactory - \tspark.driver.userClassPathFirst = true\n",
      "01:27:33.105 DEBUG ConfigFactory - \tspark.io.compression.codec = lzf\n",
      "01:27:33.105 DEBUG ConfigFactory - \tspark.executor.memoryOverhead = 600\n",
      "01:27:33.105 DEBUG ConfigFactory - \tspark.driver.extraJavaOptions = \n",
      "01:27:33.105 DEBUG ConfigFactory - \tspark.executor.extraJavaOptions = \n",
      "01:27:33.105 DEBUG ConfigFactory - \tcodec_packages = [htsjdk.variant, htsjdk.tribble, org.broadinstitute.hellbender.utils.codecs]\n",
      "01:27:33.105 DEBUG ConfigFactory - \tread_filter_packages = [org.broadinstitute.hellbender.engine.filters]\n",
      "01:27:33.106 DEBUG ConfigFactory - \tannotation_packages = [org.broadinstitute.hellbender.tools.walkers.annotator]\n",
      "01:27:33.106 DEBUG ConfigFactory - \tcloudPrefetchBuffer = 40\n",
      "01:27:33.106 DEBUG ConfigFactory - \tcloudIndexPrefetchBuffer = -1\n",
      "01:27:33.106 DEBUG ConfigFactory - \tcreateOutputBamIndex = true\n",
      "01:27:33.106 INFO  DetermineGermlineContigPloidy - Deflater: IntelDeflater\n",
      "01:27:33.106 INFO  DetermineGermlineContigPloidy - Inflater: IntelInflater\n",
      "01:27:33.106 INFO  DetermineGermlineContigPloidy - GCS max retries/reopens: 20\n",
      "01:27:33.106 INFO  DetermineGermlineContigPloidy - Requester pays: disabled\n",
      "01:27:33.108 INFO  DetermineGermlineContigPloidy - Initializing engine\n",
      "01:27:33.111 DEBUG ScriptExecutor - Executing:\n",
      "01:27:33.112 DEBUG ScriptExecutor -   python\n",
      "01:27:33.112 DEBUG ScriptExecutor -   -c\n",
      "01:27:33.112 DEBUG ScriptExecutor -   import gcnvkernel\n",
      "\n",
      "01:28:28.592 DEBUG ScriptExecutor - Result: 0\n",
      "01:28:28.597 INFO  DetermineGermlineContigPloidy - Done initializing engine\n",
      "01:28:28.610 INFO  DetermineGermlineContigPloidy - No contig-ploidy model was provided, running in cohort mode...\n",
      "01:28:28.610 INFO  DetermineGermlineContigPloidy - Intervals specified...\n",
      "01:28:29.295 DEBUG GenomeLocParser - Prepared reference sequence contig dictionary\n",
      "01:28:29.298 DEBUG GenomeLocParser -  chr1 (248956422 bp)\n",
      "01:28:29.298 DEBUG GenomeLocParser -  chr2 (242193529 bp)\n",
      "01:28:29.298 DEBUG GenomeLocParser -  chr3 (198295559 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr4 (190214555 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr5 (181538259 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr6 (170805979 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr7 (159345973 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr8 (145138636 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr9 (138394717 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr10 (133797422 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr11 (135086622 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr12 (133275309 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr13 (114364328 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr14 (107043718 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr15 (101991189 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr16 (90338345 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr17 (83257441 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr18 (80373285 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr19 (58617616 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr20 (64444167 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr21 (46709983 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chr22 (50818468 bp)\n",
      "01:28:29.299 DEBUG GenomeLocParser -  chrX (156040895 bp)\n",
      "01:28:29.300 DEBUG GenomeLocParser -  chrY (57227415 bp)\n",
      "01:28:29.300 DEBUG GenomeLocParser -  chrM (16569 bp)\n",
      "01:28:29.300 DEBUG GenomeLocParser -  chr1_KI270706v1_random (175055 bp)\n",
      "01:28:29.300 DEBUG GenomeLocParser -  chr1_KI270707v1_random (32032 bp)\n",
      "01:28:29.300 DEBUG GenomeLocParser -  chr1_KI270708v1_random (127682 bp)\n",
      "01:28:29.300 DEBUG GenomeLocParser -  chr1_KI270709v1_random (66860 bp)\n",
      "01:28:29.300 DEBUG GenomeLocParser -  chr1_KI270710v1_random (40176 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr1_KI270711v1_random (42210 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr1_KI270712v1_random (176043 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr1_KI270713v1_random (40745 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr1_KI270714v1_random (41717 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr2_KI270715v1_random (161471 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr2_KI270716v1_random (153799 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr3_GL000221v1_random (155397 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr4_GL000008v2_random (209709 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr5_GL000208v1_random (92689 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr9_KI270717v1_random (40062 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr9_KI270718v1_random (38054 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr9_KI270719v1_random (176845 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr9_KI270720v1_random (39050 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr11_KI270721v1_random (100316 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr14_GL000009v2_random (201709 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr14_GL000225v1_random (211173 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr14_KI270722v1_random (194050 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr14_GL000194v1_random (191469 bp)\n",
      "01:28:29.301 DEBUG GenomeLocParser -  chr14_KI270723v1_random (38115 bp)\n",
      "01:28:29.302 DEBUG GenomeLocParser -  chr14_KI270724v1_random (39555 bp)\n",
      "01:28:29.302 DEBUG GenomeLocParser -  chr14_KI270725v1_random (172810 bp)\n",
      "01:28:29.302 DEBUG GenomeLocParser -  chr14_KI270726v1_random (43739 bp)\n",
      "01:28:29.302 DEBUG GenomeLocParser -  chr15_KI270727v1_random (448248 bp)\n",
      "01:28:29.302 DEBUG GenomeLocParser -  chr16_KI270728v1_random (1872759 bp)\n",
      "01:28:29.302 DEBUG GenomeLocParser -  chr17_GL000205v2_random (185591 bp)\n",
      "01:28:29.302 DEBUG GenomeLocParser -  chr17_KI270729v1_random (280839 bp)\n",
      "01:28:29.302 DEBUG GenomeLocParser -  chr17_KI270730v1_random (112551 bp)\n",
      "01:28:29.302 DEBUG GenomeLocParser -  chr22_KI270731v1_random (150754 bp)\n",
      "01:28:29.302 DEBUG GenomeLocParser -  chr22_KI270732v1_random (41543 bp)\n",
      "01:28:29.302 DEBUG GenomeLocParser -  chr22_KI270733v1_random (179772 bp)\n",
      "01:28:29.302 DEBUG GenomeLocParser -  chr22_KI270734v1_random (165050 bp)\n",
      "01:28:29.303 DEBUG GenomeLocParser -  chr22_KI270735v1_random (42811 bp)\n",
      "01:28:29.303 DEBUG GenomeLocParser -  chr22_KI270736v1_random (181920 bp)\n",
      "01:28:29.304 DEBUG GenomeLocParser -  chr22_KI270737v1_random (103838 bp)\n",
      "01:28:29.304 DEBUG GenomeLocParser -  chr22_KI270738v1_random (99375 bp)\n",
      "01:28:29.304 DEBUG GenomeLocParser -  chr22_KI270739v1_random (73985 bp)\n",
      "01:28:29.305 DEBUG GenomeLocParser -  chrY_KI270740v1_random (37240 bp)\n",
      "01:28:29.305 DEBUG GenomeLocParser -  chrUn_KI270302v1 (2274 bp)\n",
      "01:28:29.305 DEBUG GenomeLocParser -  chrUn_KI270304v1 (2165 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270303v1 (1942 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270305v1 (1472 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270322v1 (21476 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270320v1 (4416 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270310v1 (1201 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270316v1 (1444 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270315v1 (2276 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270312v1 (998 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270311v1 (12399 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270317v1 (37690 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270412v1 (1179 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270411v1 (2646 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270414v1 (2489 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270419v1 (1029 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270418v1 (2145 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270420v1 (2321 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270424v1 (2140 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270417v1 (2043 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270422v1 (1445 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270423v1 (981 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270425v1 (1884 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270429v1 (1361 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270442v1 (392061 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270466v1 (1233 bp)\n",
      "01:28:29.306 DEBUG GenomeLocParser -  chrUn_KI270465v1 (1774 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270467v1 (3920 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270435v1 (92983 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270438v1 (112505 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270468v1 (4055 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270510v1 (2415 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270509v1 (2318 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270518v1 (2186 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270508v1 (1951 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270516v1 (1300 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270512v1 (22689 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270519v1 (138126 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270522v1 (5674 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270511v1 (8127 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270515v1 (6361 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270507v1 (5353 bp)\n",
      "01:28:29.307 DEBUG GenomeLocParser -  chrUn_KI270517v1 (3253 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270529v1 (1899 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270528v1 (2983 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270530v1 (2168 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270539v1 (993 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270538v1 (91309 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270544v1 (1202 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270548v1 (1599 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270583v1 (1400 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270587v1 (2969 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270580v1 (1553 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270581v1 (7046 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270579v1 (31033 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270589v1 (44474 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270590v1 (4685 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270584v1 (4513 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270582v1 (6504 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270588v1 (6158 bp)\n",
      "01:28:29.308 DEBUG GenomeLocParser -  chrUn_KI270593v1 (3041 bp)\n",
      "01:28:29.309 DEBUG GenomeLocParser -  chrUn_KI270591v1 (5796 bp)\n",
      "01:28:29.309 DEBUG GenomeLocParser -  chrUn_KI270330v1 (1652 bp)\n",
      "01:28:29.309 DEBUG GenomeLocParser -  chrUn_KI270329v1 (1040 bp)\n",
      "01:28:29.309 DEBUG GenomeLocParser -  chrUn_KI270334v1 (1368 bp)\n",
      "01:28:29.309 DEBUG GenomeLocParser -  chrUn_KI270333v1 (2699 bp)\n",
      "01:28:29.309 DEBUG GenomeLocParser -  chrUn_KI270335v1 (1048 bp)\n",
      "01:28:29.310 DEBUG GenomeLocParser -  chrUn_KI270338v1 (1428 bp)\n",
      "01:28:29.310 DEBUG GenomeLocParser -  chrUn_KI270340v1 (1428 bp)\n",
      "01:28:29.310 DEBUG GenomeLocParser -  chrUn_KI270336v1 (1026 bp)\n",
      "01:28:29.310 DEBUG GenomeLocParser -  chrUn_KI270337v1 (1121 bp)\n",
      "01:28:29.310 DEBUG GenomeLocParser -  chrUn_KI270363v1 (1803 bp)\n",
      "01:28:29.310 DEBUG GenomeLocParser -  chrUn_KI270364v1 (2855 bp)\n",
      "01:28:29.310 DEBUG GenomeLocParser -  chrUn_KI270362v1 (3530 bp)\n",
      "01:28:29.310 DEBUG GenomeLocParser -  chrUn_KI270366v1 (8320 bp)\n",
      "01:28:29.310 DEBUG GenomeLocParser -  chrUn_KI270378v1 (1048 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270379v1 (1045 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270389v1 (1298 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270390v1 (2387 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270387v1 (1537 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270395v1 (1143 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270396v1 (1880 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270388v1 (1216 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270394v1 (970 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270386v1 (1788 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270391v1 (1484 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270383v1 (1750 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270393v1 (1308 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270384v1 (1658 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270392v1 (971 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270381v1 (1930 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270385v1 (990 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270382v1 (4215 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270376v1 (1136 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270374v1 (2656 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270372v1 (1650 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270373v1 (1451 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270375v1 (2378 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270371v1 (2805 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270448v1 (7992 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_KI270521v1 (7642 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_GL000195v1 (182896 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_GL000219v1 (179198 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_GL000220v1 (161802 bp)\n",
      "01:28:29.311 DEBUG GenomeLocParser -  chrUn_GL000224v1 (179693 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270741v1 (157432 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_GL000226v1 (15008 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_GL000213v1 (164239 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270743v1 (210658 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270744v1 (168472 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270745v1 (41891 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270746v1 (66486 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270747v1 (198735 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270748v1 (93321 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270749v1 (158759 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270750v1 (148850 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270751v1 (150742 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270752v1 (27745 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270753v1 (62944 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270754v1 (40191 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270755v1 (36723 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270756v1 (79590 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270757v1 (71251 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_GL000214v1 (137718 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_KI270742v1 (186739 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_GL000216v2 (176608 bp)\n",
      "01:28:29.312 DEBUG GenomeLocParser -  chrUn_GL000218v1 (161147 bp)\n",
      "01:28:29.314 DEBUG GenomeLocParser -  chr1_KI270762v1_alt (354444 bp)\n",
      "01:28:29.315 DEBUG GenomeLocParser -  chr1_KI270766v1_alt (256271 bp)\n",
      "01:28:29.315 DEBUG GenomeLocParser -  chr1_KI270760v1_alt (109528 bp)\n",
      "01:28:29.315 DEBUG GenomeLocParser -  chr1_KI270765v1_alt (185285 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr1_GL383518v1_alt (182439 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr1_GL383519v1_alt (110268 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr1_GL383520v2_alt (366580 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr1_KI270764v1_alt (50258 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr1_KI270763v1_alt (911658 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr1_KI270759v1_alt (425601 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr1_KI270761v1_alt (165834 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_KI270770v1_alt (136240 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_KI270773v1_alt (70887 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_KI270774v1_alt (223625 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_KI270769v1_alt (120616 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_GL383521v1_alt (143390 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_KI270772v1_alt (133041 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_KI270775v1_alt (138019 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_KI270771v1_alt (110395 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_KI270768v1_alt (110099 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_GL582966v2_alt (96131 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_GL383522v1_alt (123821 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_KI270776v1_alt (174166 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr2_KI270767v1_alt (161578 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr3_JH636055v2_alt (173151 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr3_KI270783v1_alt (109187 bp)\n",
      "01:28:29.316 DEBUG GenomeLocParser -  chr3_KI270780v1_alt (224108 bp)\n",
      "01:28:29.317 DEBUG GenomeLocParser -  chr3_GL383526v1_alt (180671 bp)\n",
      "01:28:29.317 DEBUG GenomeLocParser -  chr3_KI270777v1_alt (173649 bp)\n",
      "01:28:29.317 DEBUG GenomeLocParser -  chr3_KI270778v1_alt (248252 bp)\n",
      "01:28:29.318 DEBUG GenomeLocParser -  chr3_KI270781v1_alt (113034 bp)\n",
      "01:28:29.318 DEBUG GenomeLocParser -  chr3_KI270779v1_alt (205312 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr3_KI270782v1_alt (162429 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr3_KI270784v1_alt (184404 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr4_KI270790v1_alt (220246 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr4_GL383528v1_alt (376187 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr4_KI270787v1_alt (111943 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr4_GL000257v2_alt (586476 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr4_KI270788v1_alt (158965 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr4_GL383527v1_alt (164536 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr4_KI270785v1_alt (119912 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr4_KI270789v1_alt (205944 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr4_KI270786v1_alt (244096 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr5_KI270793v1_alt (126136 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr5_KI270792v1_alt (179043 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr5_KI270791v1_alt (195710 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr5_GL383532v1_alt (82728 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr5_GL949742v1_alt (226852 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr5_KI270794v1_alt (164558 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr5_GL339449v2_alt (1612928 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr5_GL383530v1_alt (101241 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr5_KI270796v1_alt (172708 bp)\n",
      "01:28:29.319 DEBUG GenomeLocParser -  chr5_GL383531v1_alt (173459 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr5_KI270795v1_alt (131892 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr6_GL000250v2_alt (4672374 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr6_KI270800v1_alt (175808 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr6_KI270799v1_alt (152148 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr6_GL383533v1_alt (124736 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr6_KI270801v1_alt (870480 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr6_KI270802v1_alt (75005 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr6_KB021644v2_alt (185823 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr6_KI270797v1_alt (197536 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr6_KI270798v1_alt (271782 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr7_KI270804v1_alt (157952 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr7_KI270809v1_alt (209586 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr7_KI270806v1_alt (158166 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr7_GL383534v2_alt (119183 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr7_KI270803v1_alt (1111570 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr7_KI270808v1_alt (271455 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr7_KI270807v1_alt (126434 bp)\n",
      "01:28:29.320 DEBUG GenomeLocParser -  chr7_KI270805v1_alt (209988 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270818v1_alt (145606 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270812v1_alt (282736 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270811v1_alt (292436 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270821v1_alt (985506 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270813v1_alt (300230 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270822v1_alt (624492 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270814v1_alt (141812 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270810v1_alt (374415 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270819v1_alt (133535 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270820v1_alt (36640 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270817v1_alt (158983 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270816v1_alt (305841 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr8_KI270815v1_alt (132244 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr9_GL383539v1_alt (162988 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr9_GL383540v1_alt (71551 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr9_GL383541v1_alt (171286 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr9_GL383542v1_alt (60032 bp)\n",
      "01:28:29.321 DEBUG GenomeLocParser -  chr9_KI270823v1_alt (439082 bp)\n",
      "01:28:29.322 DEBUG GenomeLocParser -  chr10_GL383545v1_alt (179254 bp)\n",
      "01:28:29.322 DEBUG GenomeLocParser -  chr10_KI270824v1_alt (181496 bp)\n",
      "01:28:29.322 DEBUG GenomeLocParser -  chr10_GL383546v1_alt (309802 bp)\n",
      "01:28:29.322 DEBUG GenomeLocParser -  chr10_KI270825v1_alt (188315 bp)\n",
      "01:28:29.322 DEBUG GenomeLocParser -  chr11_KI270832v1_alt (210133 bp)\n",
      "01:28:29.322 DEBUG GenomeLocParser -  chr11_KI270830v1_alt (177092 bp)\n",
      "01:28:29.323 DEBUG GenomeLocParser -  chr11_KI270831v1_alt (296895 bp)\n",
      "01:28:29.324 DEBUG GenomeLocParser -  chr11_KI270829v1_alt (204059 bp)\n",
      "01:28:29.324 DEBUG GenomeLocParser -  chr11_GL383547v1_alt (154407 bp)\n",
      "01:28:29.325 DEBUG GenomeLocParser -  chr11_JH159136v1_alt (200998 bp)\n",
      "01:28:29.325 DEBUG GenomeLocParser -  chr11_JH159137v1_alt (191409 bp)\n",
      "01:28:29.325 DEBUG GenomeLocParser -  chr11_KI270827v1_alt (67707 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr11_KI270826v1_alt (186169 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr12_GL877875v1_alt (167313 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr12_GL877876v1_alt (408271 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr12_KI270837v1_alt (40090 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr12_GL383549v1_alt (120804 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr12_KI270835v1_alt (238139 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr12_GL383550v2_alt (169178 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr12_GL383552v1_alt (138655 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr12_GL383553v2_alt (152874 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr12_KI270834v1_alt (119498 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr12_GL383551v1_alt (184319 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr12_KI270833v1_alt (76061 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr12_KI270836v1_alt (56134 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr13_KI270840v1_alt (191684 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr13_KI270839v1_alt (180306 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr13_KI270843v1_alt (103832 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr13_KI270841v1_alt (169134 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr13_KI270838v1_alt (306913 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr13_KI270842v1_alt (37287 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr14_KI270844v1_alt (322166 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr14_KI270847v1_alt (1511111 bp)\n",
      "01:28:29.326 DEBUG GenomeLocParser -  chr14_KI270845v1_alt (180703 bp)\n",
      "01:28:29.327 DEBUG GenomeLocParser -  chr14_KI270846v1_alt (1351393 bp)\n",
      "01:28:29.327 DEBUG GenomeLocParser -  chr15_KI270852v1_alt (478999 bp)\n",
      "01:28:29.327 DEBUG GenomeLocParser -  chr15_KI270851v1_alt (263054 bp)\n",
      "01:28:29.327 DEBUG GenomeLocParser -  chr15_KI270848v1_alt (327382 bp)\n",
      "01:28:29.327 DEBUG GenomeLocParser -  chr15_GL383554v1_alt (296527 bp)\n",
      "01:28:29.328 DEBUG GenomeLocParser -  chr15_KI270849v1_alt (244917 bp)\n",
      "01:28:29.328 DEBUG GenomeLocParser -  chr15_GL383555v2_alt (388773 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr15_KI270850v1_alt (430880 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr16_KI270854v1_alt (134193 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr16_KI270856v1_alt (63982 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr16_KI270855v1_alt (232857 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr16_KI270853v1_alt (2659700 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr16_GL383556v1_alt (192462 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr16_GL383557v1_alt (89672 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr17_GL383563v3_alt (375691 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr17_KI270862v1_alt (391357 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr17_KI270861v1_alt (196688 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr17_KI270857v1_alt (2877074 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr17_JH159146v1_alt (278131 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr17_JH159147v1_alt (70345 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr17_GL383564v2_alt (133151 bp)\n",
      "01:28:29.329 DEBUG GenomeLocParser -  chr17_GL000258v2_alt (1821992 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr17_GL383565v1_alt (223995 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr17_KI270858v1_alt (235827 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr17_KI270859v1_alt (108763 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr17_GL383566v1_alt (90219 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr17_KI270860v1_alt (178921 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr18_KI270864v1_alt (111737 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr18_GL383567v1_alt (289831 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr18_GL383570v1_alt (164789 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr18_GL383571v1_alt (198278 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr18_GL383568v1_alt (104552 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr18_GL383569v1_alt (167950 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr18_GL383572v1_alt (159547 bp)\n",
      "01:28:29.330 DEBUG GenomeLocParser -  chr18_KI270863v1_alt (167999 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr19_KI270868v1_alt (61734 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr19_KI270865v1_alt (52969 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr19_GL383573v1_alt (385657 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr19_GL383575v2_alt (170222 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr19_GL383576v1_alt (188024 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr19_GL383574v1_alt (155864 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr19_KI270866v1_alt (43156 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr19_KI270867v1_alt (233762 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr19_GL949746v1_alt (987716 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr20_GL383577v2_alt (128386 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr20_KI270869v1_alt (118774 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr20_KI270871v1_alt (58661 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr20_KI270870v1_alt (183433 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr21_GL383578v2_alt (63917 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr21_KI270874v1_alt (166743 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr21_KI270873v1_alt (143900 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr21_GL383579v2_alt (201197 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr21_GL383580v2_alt (74653 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr21_GL383581v2_alt (116689 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr21_KI270872v1_alt (82692 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr22_KI270875v1_alt (259914 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr22_KI270878v1_alt (186262 bp)\n",
      "01:28:29.331 DEBUG GenomeLocParser -  chr22_KI270879v1_alt (304135 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr22_KI270876v1_alt (263666 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr22_KI270877v1_alt (101331 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr22_GL383583v2_alt (96924 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr22_GL383582v2_alt (162811 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chrX_KI270880v1_alt (284869 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chrX_KI270881v1_alt (144206 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr19_KI270882v1_alt (248807 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr19_KI270883v1_alt (170399 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr19_KI270884v1_alt (157053 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr19_KI270885v1_alt (171027 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr19_KI270886v1_alt (204239 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr19_KI270887v1_alt (209512 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr19_KI270888v1_alt (155532 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr19_KI270889v1_alt (170698 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr19_KI270890v1_alt (184499 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr19_KI270891v1_alt (170680 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr1_KI270892v1_alt (162212 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr2_KI270894v1_alt (214158 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr2_KI270893v1_alt (161218 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr3_KI270895v1_alt (162896 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr4_KI270896v1_alt (378547 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr5_KI270897v1_alt (1144418 bp)\n",
      "01:28:29.332 DEBUG GenomeLocParser -  chr5_KI270898v1_alt (130957 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr6_GL000251v2_alt (4795265 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr7_KI270899v1_alt (190869 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr8_KI270901v1_alt (136959 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr8_KI270900v1_alt (318687 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr11_KI270902v1_alt (106711 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr11_KI270903v1_alt (214625 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr12_KI270904v1_alt (572349 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr15_KI270906v1_alt (196384 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr15_KI270905v1_alt (5161414 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr17_KI270907v1_alt (137721 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr17_KI270910v1_alt (157099 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr17_KI270909v1_alt (325800 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr17_JH159148v1_alt (88070 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr17_KI270908v1_alt (1423190 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr18_KI270912v1_alt (174061 bp)\n",
      "01:28:29.333 DEBUG GenomeLocParser -  chr18_KI270911v1_alt (157710 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_GL949747v2_alt (729520 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr22_KB663609v1_alt (74013 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chrX_KI270913v1_alt (274009 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270914v1_alt (205194 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270915v1_alt (170665 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270916v1_alt (184516 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270917v1_alt (190932 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270918v1_alt (123111 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270919v1_alt (170701 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270920v1_alt (198005 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270921v1_alt (282224 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270922v1_alt (187935 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270923v1_alt (189352 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr3_KI270924v1_alt (166540 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr4_KI270925v1_alt (555799 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr6_GL000252v2_alt (4604811 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr8_KI270926v1_alt (229282 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr11_KI270927v1_alt (218612 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_GL949748v2_alt (1064304 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr22_KI270928v1_alt (176103 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270929v1_alt (186203 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270930v1_alt (200773 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270931v1_alt (170148 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270932v1_alt (215732 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_KI270933v1_alt (170537 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_GL000209v2_alt (177381 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr3_KI270934v1_alt (163458 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr6_GL000253v2_alt (4677643 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr19_GL949749v2_alt (1091841 bp)\n",
      "01:28:29.334 DEBUG GenomeLocParser -  chr3_KI270935v1_alt (197351 bp)\n",
      "01:28:29.335 DEBUG GenomeLocParser -  chr6_GL000254v2_alt (4827813 bp)\n",
      "01:28:29.335 DEBUG GenomeLocParser -  chr19_GL949750v2_alt (1066390 bp)\n",
      "01:28:29.335 DEBUG GenomeLocParser -  chr3_KI270936v1_alt (164170 bp)\n",
      "01:28:29.335 DEBUG GenomeLocParser -  chr6_GL000255v2_alt (4606388 bp)\n",
      "01:28:29.335 DEBUG GenomeLocParser -  chr19_GL949751v2_alt (1002683 bp)\n",
      "01:28:29.336 DEBUG GenomeLocParser -  chr3_KI270937v1_alt (165607 bp)\n",
      "01:28:29.337 DEBUG GenomeLocParser -  chr6_GL000256v2_alt (4929269 bp)\n",
      "01:28:29.338 DEBUG GenomeLocParser -  chr19_GL949752v1_alt (987100 bp)\n",
      "01:28:29.338 DEBUG GenomeLocParser -  chr6_KI270758v1_alt (76752 bp)\n",
      "01:28:29.339 DEBUG GenomeLocParser -  chr19_GL949753v2_alt (796479 bp)\n",
      "01:28:29.339 DEBUG GenomeLocParser -  chr19_KI270938v1_alt (1066800 bp)\n",
      "01:28:29.340 DEBUG GenomeLocParser -  chrEBV (171823 bp)\n",
      "01:28:29.340 DEBUG GenomeLocParser -  chrUn_KN707606v1_decoy (2200 bp)\n",
      "01:28:29.340 DEBUG GenomeLocParser -  chrUn_KN707607v1_decoy (3033 bp)\n",
      "01:28:29.340 DEBUG GenomeLocParser -  chrUn_KN707608v1_decoy (3112 bp)\n",
      "01:28:29.340 DEBUG GenomeLocParser -  chrUn_KN707609v1_decoy (1642 bp)\n",
      "01:28:29.340 DEBUG GenomeLocParser -  chrUn_KN707610v1_decoy (1393 bp)\n",
      "01:28:29.340 DEBUG GenomeLocParser -  chrUn_KN707611v1_decoy (1103 bp)\n",
      "01:28:29.340 DEBUG GenomeLocParser -  chrUn_KN707612v1_decoy (1039 bp)\n",
      "01:28:29.340 DEBUG GenomeLocParser -  chrUn_KN707613v1_decoy (1619 bp)\n",
      "01:28:29.340 DEBUG GenomeLocParser -  chrUn_KN707614v1_decoy (3122 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707615v1_decoy (1934 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707616v1_decoy (3111 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707617v1_decoy (2545 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707618v1_decoy (2295 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707619v1_decoy (1551 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707620v1_decoy (2046 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707621v1_decoy (1222 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707622v1_decoy (1535 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707623v1_decoy (3784 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707624v1_decoy (1329 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707625v1_decoy (1238 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707626v1_decoy (5623 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707627v1_decoy (5821 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707628v1_decoy (2960 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707629v1_decoy (1848 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707630v1_decoy (2315 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707631v1_decoy (1945 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707632v1_decoy (1424 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707633v1_decoy (1274 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707634v1_decoy (1007 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707635v1_decoy (1414 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707636v1_decoy (1725 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707637v1_decoy (5354 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707638v1_decoy (2189 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707639v1_decoy (1294 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707640v1_decoy (1831 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707641v1_decoy (1647 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707642v1_decoy (2943 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707643v1_decoy (2857 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707644v1_decoy (1030 bp)\n",
      "01:28:29.341 DEBUG GenomeLocParser -  chrUn_KN707645v1_decoy (1070 bp)\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707646v1_decoy (1735 bp)\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707647v1_decoy (1982 bp)\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707648v1_decoy (1564 bp)\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707649v1_decoy (1775 bp)\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707650v1_decoy (1540 bp)\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707651v1_decoy (2013 bp)\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707652v1_decoy (1176 bp)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707653v1_decoy (1890 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707654v1_decoy (3644 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707655v1_decoy (2785 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707656v1_decoy (1017 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707657v1_decoy (1068 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707658v1_decoy (1007 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707659v1_decoy (2605 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707660v1_decoy (8410 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707661v1_decoy (5534 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707662v1_decoy (2173 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707663v1_decoy (1065 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707664v1_decoy (8683 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707665v1_decoy (2670 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707666v1_decoy (2420 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707667v1_decoy (2189 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707668v1_decoy (2093 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707669v1_decoy (1184 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707670v1_decoy (1205 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707671v1_decoy (2786 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707672v1_decoy (2794 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707673v1_decoy (19544 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707674v1_decoy (2848 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707675v1_decoy (10556 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707676v1_decoy (9066 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707677v1_decoy (7267 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707678v1_decoy (2462 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707679v1_decoy (1774 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707680v1_decoy (1297 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707681v1_decoy (4379 bp)\r\n",
      "01:28:29.342 DEBUG GenomeLocParser -  chrUn_KN707682v1_decoy (4208 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707683v1_decoy (4068 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707684v1_decoy (2940 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707685v1_decoy (3938 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707686v1_decoy (2072 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707687v1_decoy (1136 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707688v1_decoy (4248 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707689v1_decoy (5823 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707690v1_decoy (3715 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707691v1_decoy (4885 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707692v1_decoy (4813 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707693v1_decoy (2899 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707694v1_decoy (1228 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707695v1_decoy (3119 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707696v1_decoy (3828 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707697v1_decoy (1186 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707698v1_decoy (1908 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707699v1_decoy (2795 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707700v1_decoy (3703 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707701v1_decoy (6722 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707702v1_decoy (6466 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707703v1_decoy (2235 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707704v1_decoy (2871 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707705v1_decoy (4632 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707706v1_decoy (4225 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707707v1_decoy (4339 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707708v1_decoy (2305 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707709v1_decoy (3273 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707710v1_decoy (5701 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707711v1_decoy (4154 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707712v1_decoy (1243 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707713v1_decoy (1308 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707714v1_decoy (2922 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707715v1_decoy (3044 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707716v1_decoy (2888 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707717v1_decoy (1742 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707718v1_decoy (4969 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707719v1_decoy (3270 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707720v1_decoy (6028 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707721v1_decoy (1105 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707722v1_decoy (2884 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707723v1_decoy (1124 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707724v1_decoy (1454 bp)\r\n",
      "01:28:29.343 DEBUG GenomeLocParser -  chrUn_KN707725v1_decoy (2565 bp)\r\n",
      "01:28:29.344 DEBUG GenomeLocParser -  chrUn_KN707726v1_decoy (2149 bp)\r\n",
      "01:28:29.344 DEBUG GenomeLocParser -  chrUn_KN707727v1_decoy (2630 bp)\r\n",
      "01:28:29.344 DEBUG GenomeLocParser -  chrUn_KN707728v1_decoy (14625 bp)\r\n",
      "01:28:29.345 DEBUG GenomeLocParser -  chrUn_KN707729v1_decoy (7431 bp)\r\n",
      "01:28:29.345 DEBUG GenomeLocParser -  chrUn_KN707730v1_decoy (5776 bp)\r\n",
      "01:28:29.346 DEBUG GenomeLocParser -  chrUn_KN707731v1_decoy (4820 bp)\r\n",
      "01:28:29.347 DEBUG GenomeLocParser -  chrUn_KN707732v1_decoy (1227 bp)\r\n",
      "01:28:29.347 DEBUG GenomeLocParser -  chrUn_KN707733v1_decoy (7503 bp)\r\n",
      "01:28:29.347 DEBUG GenomeLocParser -  chrUn_KN707734v1_decoy (9652 bp)\r\n",
      "01:28:29.347 DEBUG GenomeLocParser -  chrUn_KN707735v1_decoy (1091 bp)\r\n",
      "01:28:29.347 DEBUG GenomeLocParser -  chrUn_KN707736v1_decoy (2467 bp)\r\n",
      "01:28:29.347 DEBUG GenomeLocParser -  chrUn_KN707737v1_decoy (1270 bp)\r\n",
      "01:28:29.347 DEBUG GenomeLocParser -  chrUn_KN707738v1_decoy (4365 bp)\r\n",
      "01:28:29.347 DEBUG GenomeLocParser -  chrUn_KN707739v1_decoy (4284 bp)\r\n",
      "01:28:29.347 DEBUG GenomeLocParser -  chrUn_KN707740v1_decoy (10282 bp)\r\n",
      "01:28:29.347 DEBUG GenomeLocParser -  chrUn_KN707741v1_decoy (5601 bp)\r\n",
      "01:28:29.347 DEBUG GenomeLocParser -  chrUn_KN707742v1_decoy (4758 bp)\r\n",
      "01:28:29.347 DEBUG GenomeLocParser -  chrUn_KN707743v1_decoy (1624 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707744v1_decoy (4024 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707745v1_decoy (1276 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707746v1_decoy (5083 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707747v1_decoy (2075 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707748v1_decoy (3553 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707749v1_decoy (7010 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707750v1_decoy (4718 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707751v1_decoy (3546 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707752v1_decoy (2873 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707753v1_decoy (2144 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707754v1_decoy (2243 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707755v1_decoy (5343 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707756v1_decoy (4877 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707757v1_decoy (3034 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707758v1_decoy (2826 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707759v1_decoy (1221 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707760v1_decoy (1169 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707761v1_decoy (2319 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707762v1_decoy (3450 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707763v1_decoy (2674 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707764v1_decoy (3912 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707765v1_decoy (6020 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707766v1_decoy (2303 bp)\r\n",
      "01:28:29.348 DEBUG GenomeLocParser -  chrUn_KN707767v1_decoy (2552 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707768v1_decoy (3656 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707769v1_decoy (1591 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707770v1_decoy (1209 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707771v1_decoy (3176 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707772v1_decoy (8915 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707773v1_decoy (4902 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707774v1_decoy (3324 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707775v1_decoy (5997 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707776v1_decoy (2618 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707777v1_decoy (10311 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707778v1_decoy (2440 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707779v1_decoy (12444 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707780v1_decoy (5691 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707781v1_decoy (2717 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707782v1_decoy (5277 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707783v1_decoy (4373 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707784v1_decoy (3224 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707785v1_decoy (2631 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707786v1_decoy (5385 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707787v1_decoy (3678 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707788v1_decoy (1412 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707789v1_decoy (1443 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707790v1_decoy (1098 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707791v1_decoy (3240 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707792v1_decoy (1915 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707793v1_decoy (4667 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707794v1_decoy (7219 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707795v1_decoy (3277 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707796v1_decoy (3473 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707797v1_decoy (4243 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707798v1_decoy (17599 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707799v1_decoy (5095 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707800v1_decoy (2237 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707801v1_decoy (2901 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707802v1_decoy (2666 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707803v1_decoy (5336 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707804v1_decoy (4383 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707805v1_decoy (5446 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707806v1_decoy (6252 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707807v1_decoy (4616 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707808v1_decoy (3021 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707809v1_decoy (3667 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707810v1_decoy (4563 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707811v1_decoy (1120 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707812v1_decoy (3845 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707813v1_decoy (2272 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707814v1_decoy (4764 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707815v1_decoy (5410 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707816v1_decoy (7150 bp)\r\n",
      "01:28:29.349 DEBUG GenomeLocParser -  chrUn_KN707817v1_decoy (1762 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707818v1_decoy (1207 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707819v1_decoy (1331 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707820v1_decoy (8307 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707821v1_decoy (2276 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707822v1_decoy (2575 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707823v1_decoy (3970 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707824v1_decoy (1352 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707825v1_decoy (3040 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707826v1_decoy (2070 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707827v1_decoy (2913 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707828v1_decoy (2389 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707829v1_decoy (1835 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707830v1_decoy (4807 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707831v1_decoy (2201 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707832v1_decoy (1265 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707833v1_decoy (1961 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707834v1_decoy (1064 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707835v1_decoy (1932 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707836v1_decoy (3213 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707837v1_decoy (1178 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707838v1_decoy (2926 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707839v1_decoy (1038 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707840v1_decoy (3298 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707841v1_decoy (8992 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707842v1_decoy (6698 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707843v1_decoy (4880 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707844v1_decoy (1766 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707845v1_decoy (3532 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707846v1_decoy (2297 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707847v1_decoy (1234 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707848v1_decoy (1205 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707849v1_decoy (2790 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707850v1_decoy (2006 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707851v1_decoy (4593 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707852v1_decoy (1579 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707853v1_decoy (9597 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707854v1_decoy (10451 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707855v1_decoy (3219 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707856v1_decoy (2300 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707857v1_decoy (5985 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707858v1_decoy (2959 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707859v1_decoy (1340 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707860v1_decoy (3148 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707861v1_decoy (2242 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707862v1_decoy (16513 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707863v1_decoy (7821 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707864v1_decoy (2159 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707865v1_decoy (2114 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707866v1_decoy (4109 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707867v1_decoy (1544 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707868v1_decoy (1005 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707869v1_decoy (8632 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707870v1_decoy (1012 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707871v1_decoy (4728 bp)\r\n",
      "01:28:29.350 DEBUG GenomeLocParser -  chrUn_KN707872v1_decoy (2165 bp)\r\n",
      "01:28:29.351 DEBUG GenomeLocParser -  chrUn_KN707873v1_decoy (7591 bp)\r\n",
      "01:28:29.351 DEBUG GenomeLocParser -  chrUn_KN707874v1_decoy (5202 bp)\r\n",
      "01:28:29.351 DEBUG GenomeLocParser -  chrUn_KN707875v1_decoy (4241 bp)\r\n",
      "01:28:29.351 DEBUG GenomeLocParser -  chrUn_KN707876v1_decoy (4131 bp)\r\n",
      "01:28:29.352 DEBUG GenomeLocParser -  chrUn_KN707877v1_decoy (2272 bp)\r\n",
      "01:28:29.353 DEBUG GenomeLocParser -  chrUn_KN707878v1_decoy (2085 bp)\r\n",
      "01:28:29.353 DEBUG GenomeLocParser -  chrUn_KN707879v1_decoy (4346 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707880v1_decoy (1208 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707881v1_decoy (4543 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707882v1_decoy (2772 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707883v1_decoy (2490 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707884v1_decoy (4568 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707885v1_decoy (1776 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707886v1_decoy (2699 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707887v1_decoy (3534 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707888v1_decoy (2424 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707889v1_decoy (1747 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707890v1_decoy (1088 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707891v1_decoy (1143 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707892v1_decoy (2530 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707893v1_decoy (8049 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707894v1_decoy (1366 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707895v1_decoy (4284 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707896v1_decoy (33125 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707897v1_decoy (2137 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707898v1_decoy (3840 bp)\r\n",
      "01:28:29.354 DEBUG GenomeLocParser -  chrUn_KN707899v1_decoy (3087 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707900v1_decoy (2041 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707901v1_decoy (3344 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707902v1_decoy (2921 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707903v1_decoy (6581 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707904v1_decoy (3968 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707905v1_decoy (2339 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707906v1_decoy (1243 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707907v1_decoy (7776 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707908v1_decoy (19837 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707909v1_decoy (1737 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707910v1_decoy (1098 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707911v1_decoy (1893 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707912v1_decoy (1281 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707913v1_decoy (1527 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707914v1_decoy (2055 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707915v1_decoy (2527 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707916v1_decoy (3275 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707917v1_decoy (1265 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707918v1_decoy (2623 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707919v1_decoy (4850 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707920v1_decoy (3584 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707921v1_decoy (2561 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707922v1_decoy (3041 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707923v1_decoy (1409 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707924v1_decoy (4596 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707925v1_decoy (11555 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707926v1_decoy (1266 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707927v1_decoy (1079 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707928v1_decoy (1087 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707929v1_decoy (1226 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707930v1_decoy (1131 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707931v1_decoy (1199 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707932v1_decoy (1084 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707933v1_decoy (2038 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707934v1_decoy (1070 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707935v1_decoy (1312 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707936v1_decoy (4031 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707937v1_decoy (7445 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707938v1_decoy (1770 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707939v1_decoy (5600 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707940v1_decoy (1882 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707941v1_decoy (1170 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707942v1_decoy (1300 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707943v1_decoy (5325 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707944v1_decoy (2043 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707945v1_decoy (1072 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707946v1_decoy (2463 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707947v1_decoy (1010 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707948v1_decoy (1432 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707949v1_decoy (1162 bp)\r\n",
      "01:28:29.355 DEBUG GenomeLocParser -  chrUn_KN707950v1_decoy (1095 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707951v1_decoy (1118 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707952v1_decoy (1383 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707953v1_decoy (2289 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707954v1_decoy (1648 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707955v1_decoy (2203 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707956v1_decoy (3270 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707957v1_decoy (11499 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707958v1_decoy (2474 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707959v1_decoy (2294 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707960v1_decoy (1238 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707961v1_decoy (3410 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707962v1_decoy (1523 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707963v1_decoy (62955 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707964v1_decoy (6282 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707965v1_decoy (3836 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707966v1_decoy (6486 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707967v1_decoy (15368 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707968v1_decoy (9572 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707969v1_decoy (6413 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707970v1_decoy (4104 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707971v1_decoy (12943 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707972v1_decoy (4650 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707973v1_decoy (3080 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707974v1_decoy (3134 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707975v1_decoy (6211 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707976v1_decoy (1126 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707977v1_decoy (1101 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707978v1_decoy (1101 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707979v1_decoy (2648 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707980v1_decoy (2973 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707981v1_decoy (2520 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707982v1_decoy (2318 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707983v1_decoy (2606 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707984v1_decoy (2205 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707985v1_decoy (2929 bp)\r\n",
      "01:28:29.356 DEBUG GenomeLocParser -  chrUn_KN707986v1_decoy (3869 bp)\r\n",
      "01:28:29.357 DEBUG GenomeLocParser -  chrUn_KN707987v1_decoy (1117 bp)\r\n",
      "01:28:29.357 DEBUG GenomeLocParser -  chrUn_KN707988v1_decoy (2960 bp)\r\n",
      "01:28:29.357 DEBUG GenomeLocParser -  chrUn_KN707989v1_decoy (1009 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_KN707990v1_decoy (4048 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_KN707991v1_decoy (2193 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_KN707992v1_decoy (1830 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_JTFH01000001v1_decoy (25139 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_JTFH01000002v1_decoy (18532 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_JTFH01000003v1_decoy (15240 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_JTFH01000004v1_decoy (13739 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_JTFH01000005v1_decoy (11297 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_JTFH01000006v1_decoy (10074 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_JTFH01000007v1_decoy (9891 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_JTFH01000008v1_decoy (9774 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_JTFH01000009v1_decoy (9727 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_JTFH01000010v1_decoy (9358 bp)\r\n",
      "01:28:29.358 DEBUG GenomeLocParser -  chrUn_JTFH01000011v1_decoy (8920 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000012v1_decoy (8479 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000013v1_decoy (8312 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000014v1_decoy (8261 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000015v1_decoy (8131 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000016v1_decoy (8051 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000017v1_decoy (7832 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000018v1_decoy (7710 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000019v1_decoy (7702 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000020v1_decoy (7479 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000021v1_decoy (7368 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000022v1_decoy (7162 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000023v1_decoy (7065 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000024v1_decoy (7019 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000025v1_decoy (6997 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000026v1_decoy (6994 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000027v1_decoy (6979 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000028v1_decoy (6797 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000029v1_decoy (6525 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000030v1_decoy (6246 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000031v1_decoy (5926 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000032v1_decoy (5914 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000033v1_decoy (5898 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000034v1_decoy (5879 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000035v1_decoy (5834 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000036v1_decoy (5743 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000037v1_decoy (5577 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000038v1_decoy (5413 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000039v1_decoy (5250 bp)\r\n",
      "01:28:29.359 DEBUG GenomeLocParser -  chrUn_JTFH01000040v1_decoy (5246 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000041v1_decoy (5118 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000042v1_decoy (5058 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000043v1_decoy (4959 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000044v1_decoy (4853 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000045v1_decoy (4828 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000046v1_decoy (4819 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000047v1_decoy (4809 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000048v1_decoy (4710 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000049v1_decoy (4680 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000050v1_decoy (4645 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000051v1_decoy (4514 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000052v1_decoy (4439 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000053v1_decoy (4416 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000054v1_decoy (4409 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000055v1_decoy (4392 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000056v1_decoy (4359 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000057v1_decoy (4319 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000058v1_decoy (4290 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000059v1_decoy (4242 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000060v1_decoy (4228 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000061v1_decoy (4222 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000062v1_decoy (4216 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000063v1_decoy (4210 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000064v1_decoy (4206 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000065v1_decoy (4102 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000066v1_decoy (4101 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000067v1_decoy (4083 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000068v1_decoy (3967 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000069v1_decoy (3955 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000070v1_decoy (3945 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000071v1_decoy (3930 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000072v1_decoy (3929 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000073v1_decoy (3924 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000074v1_decoy (3919 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000075v1_decoy (3908 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000076v1_decoy (3892 bp)\r\n",
      "01:28:29.360 DEBUG GenomeLocParser -  chrUn_JTFH01000077v1_decoy (3890 bp)\r\n",
      "01:28:29.361 DEBUG GenomeLocParser -  chrUn_JTFH01000078v1_decoy (3859 bp)\r\n",
      "01:28:29.361 DEBUG GenomeLocParser -  chrUn_JTFH01000079v1_decoy (3846 bp)\r\n",
      "01:28:29.361 DEBUG GenomeLocParser -  chrUn_JTFH01000080v1_decoy (3835 bp)\r\n",
      "01:28:29.361 DEBUG GenomeLocParser -  chrUn_JTFH01000081v1_decoy (3830 bp)\r\n",
      "01:28:29.361 DEBUG GenomeLocParser -  chrUn_JTFH01000082v1_decoy (3828 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000083v1_decoy (3825 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000084v1_decoy (3821 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000085v1_decoy (3809 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000086v1_decoy (3801 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000087v1_decoy (3799 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000088v1_decoy (3737 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000089v1_decoy (3701 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000090v1_decoy (3698 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000091v1_decoy (3692 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000092v1_decoy (3686 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000093v1_decoy (3677 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000094v1_decoy (3664 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000095v1_decoy (3613 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000096v1_decoy (3611 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000097v1_decoy (3606 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000098v1_decoy (3584 bp)\r\n",
      "01:28:29.362 DEBUG GenomeLocParser -  chrUn_JTFH01000099v1_decoy (3581 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000100v1_decoy (3543 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000101v1_decoy (3528 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000102v1_decoy (3527 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000103v1_decoy (3496 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000104v1_decoy (3493 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000105v1_decoy (3484 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000106v1_decoy (3435 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000107v1_decoy (3391 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000108v1_decoy (3374 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000109v1_decoy (3371 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000110v1_decoy (3361 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000111v1_decoy (3351 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000112v1_decoy (3340 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000113v1_decoy (3320 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000114v1_decoy (3282 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000115v1_decoy (3278 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000116v1_decoy (3260 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000117v1_decoy (3258 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000118v1_decoy (3253 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000119v1_decoy (3247 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000120v1_decoy (3230 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000121v1_decoy (3224 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000122v1_decoy (3216 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000123v1_decoy (3212 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000124v1_decoy (3194 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000125v1_decoy (3189 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000126v1_decoy (3177 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000127v1_decoy (3176 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000128v1_decoy (3173 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000129v1_decoy (3170 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000130v1_decoy (3166 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000131v1_decoy (3163 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000132v1_decoy (3143 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000133v1_decoy (3137 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000134v1_decoy (3116 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000135v1_decoy (3106 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000136v1_decoy (3093 bp)\r\n",
      "01:28:29.363 DEBUG GenomeLocParser -  chrUn_JTFH01000137v1_decoy (3079 bp)\r\n",
      "01:28:29.364 DEBUG GenomeLocParser -  chrUn_JTFH01000138v1_decoy (3053 bp)\r\n",
      "01:28:29.365 DEBUG GenomeLocParser -  chrUn_JTFH01000139v1_decoy (3051 bp)\r\n",
      "01:28:29.366 DEBUG GenomeLocParser -  chrUn_JTFH01000140v1_decoy (3015 bp)\r\n",
      "01:28:29.368 DEBUG GenomeLocParser -  chrUn_JTFH01000141v1_decoy (3012 bp)\r\n",
      "01:28:29.369 DEBUG GenomeLocParser -  chrUn_JTFH01000142v1_decoy (3009 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000143v1_decoy (2997 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000144v1_decoy (2997 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000145v1_decoy (2983 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000146v1_decoy (2979 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000147v1_decoy (2967 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000148v1_decoy (2967 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000149v1_decoy (2966 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000150v1_decoy (2954 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000151v1_decoy (2952 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000152v1_decoy (2934 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000153v1_decoy (2918 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000154v1_decoy (2895 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000155v1_decoy (2887 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000156v1_decoy (2879 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000157v1_decoy (2878 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000158v1_decoy (2872 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000159v1_decoy (2868 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000160v1_decoy (2866 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000161v1_decoy (2865 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000162v1_decoy (2864 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000163v1_decoy (2859 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000164v1_decoy (2854 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000165v1_decoy (2830 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000166v1_decoy (2828 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000167v1_decoy (2824 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000168v1_decoy (2819 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000169v1_decoy (2813 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000170v1_decoy (2809 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000171v1_decoy (2802 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000172v1_decoy (2791 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000173v1_decoy (2783 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000174v1_decoy (2778 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000175v1_decoy (2777 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000176v1_decoy (2770 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000177v1_decoy (2769 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000178v1_decoy (2767 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000179v1_decoy (2763 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000180v1_decoy (2745 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000181v1_decoy (2742 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000182v1_decoy (2736 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000183v1_decoy (2729 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000184v1_decoy (2726 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000185v1_decoy (2719 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000186v1_decoy (2715 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000187v1_decoy (2708 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000188v1_decoy (2704 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000189v1_decoy (2692 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000190v1_decoy (2691 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000191v1_decoy (2690 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000192v1_decoy (2687 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000193v1_decoy (2677 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000194v1_decoy (2668 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000195v1_decoy (2668 bp)\r\n",
      "01:28:29.370 DEBUG GenomeLocParser -  chrUn_JTFH01000196v1_decoy (2663 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000197v1_decoy (2655 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000198v1_decoy (2644 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000199v1_decoy (2642 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000200v1_decoy (2632 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000201v1_decoy (2632 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000202v1_decoy (2628 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000203v1_decoy (2623 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000204v1_decoy (2622 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000205v1_decoy (2619 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000206v1_decoy (2605 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000207v1_decoy (2603 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000208v1_decoy (2601 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000209v1_decoy (2598 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000210v1_decoy (2597 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000211v1_decoy (2596 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000212v1_decoy (2594 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000213v1_decoy (2586 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000214v1_decoy (2585 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000215v1_decoy (2583 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000216v1_decoy (2578 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000217v1_decoy (2569 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000218v1_decoy (2569 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000219v1_decoy (2551 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000220v1_decoy (2548 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000221v1_decoy (2548 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000222v1_decoy (2546 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000223v1_decoy (2545 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000224v1_decoy (2534 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000225v1_decoy (2533 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000226v1_decoy (2522 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000227v1_decoy (2522 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000228v1_decoy (2515 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000229v1_decoy (2513 bp)\r\n",
      "01:28:29.371 DEBUG GenomeLocParser -  chrUn_JTFH01000230v1_decoy (2507 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000231v1_decoy (2504 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000232v1_decoy (2497 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000233v1_decoy (2471 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000234v1_decoy (2465 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000235v1_decoy (2464 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000236v1_decoy (2459 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000237v1_decoy (2457 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000238v1_decoy (2450 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000239v1_decoy (2435 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000240v1_decoy (2434 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000241v1_decoy (2432 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000242v1_decoy (2427 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000243v1_decoy (2421 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000244v1_decoy (2420 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000245v1_decoy (2414 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000246v1_decoy (2404 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000247v1_decoy (2403 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000248v1_decoy (2402 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000249v1_decoy (2397 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000250v1_decoy (2395 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000251v1_decoy (2394 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000252v1_decoy (2388 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000253v1_decoy (2382 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000254v1_decoy (2381 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000255v1_decoy (2380 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000256v1_decoy (2368 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000257v1_decoy (2364 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000258v1_decoy (2363 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000259v1_decoy (2348 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000260v1_decoy (2339 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000261v1_decoy (2335 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000262v1_decoy (2332 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000263v1_decoy (2331 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000264v1_decoy (2330 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000265v1_decoy (2323 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000266v1_decoy (2319 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000267v1_decoy (2314 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000268v1_decoy (2308 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000269v1_decoy (2306 bp)\r\n",
      "01:28:29.372 DEBUG GenomeLocParser -  chrUn_JTFH01000270v1_decoy (2296 bp)\r\n",
      "01:28:29.373 DEBUG GenomeLocParser -  chrUn_JTFH01000271v1_decoy (2287 bp)\r\n",
      "01:28:29.373 DEBUG GenomeLocParser -  chrUn_JTFH01000272v1_decoy (2279 bp)\r\n",
      "01:28:29.373 DEBUG GenomeLocParser -  chrUn_JTFH01000273v1_decoy (2276 bp)\r\n",
      "01:28:29.373 DEBUG GenomeLocParser -  chrUn_JTFH01000274v1_decoy (2273 bp)\r\n",
      "01:28:29.374 DEBUG GenomeLocParser -  chrUn_JTFH01000275v1_decoy (2262 bp)\r\n",
      "01:28:29.374 DEBUG GenomeLocParser -  chrUn_JTFH01000276v1_decoy (2254 bp)\r\n",
      "01:28:29.375 DEBUG GenomeLocParser -  chrUn_JTFH01000277v1_decoy (2252 bp)\r\n",
      "01:28:29.376 DEBUG GenomeLocParser -  chrUn_JTFH01000278v1_decoy (2245 bp)\r\n",
      "01:28:29.377 DEBUG GenomeLocParser -  chrUn_JTFH01000279v1_decoy (2239 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000280v1_decoy (2223 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000281v1_decoy (2220 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000282v1_decoy (2218 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000283v1_decoy (2215 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000284v1_decoy (2213 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000285v1_decoy (2203 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000286v1_decoy (2200 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000287v1_decoy (2197 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000288v1_decoy (2194 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000289v1_decoy (2183 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000290v1_decoy (2179 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000291v1_decoy (2177 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000292v1_decoy (2177 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000293v1_decoy (2177 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000294v1_decoy (2168 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000295v1_decoy (2160 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000296v1_decoy (2155 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000297v1_decoy (2144 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000298v1_decoy (2143 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000299v1_decoy (2136 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000300v1_decoy (2134 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000301v1_decoy (2129 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000302v1_decoy (2128 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000303v1_decoy (2125 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000304v1_decoy (2125 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000305v1_decoy (2122 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000306v1_decoy (2111 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000307v1_decoy (2106 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000308v1_decoy (2094 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000309v1_decoy (2093 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000310v1_decoy (2088 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000311v1_decoy (2086 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000312v1_decoy (2086 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000313v1_decoy (2084 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000314v1_decoy (2080 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000315v1_decoy (2079 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000316v1_decoy (2076 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000317v1_decoy (2071 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000318v1_decoy (2066 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000319v1_decoy (2061 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000320v1_decoy (2055 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000321v1_decoy (2053 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000322v1_decoy (2040 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000323v1_decoy (2036 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000324v1_decoy (2035 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000325v1_decoy (2034 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000326v1_decoy (2032 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000327v1_decoy (2029 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000328v1_decoy (2025 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000329v1_decoy (2021 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000330v1_decoy (2018 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000331v1_decoy (2015 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000332v1_decoy (2009 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000333v1_decoy (2007 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000334v1_decoy (2005 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000335v1_decoy (2003 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000336v1_decoy (2001 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000337v1_decoy (2001 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000338v1_decoy (2000 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000339v1_decoy (1996 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000340v1_decoy (1992 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000341v1_decoy (1985 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000342v1_decoy (1981 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000343v1_decoy (1977 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000344v1_decoy (1971 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000345v1_decoy (1968 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000346v1_decoy (1962 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000347v1_decoy (1961 bp)\r\n",
      "01:28:29.378 DEBUG GenomeLocParser -  chrUn_JTFH01000348v1_decoy (1960 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000349v1_decoy (1960 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000350v1_decoy (1954 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000351v1_decoy (1952 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000352v1_decoy (1947 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000353v1_decoy (1944 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000354v1_decoy (1943 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000355v1_decoy (1941 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000356v1_decoy (1937 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000357v1_decoy (1934 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000358v1_decoy (1929 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000359v1_decoy (1924 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000360v1_decoy (1924 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000361v1_decoy (1923 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000362v1_decoy (1921 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000363v1_decoy (1918 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000364v1_decoy (1915 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000365v1_decoy (1915 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000366v1_decoy (1914 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000367v1_decoy (1912 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000368v1_decoy (1910 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000369v1_decoy (1907 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000370v1_decoy (1904 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000371v1_decoy (1897 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000372v1_decoy (1891 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000373v1_decoy (1890 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000374v1_decoy (1888 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000375v1_decoy (1888 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000376v1_decoy (1885 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000377v1_decoy (1881 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000378v1_decoy (1879 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000379v1_decoy (1877 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000380v1_decoy (1876 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000381v1_decoy (1876 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000382v1_decoy (1874 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000383v1_decoy (1872 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000384v1_decoy (1869 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000385v1_decoy (1866 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000386v1_decoy (1865 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000387v1_decoy (1865 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000388v1_decoy (1865 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000389v1_decoy (1862 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000390v1_decoy (1862 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000391v1_decoy (1859 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000392v1_decoy (1856 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000393v1_decoy (1856 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000394v1_decoy (1854 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000395v1_decoy (1850 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000396v1_decoy (1849 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000397v1_decoy (1849 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000398v1_decoy (1847 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000399v1_decoy (1839 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000400v1_decoy (1834 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000401v1_decoy (1821 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000402v1_decoy (1815 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000403v1_decoy (1811 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000404v1_decoy (1808 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000405v1_decoy (1808 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000406v1_decoy (1807 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000407v1_decoy (1807 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000408v1_decoy (1802 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000409v1_decoy (1801 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000410v1_decoy (1800 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000411v1_decoy (1795 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000412v1_decoy (1794 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000413v1_decoy (1792 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000414v1_decoy (1788 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000415v1_decoy (1786 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000416v1_decoy (1782 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000417v1_decoy (1782 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000418v1_decoy (1781 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000419v1_decoy (1781 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000420v1_decoy (1779 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000421v1_decoy (1777 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000422v1_decoy (1764 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000423v1_decoy (1762 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000424v1_decoy (1755 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000425v1_decoy (1749 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000426v1_decoy (1747 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000427v1_decoy (1746 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000428v1_decoy (1745 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000429v1_decoy (1744 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000430v1_decoy (1742 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000431v1_decoy (1740 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000432v1_decoy (1740 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000433v1_decoy (1736 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000434v1_decoy (1735 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000435v1_decoy (1732 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000436v1_decoy (1732 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000437v1_decoy (1730 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000438v1_decoy (1727 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000439v1_decoy (1722 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000440v1_decoy (1718 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000441v1_decoy (1716 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000442v1_decoy (1710 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000443v1_decoy (1708 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000444v1_decoy (1707 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000445v1_decoy (1706 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000446v1_decoy (1705 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000447v1_decoy (1704 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000448v1_decoy (1699 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000449v1_decoy (1698 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000450v1_decoy (1697 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000451v1_decoy (1697 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000452v1_decoy (1695 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000453v1_decoy (1695 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000454v1_decoy (1693 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000455v1_decoy (1687 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000456v1_decoy (1686 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000457v1_decoy (1680 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000458v1_decoy (1679 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000459v1_decoy (1679 bp)\r\n",
      "01:28:29.379 DEBUG GenomeLocParser -  chrUn_JTFH01000460v1_decoy (1678 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000461v1_decoy (1674 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000462v1_decoy (1674 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000463v1_decoy (1671 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000464v1_decoy (1669 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000465v1_decoy (1665 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000466v1_decoy (1663 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000467v1_decoy (1657 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000468v1_decoy (1653 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000469v1_decoy (1652 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000470v1_decoy (1650 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000471v1_decoy (1649 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000472v1_decoy (1649 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000473v1_decoy (1640 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000474v1_decoy (1638 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000475v1_decoy (1636 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000476v1_decoy (1632 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000477v1_decoy (1631 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000478v1_decoy (1630 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000479v1_decoy (1627 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000480v1_decoy (1624 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000481v1_decoy (1617 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000482v1_decoy (1616 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000483v1_decoy (1615 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000484v1_decoy (1611 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000485v1_decoy (1611 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000486v1_decoy (1606 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000487v1_decoy (1605 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000488v1_decoy (1605 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000489v1_decoy (1600 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000490v1_decoy (1598 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000491v1_decoy (1598 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000492v1_decoy (1597 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000493v1_decoy (1596 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000494v1_decoy (1595 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000495v1_decoy (1592 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000496v1_decoy (1589 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000497v1_decoy (1585 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000498v1_decoy (1579 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000499v1_decoy (1578 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000500v1_decoy (1577 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000501v1_decoy (1577 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000502v1_decoy (1577 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000503v1_decoy (1576 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000504v1_decoy (1575 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000505v1_decoy (1574 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000506v1_decoy (1572 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000507v1_decoy (1571 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000508v1_decoy (1563 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000509v1_decoy (1561 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000510v1_decoy (1561 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000511v1_decoy (1560 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000512v1_decoy (1560 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000513v1_decoy (1554 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000514v1_decoy (1552 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000515v1_decoy (1548 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000516v1_decoy (1546 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000517v1_decoy (1541 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000518v1_decoy (1536 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000519v1_decoy (1533 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000520v1_decoy (1532 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000521v1_decoy (1532 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000522v1_decoy (1530 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000523v1_decoy (1527 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000524v1_decoy (1526 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000525v1_decoy (1524 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000526v1_decoy (1523 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000527v1_decoy (1523 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000528v1_decoy (1522 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000529v1_decoy (1522 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000530v1_decoy (1519 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000531v1_decoy (1513 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000532v1_decoy (1508 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000533v1_decoy (1508 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000534v1_decoy (1505 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000535v1_decoy (1503 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000536v1_decoy (1496 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000537v1_decoy (1491 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000538v1_decoy (1490 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000539v1_decoy (1490 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000540v1_decoy (1487 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000541v1_decoy (1486 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000542v1_decoy (1485 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000543v1_decoy (1484 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000544v1_decoy (1483 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000545v1_decoy (1479 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000546v1_decoy (1479 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000547v1_decoy (1476 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000548v1_decoy (1475 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000549v1_decoy (1472 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000550v1_decoy (1469 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000551v1_decoy (1468 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000552v1_decoy (1467 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000553v1_decoy (1465 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000554v1_decoy (1464 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000555v1_decoy (1463 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000556v1_decoy (1463 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000557v1_decoy (1459 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000558v1_decoy (1459 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000559v1_decoy (1458 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000560v1_decoy (1458 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000561v1_decoy (1454 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000562v1_decoy (1449 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000563v1_decoy (1449 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000564v1_decoy (1448 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000565v1_decoy (1446 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000566v1_decoy (1442 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000567v1_decoy (1441 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000568v1_decoy (1440 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000569v1_decoy (1439 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000570v1_decoy (1437 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000571v1_decoy (1436 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000572v1_decoy (1429 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000573v1_decoy (1429 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000574v1_decoy (1427 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000575v1_decoy (1426 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000576v1_decoy (1425 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000577v1_decoy (1424 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000578v1_decoy (1424 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000579v1_decoy (1423 bp)\r\n",
      "01:28:29.380 DEBUG GenomeLocParser -  chrUn_JTFH01000580v1_decoy (1423 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000581v1_decoy (1423 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000582v1_decoy (1414 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000583v1_decoy (1414 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000584v1_decoy (1413 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000585v1_decoy (1413 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000586v1_decoy (1410 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000587v1_decoy (1409 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000588v1_decoy (1409 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000589v1_decoy (1406 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000590v1_decoy (1405 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000591v1_decoy (1405 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000592v1_decoy (1404 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000593v1_decoy (1404 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000594v1_decoy (1402 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000595v1_decoy (1402 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000596v1_decoy (1402 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000597v1_decoy (1402 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000598v1_decoy (1400 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000599v1_decoy (1398 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000600v1_decoy (1396 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000601v1_decoy (1395 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000602v1_decoy (1394 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000603v1_decoy (1393 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000604v1_decoy (1391 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000605v1_decoy (1389 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000606v1_decoy (1389 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000607v1_decoy (1388 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000608v1_decoy (1387 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000609v1_decoy (1384 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000610v1_decoy (1381 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000611v1_decoy (1381 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000612v1_decoy (1379 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000613v1_decoy (1377 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000614v1_decoy (1376 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000615v1_decoy (1376 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000616v1_decoy (1375 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000617v1_decoy (1374 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000618v1_decoy (1372 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000619v1_decoy (1371 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000620v1_decoy (1370 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000621v1_decoy (1370 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000622v1_decoy (1366 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000623v1_decoy (1363 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000624v1_decoy (1360 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000625v1_decoy (1356 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000626v1_decoy (1355 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000627v1_decoy (1355 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000628v1_decoy (1352 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000629v1_decoy (1345 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000630v1_decoy (1344 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000631v1_decoy (1344 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000632v1_decoy (1342 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000633v1_decoy (1342 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000634v1_decoy (1336 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000635v1_decoy (1334 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000636v1_decoy (1334 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000637v1_decoy (1333 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000638v1_decoy (1332 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000639v1_decoy (1328 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000640v1_decoy (1328 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000641v1_decoy (1328 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000642v1_decoy (1327 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000643v1_decoy (1325 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000644v1_decoy (1322 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000645v1_decoy (1320 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000646v1_decoy (1319 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000647v1_decoy (1318 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000648v1_decoy (1315 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000649v1_decoy (1314 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000650v1_decoy (1313 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000651v1_decoy (1313 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000652v1_decoy (1312 bp)\r\n",
      "01:28:29.381 DEBUG GenomeLocParser -  chrUn_JTFH01000653v1_decoy (1310 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000654v1_decoy (1309 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000655v1_decoy (1309 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000656v1_decoy (1307 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000657v1_decoy (1307 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000658v1_decoy (1305 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000659v1_decoy (1304 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000660v1_decoy (1303 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000661v1_decoy (1302 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000662v1_decoy (1302 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000663v1_decoy (1301 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000664v1_decoy (1301 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000665v1_decoy (1300 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000666v1_decoy (1299 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000667v1_decoy (1297 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000668v1_decoy (1295 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000669v1_decoy (1294 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000670v1_decoy (1293 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000671v1_decoy (1291 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000672v1_decoy (1291 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000673v1_decoy (1289 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000674v1_decoy (1288 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000675v1_decoy (1288 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000676v1_decoy (1287 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000677v1_decoy (1287 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000678v1_decoy (1287 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000679v1_decoy (1286 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000680v1_decoy (1283 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000681v1_decoy (1281 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000682v1_decoy (1277 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000683v1_decoy (1274 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000684v1_decoy (1270 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000685v1_decoy (1267 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000686v1_decoy (1266 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000687v1_decoy (1260 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000688v1_decoy (1259 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000689v1_decoy (1258 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000690v1_decoy (1258 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000691v1_decoy (1258 bp)\r\n",
      "01:28:29.382 DEBUG GenomeLocParser -  chrUn_JTFH01000692v1_decoy (1256 bp)\r\n",
      "01:28:29.383 DEBUG GenomeLocParser -  chrUn_JTFH01000693v1_decoy (1255 bp)\r\n",
      "01:28:29.383 DEBUG GenomeLocParser -  chrUn_JTFH01000694v1_decoy (1254 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000695v1_decoy (1254 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000696v1_decoy (1253 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000697v1_decoy (1250 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000698v1_decoy (1249 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000699v1_decoy (1248 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000700v1_decoy (1248 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000701v1_decoy (1247 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000702v1_decoy (1242 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000703v1_decoy (1242 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000704v1_decoy (1241 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000705v1_decoy (1241 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000706v1_decoy (1241 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000707v1_decoy (1239 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000708v1_decoy (1238 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000709v1_decoy (1237 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000710v1_decoy (1236 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000711v1_decoy (1235 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000712v1_decoy (1234 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000713v1_decoy (1234 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000714v1_decoy (1234 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000715v1_decoy (1233 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000716v1_decoy (1232 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000717v1_decoy (1232 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000718v1_decoy (1231 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000719v1_decoy (1230 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000720v1_decoy (1228 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000721v1_decoy (1227 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000722v1_decoy (1227 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000723v1_decoy (1226 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000724v1_decoy (1224 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000725v1_decoy (1224 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000726v1_decoy (1220 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000727v1_decoy (1220 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000728v1_decoy (1219 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000729v1_decoy (1217 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000730v1_decoy (1216 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000731v1_decoy (1215 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000732v1_decoy (1214 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000733v1_decoy (1214 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000734v1_decoy (1214 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000735v1_decoy (1213 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000736v1_decoy (1212 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000737v1_decoy (1209 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000738v1_decoy (1208 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000739v1_decoy (1207 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000740v1_decoy (1207 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000741v1_decoy (1207 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000742v1_decoy (1206 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000743v1_decoy (1206 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000744v1_decoy (1205 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000745v1_decoy (1205 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000746v1_decoy (1204 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000747v1_decoy (1204 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000748v1_decoy (1204 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000749v1_decoy (1203 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000750v1_decoy (1201 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000751v1_decoy (1201 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000752v1_decoy (1200 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000753v1_decoy (1200 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000754v1_decoy (1199 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000755v1_decoy (1198 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000756v1_decoy (1197 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000757v1_decoy (1196 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000758v1_decoy (1195 bp)\r\n",
      "01:28:29.384 DEBUG GenomeLocParser -  chrUn_JTFH01000759v1_decoy (1194 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000760v1_decoy (1194 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000761v1_decoy (1191 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000762v1_decoy (1189 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000763v1_decoy (1186 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000764v1_decoy (1186 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000765v1_decoy (1184 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000766v1_decoy (1183 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000767v1_decoy (1183 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000768v1_decoy (1182 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000769v1_decoy (1181 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000770v1_decoy (1181 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000771v1_decoy (1181 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000772v1_decoy (1181 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000773v1_decoy (1179 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000774v1_decoy (1178 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000775v1_decoy (1178 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000776v1_decoy (1177 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000777v1_decoy (1177 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000778v1_decoy (1171 bp)\r\n",
      "01:28:29.385 DEBUG GenomeLocParser -  chrUn_JTFH01000779v1_decoy (1171 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000780v1_decoy (1171 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000781v1_decoy (1170 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000782v1_decoy (1170 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000783v1_decoy (1167 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000784v1_decoy (1167 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000785v1_decoy (1167 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000786v1_decoy (1165 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000787v1_decoy (1165 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000788v1_decoy (1162 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000789v1_decoy (1157 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000790v1_decoy (1156 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000791v1_decoy (1156 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000792v1_decoy (1154 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000793v1_decoy (1154 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000794v1_decoy (1151 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000795v1_decoy (1151 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000796v1_decoy (1150 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000797v1_decoy (1150 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000798v1_decoy (1147 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000799v1_decoy (1147 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000800v1_decoy (1146 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000801v1_decoy (1144 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000802v1_decoy (1144 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000803v1_decoy (1143 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000804v1_decoy (1142 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000805v1_decoy (1141 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000806v1_decoy (1141 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000807v1_decoy (1140 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000808v1_decoy (1138 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000809v1_decoy (1134 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000810v1_decoy (1134 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000811v1_decoy (1132 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000812v1_decoy (1131 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000813v1_decoy (1131 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000814v1_decoy (1130 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000815v1_decoy (1127 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000816v1_decoy (1126 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000817v1_decoy (1124 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000818v1_decoy (1122 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000819v1_decoy (1122 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000820v1_decoy (1121 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000821v1_decoy (1119 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000822v1_decoy (1119 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000823v1_decoy (1119 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000824v1_decoy (1119 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000825v1_decoy (1118 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000826v1_decoy (1116 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000827v1_decoy (1116 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000828v1_decoy (1115 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000829v1_decoy (1115 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000830v1_decoy (1115 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000831v1_decoy (1114 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000832v1_decoy (1113 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000833v1_decoy (1113 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000834v1_decoy (1110 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000835v1_decoy (1110 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000836v1_decoy (1109 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000837v1_decoy (1108 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000838v1_decoy (1107 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000839v1_decoy (1107 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000840v1_decoy (1107 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000841v1_decoy (1107 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000842v1_decoy (1106 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000843v1_decoy (1103 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000844v1_decoy (1103 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000845v1_decoy (1103 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000846v1_decoy (1100 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000847v1_decoy (1099 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000848v1_decoy (1098 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000849v1_decoy (1097 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000850v1_decoy (1096 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000851v1_decoy (1096 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000852v1_decoy (1094 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000853v1_decoy (1093 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000854v1_decoy (1090 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000855v1_decoy (1088 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000856v1_decoy (1087 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000857v1_decoy (1086 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000858v1_decoy (1085 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000859v1_decoy (1084 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000860v1_decoy (1084 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000861v1_decoy (1084 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000862v1_decoy (1084 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000863v1_decoy (1083 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000864v1_decoy (1083 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000865v1_decoy (1082 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000866v1_decoy (1082 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000867v1_decoy (1081 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000868v1_decoy (1081 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000869v1_decoy (1079 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000870v1_decoy (1076 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000871v1_decoy (1074 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000872v1_decoy (1073 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000873v1_decoy (1073 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000874v1_decoy (1071 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000875v1_decoy (1069 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000876v1_decoy (1067 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000877v1_decoy (1067 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000878v1_decoy (1067 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000879v1_decoy (1066 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000880v1_decoy (1065 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000881v1_decoy (1065 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000882v1_decoy (1065 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000883v1_decoy (1065 bp)\r\n",
      "01:28:29.386 DEBUG GenomeLocParser -  chrUn_JTFH01000884v1_decoy (1065 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000885v1_decoy (1064 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000886v1_decoy (1064 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000887v1_decoy (1064 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000888v1_decoy (1063 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000889v1_decoy (1062 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000890v1_decoy (1062 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000891v1_decoy (1062 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000892v1_decoy (1061 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000893v1_decoy (1060 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000894v1_decoy (1057 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000895v1_decoy (1057 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000896v1_decoy (1056 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000897v1_decoy (1055 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000898v1_decoy (1055 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000899v1_decoy (1055 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000900v1_decoy (1055 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000901v1_decoy (1054 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000902v1_decoy (1051 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000903v1_decoy (1050 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000904v1_decoy (1050 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000905v1_decoy (1049 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000906v1_decoy (1048 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000907v1_decoy (1047 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000908v1_decoy (1046 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000909v1_decoy (1046 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000910v1_decoy (1046 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000911v1_decoy (1045 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000912v1_decoy (1045 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000913v1_decoy (1045 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000914v1_decoy (1044 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000915v1_decoy (1042 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000916v1_decoy (1041 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000917v1_decoy (1039 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000918v1_decoy (1039 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000919v1_decoy (1038 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000920v1_decoy (1036 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000921v1_decoy (1036 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000922v1_decoy (1035 bp)\r\n",
      "01:28:29.387 DEBUG GenomeLocParser -  chrUn_JTFH01000923v1_decoy (1035 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000924v1_decoy (1033 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000925v1_decoy (1032 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000926v1_decoy (1031 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000927v1_decoy (1031 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000928v1_decoy (1031 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000929v1_decoy (1027 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000930v1_decoy (1027 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000931v1_decoy (1026 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000932v1_decoy (1026 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000933v1_decoy (1024 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000934v1_decoy (1024 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000935v1_decoy (1022 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000936v1_decoy (1022 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000937v1_decoy (1021 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000938v1_decoy (1020 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000939v1_decoy (1019 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000940v1_decoy (1018 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000941v1_decoy (1018 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000942v1_decoy (1018 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000943v1_decoy (1016 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000944v1_decoy (1010 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000945v1_decoy (1010 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000946v1_decoy (1009 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000947v1_decoy (1008 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000948v1_decoy (1007 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000949v1_decoy (1006 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000950v1_decoy (1005 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000951v1_decoy (1005 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000952v1_decoy (1004 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000953v1_decoy (1004 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000954v1_decoy (1003 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000955v1_decoy (1003 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000956v1_decoy (1003 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000957v1_decoy (1003 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000958v1_decoy (1002 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000959v1_decoy (1002 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000960v1_decoy (1000 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000961v1_decoy (1000 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000962v1_decoy (8358 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000963v1_decoy (7932 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000964v1_decoy (6846 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000965v1_decoy (4591 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000966v1_decoy (4041 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000967v1_decoy (3841 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000968v1_decoy (3754 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000969v1_decoy (3743 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000970v1_decoy (3702 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000971v1_decoy (3625 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000972v1_decoy (3529 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000973v1_decoy (3508 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000974v1_decoy (3359 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000975v1_decoy (3320 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000976v1_decoy (3231 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000977v1_decoy (3220 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000978v1_decoy (3212 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000979v1_decoy (3192 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000980v1_decoy (3092 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000981v1_decoy (3087 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000982v1_decoy (3048 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000983v1_decoy (3005 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000984v1_decoy (3004 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000985v1_decoy (2959 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000986v1_decoy (2934 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000987v1_decoy (2933 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000988v1_decoy (2827 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000989v1_decoy (2794 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000990v1_decoy (2749 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000991v1_decoy (2745 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000992v1_decoy (2733 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000993v1_decoy (2698 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000994v1_decoy (2665 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000995v1_decoy (2634 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000996v1_decoy (2492 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000997v1_decoy (2489 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000998v1_decoy (2468 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01000999v1_decoy (2414 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001000v1_decoy (2395 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001001v1_decoy (2356 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001002v1_decoy (2339 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001003v1_decoy (2310 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001004v1_decoy (2288 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001005v1_decoy (2285 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001006v1_decoy (2269 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001007v1_decoy (2253 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001008v1_decoy (2203 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001009v1_decoy (2176 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001010v1_decoy (2159 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001011v1_decoy (2155 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001012v1_decoy (2149 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001013v1_decoy (2129 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001014v1_decoy (2116 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001015v1_decoy (2113 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001016v1_decoy (2098 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001017v1_decoy (2066 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001018v1_decoy (2066 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001019v1_decoy (2059 bp)\r\n",
      "01:28:29.388 DEBUG GenomeLocParser -  chrUn_JTFH01001020v1_decoy (2047 bp)\r\n",
      "01:28:29.389 DEBUG GenomeLocParser -  chrUn_JTFH01001021v1_decoy (2040 bp)\r\n",
      "01:28:29.389 DEBUG GenomeLocParser -  chrUn_JTFH01001022v1_decoy (2030 bp)\r\n",
      "01:28:29.389 DEBUG GenomeLocParser -  chrUn_JTFH01001023v1_decoy (2024 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001024v1_decoy (2001 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001025v1_decoy (1992 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001026v1_decoy (1981 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001027v1_decoy (1979 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001028v1_decoy (1957 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001029v1_decoy (1953 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001030v1_decoy (1944 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001031v1_decoy (1936 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001032v1_decoy (1932 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001033v1_decoy (1882 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001034v1_decoy (1878 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001035v1_decoy (1870 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001036v1_decoy (1821 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001037v1_decoy (1813 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001038v1_decoy (1809 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001039v1_decoy (1804 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001040v1_decoy (1797 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001041v1_decoy (1791 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001042v1_decoy (1781 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001043v1_decoy (1766 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001044v1_decoy (1764 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001045v1_decoy (1743 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001046v1_decoy (1741 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001047v1_decoy (1709 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001048v1_decoy (1706 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001049v1_decoy (1701 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001050v1_decoy (1689 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001051v1_decoy (1646 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001052v1_decoy (1641 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001053v1_decoy (1639 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001054v1_decoy (1636 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001055v1_decoy (1632 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001056v1_decoy (1629 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001057v1_decoy (1623 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001058v1_decoy (1622 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001059v1_decoy (1622 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001060v1_decoy (1619 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001061v1_decoy (1606 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001062v1_decoy (1593 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001063v1_decoy (1592 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001064v1_decoy (1558 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001065v1_decoy (1545 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001066v1_decoy (1542 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001067v1_decoy (1540 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001068v1_decoy (1529 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001069v1_decoy (1518 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001070v1_decoy (1515 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001071v1_decoy (1513 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001072v1_decoy (1507 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001073v1_decoy (1504 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001074v1_decoy (1499 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001075v1_decoy (1495 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001076v1_decoy (1495 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001077v1_decoy (1492 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001078v1_decoy (1492 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001079v1_decoy (1489 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001080v1_decoy (1485 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001081v1_decoy (1483 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001082v1_decoy (1473 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001083v1_decoy (1470 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001084v1_decoy (1463 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001085v1_decoy (1460 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001086v1_decoy (1458 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001087v1_decoy (1456 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001088v1_decoy (1453 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001089v1_decoy (1443 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001090v1_decoy (1441 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001091v1_decoy (1426 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001092v1_decoy (1425 bp)\r\n",
      "01:28:29.390 DEBUG GenomeLocParser -  chrUn_JTFH01001093v1_decoy (1418 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001094v1_decoy (1413 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001095v1_decoy (1413 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001096v1_decoy (1412 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001097v1_decoy (1407 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001098v1_decoy (1406 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001099v1_decoy (1396 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001100v1_decoy (1390 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001101v1_decoy (1382 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001102v1_decoy (1376 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001103v1_decoy (1375 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001104v1_decoy (1371 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001105v1_decoy (1367 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001106v1_decoy (1364 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001107v1_decoy (1356 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001108v1_decoy (1355 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001109v1_decoy (1352 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001110v1_decoy (1350 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001111v1_decoy (1346 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001112v1_decoy (1345 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001113v1_decoy (1340 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001114v1_decoy (1330 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001115v1_decoy (1329 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001116v1_decoy (1324 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001117v1_decoy (1316 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001118v1_decoy (1307 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001119v1_decoy (1304 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001120v1_decoy (1304 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001121v1_decoy (1303 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001122v1_decoy (1301 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001123v1_decoy (1300 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001124v1_decoy (1297 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001125v1_decoy (1296 bp)\r\n",
      "01:28:29.391 DEBUG GenomeLocParser -  chrUn_JTFH01001126v1_decoy (1290 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001127v1_decoy (1284 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001128v1_decoy (1282 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001129v1_decoy (1281 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001130v1_decoy (1280 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001131v1_decoy (1279 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001132v1_decoy (1272 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001133v1_decoy (1267 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001134v1_decoy (1267 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001135v1_decoy (1266 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001136v1_decoy (1264 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001137v1_decoy (1264 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001138v1_decoy (1264 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001139v1_decoy (1263 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001140v1_decoy (1249 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001141v1_decoy (1240 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001142v1_decoy (1239 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001143v1_decoy (1235 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001144v1_decoy (1235 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001145v1_decoy (1233 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001146v1_decoy (1232 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001147v1_decoy (1230 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001148v1_decoy (1226 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001149v1_decoy (1223 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001150v1_decoy (1214 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001151v1_decoy (1213 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001152v1_decoy (1211 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001153v1_decoy (1209 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001154v1_decoy (1202 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001155v1_decoy (1199 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001156v1_decoy (1197 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001157v1_decoy (1193 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001158v1_decoy (1191 bp)\r\n",
      "01:28:29.392 DEBUG GenomeLocParser -  chrUn_JTFH01001159v1_decoy (1187 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001160v1_decoy (1186 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001161v1_decoy (1184 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001162v1_decoy (1184 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001163v1_decoy (1182 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001164v1_decoy (1179 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001165v1_decoy (1173 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001166v1_decoy (1169 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001167v1_decoy (1167 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001168v1_decoy (1166 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001169v1_decoy (1165 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001170v1_decoy (1164 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001171v1_decoy (1163 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001172v1_decoy (1158 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001173v1_decoy (1158 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001174v1_decoy (1157 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001175v1_decoy (1157 bp)\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001176v1_decoy (1157 bp)\r\n",
      "01:28:29.393 DEBUG GenomeLocParser -  chrUn_JTFH01001177v1_decoy (1155 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001178v1_decoy (1154 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001179v1_decoy (1149 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001180v1_decoy (1148 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001181v1_decoy (1148 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001182v1_decoy (1146 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001183v1_decoy (1144 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001184v1_decoy (1140 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001185v1_decoy (1136 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001186v1_decoy (1134 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001187v1_decoy (1133 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001188v1_decoy (1129 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001189v1_decoy (1127 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001190v1_decoy (1127 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001191v1_decoy (1118 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001192v1_decoy (1110 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001193v1_decoy (1104 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001194v1_decoy (1104 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001195v1_decoy (1101 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001196v1_decoy (1098 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001197v1_decoy (1096 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001198v1_decoy (1094 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001199v1_decoy (1091 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001200v1_decoy (1089 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001201v1_decoy (1086 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001202v1_decoy (1085 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001203v1_decoy (1084 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001204v1_decoy (1083 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001205v1_decoy (1083 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001206v1_decoy (1079 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001207v1_decoy (1076 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001208v1_decoy (1069 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001209v1_decoy (1068 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001210v1_decoy (1067 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001211v1_decoy (1067 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001212v1_decoy (1067 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001213v1_decoy (1063 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001214v1_decoy (1062 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001215v1_decoy (1059 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001216v1_decoy (1058 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001217v1_decoy (1058 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001218v1_decoy (1055 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001219v1_decoy (1054 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001220v1_decoy (1054 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001221v1_decoy (1053 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001222v1_decoy (1053 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001223v1_decoy (1052 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001224v1_decoy (1051 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001225v1_decoy (1049 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001226v1_decoy (1047 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001227v1_decoy (1044 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001228v1_decoy (1043 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001229v1_decoy (1043 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001230v1_decoy (1042 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001231v1_decoy (1042 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001232v1_decoy (1041 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001233v1_decoy (1040 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001234v1_decoy (1039 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001235v1_decoy (1038 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001236v1_decoy (1037 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001237v1_decoy (1037 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001238v1_decoy (1035 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001239v1_decoy (1027 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001240v1_decoy (1021 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001241v1_decoy (1021 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001242v1_decoy (1019 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001243v1_decoy (1019 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001244v1_decoy (1016 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001245v1_decoy (1014 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001246v1_decoy (1013 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001247v1_decoy (1009 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001248v1_decoy (1008 bp)\r\n",
      "01:28:29.394 DEBUG GenomeLocParser -  chrUn_JTFH01001249v1_decoy (1007 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001250v1_decoy (1004 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001251v1_decoy (1004 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001252v1_decoy (1003 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001253v1_decoy (1001 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001254v1_decoy (1000 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001255v1_decoy (1000 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001256v1_decoy (1000 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001257v1_decoy (17929 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001258v1_decoy (9749 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001259v1_decoy (8053 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001260v1_decoy (7826 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001261v1_decoy (7768 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001262v1_decoy (5691 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001263v1_decoy (5444 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001264v1_decoy (5077 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001265v1_decoy (4990 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001266v1_decoy (4545 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001267v1_decoy (4544 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001268v1_decoy (4202 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001269v1_decoy (4195 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001270v1_decoy (3807 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001271v1_decoy (3741 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001272v1_decoy (3699 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001273v1_decoy (3640 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001274v1_decoy (3531 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001275v1_decoy (3455 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001276v1_decoy (3411 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001277v1_decoy (3387 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001278v1_decoy (3358 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001279v1_decoy (3285 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001280v1_decoy (3273 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001281v1_decoy (3262 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001282v1_decoy (3259 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001283v1_decoy (3222 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001284v1_decoy (3127 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001285v1_decoy (3110 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001286v1_decoy (3104 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001287v1_decoy (3071 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001288v1_decoy (3063 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001289v1_decoy (3059 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001290v1_decoy (2990 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001291v1_decoy (2986 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001292v1_decoy (2928 bp)\r\n",
      "01:28:29.395 DEBUG GenomeLocParser -  chrUn_JTFH01001293v1_decoy (2922 bp)\r\n",
      "01:28:29.396 DEBUG GenomeLocParser -  chrUn_JTFH01001294v1_decoy (2875 bp)\r\n",
      "01:28:29.396 DEBUG GenomeLocParser -  chrUn_JTFH01001295v1_decoy (2859 bp)\r\n",
      "01:28:29.396 DEBUG GenomeLocParser -  chrUn_JTFH01001296v1_decoy (2850 bp)\r\n",
      "01:28:29.396 DEBUG GenomeLocParser -  chrUn_JTFH01001297v1_decoy (2813 bp)\r\n",
      "01:28:29.396 DEBUG GenomeLocParser -  chrUn_JTFH01001298v1_decoy (2785 bp)\r\n",
      "01:28:29.396 DEBUG GenomeLocParser -  chrUn_JTFH01001299v1_decoy (2736 bp)\r\n",
      "01:28:29.396 DEBUG GenomeLocParser -  chrUn_JTFH01001300v1_decoy (2688 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001301v1_decoy (2658 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001302v1_decoy (2643 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001303v1_decoy (2618 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001304v1_decoy (2605 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001305v1_decoy (2583 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001306v1_decoy (2534 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001307v1_decoy (2512 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001308v1_decoy (2500 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001309v1_decoy (2481 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001310v1_decoy (2478 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001311v1_decoy (2473 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001312v1_decoy (2467 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001313v1_decoy (2442 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001314v1_decoy (2430 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001315v1_decoy (2417 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001316v1_decoy (2408 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001317v1_decoy (2395 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001318v1_decoy (2352 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001319v1_decoy (2337 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001320v1_decoy (2322 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001321v1_decoy (2307 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001322v1_decoy (2306 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001323v1_decoy (2292 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001324v1_decoy (2271 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001325v1_decoy (2265 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001326v1_decoy (2260 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001327v1_decoy (2240 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001328v1_decoy (2238 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001329v1_decoy (2228 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001330v1_decoy (2215 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001331v1_decoy (2205 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001332v1_decoy (2191 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001333v1_decoy (2191 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001334v1_decoy (2190 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001335v1_decoy (2184 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001336v1_decoy (2166 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001337v1_decoy (2165 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001338v1_decoy (2162 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001339v1_decoy (2146 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001340v1_decoy (2116 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001341v1_decoy (2112 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001342v1_decoy (2108 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001343v1_decoy (2106 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001344v1_decoy (2106 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001345v1_decoy (2106 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001346v1_decoy (2097 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001347v1_decoy (2081 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001348v1_decoy (2058 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001349v1_decoy (2055 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001350v1_decoy (2054 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001351v1_decoy (2037 bp)\r\n",
      "01:28:29.397 DEBUG GenomeLocParser -  chrUn_JTFH01001352v1_decoy (2032 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001353v1_decoy (2032 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001354v1_decoy (2020 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001355v1_decoy (2018 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001356v1_decoy (2014 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001357v1_decoy (2001 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001358v1_decoy (2001 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001359v1_decoy (1991 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001360v1_decoy (1990 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001361v1_decoy (1983 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001362v1_decoy (1981 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001363v1_decoy (1981 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001364v1_decoy (1979 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001365v1_decoy (1963 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001366v1_decoy (1932 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001367v1_decoy (1929 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001368v1_decoy (1881 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001369v1_decoy (1874 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001370v1_decoy (1849 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001371v1_decoy (1849 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001372v1_decoy (1833 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001373v1_decoy (1832 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001374v1_decoy (1826 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001375v1_decoy (1814 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001376v1_decoy (1814 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001377v1_decoy (1791 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001378v1_decoy (1789 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001379v1_decoy (1786 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001380v1_decoy (1778 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001381v1_decoy (1776 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001382v1_decoy (1762 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001383v1_decoy (1758 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001384v1_decoy (1757 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001385v1_decoy (1754 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001386v1_decoy (1752 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001387v1_decoy (1751 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001388v1_decoy (1749 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001389v1_decoy (1738 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001390v1_decoy (1729 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001391v1_decoy (1726 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001392v1_decoy (1716 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001393v1_decoy (1712 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001394v1_decoy (1711 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001395v1_decoy (1703 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001396v1_decoy (1702 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001397v1_decoy (1699 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001398v1_decoy (1686 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001399v1_decoy (1684 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001400v1_decoy (1680 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001401v1_decoy (1678 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001402v1_decoy (1678 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001403v1_decoy (1677 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001404v1_decoy (1676 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001405v1_decoy (1672 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001406v1_decoy (1669 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001407v1_decoy (1668 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001408v1_decoy (1663 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001409v1_decoy (1660 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001410v1_decoy (1660 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001411v1_decoy (1658 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001412v1_decoy (1656 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001413v1_decoy (1656 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001414v1_decoy (1652 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001415v1_decoy (1647 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001416v1_decoy (1645 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001417v1_decoy (1641 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001418v1_decoy (1638 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001419v1_decoy (1633 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001420v1_decoy (1626 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001421v1_decoy (1614 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001422v1_decoy (1612 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001423v1_decoy (1605 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001424v1_decoy (1603 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001425v1_decoy (1599 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001426v1_decoy (1589 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001427v1_decoy (1588 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001428v1_decoy (1585 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001429v1_decoy (1584 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001430v1_decoy (1584 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001431v1_decoy (1580 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001432v1_decoy (1572 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001433v1_decoy (1570 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001434v1_decoy (1569 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001435v1_decoy (1568 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001436v1_decoy (1567 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001437v1_decoy (1565 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001438v1_decoy (1559 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001439v1_decoy (1559 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001440v1_decoy (1556 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001441v1_decoy (1554 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001442v1_decoy (1549 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001443v1_decoy (1542 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001444v1_decoy (1541 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001445v1_decoy (1538 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001446v1_decoy (1537 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001447v1_decoy (1535 bp)\r\n",
      "01:28:29.398 DEBUG GenomeLocParser -  chrUn_JTFH01001448v1_decoy (1530 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001449v1_decoy (1528 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001450v1_decoy (1522 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001451v1_decoy (1514 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001452v1_decoy (1509 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001453v1_decoy (1507 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001454v1_decoy (1500 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001455v1_decoy (1499 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001456v1_decoy (1499 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001457v1_decoy (1497 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001458v1_decoy (1496 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001459v1_decoy (1488 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001460v1_decoy (1486 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001461v1_decoy (1485 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001462v1_decoy (1481 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001463v1_decoy (1479 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001464v1_decoy (1472 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001465v1_decoy (1472 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001466v1_decoy (1470 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001467v1_decoy (1466 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001468v1_decoy (1465 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001469v1_decoy (1461 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001470v1_decoy (1458 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001471v1_decoy (1457 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001472v1_decoy (1448 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001473v1_decoy (1447 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001474v1_decoy (1444 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001475v1_decoy (1443 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001476v1_decoy (1443 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001477v1_decoy (1438 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001478v1_decoy (1432 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001479v1_decoy (1430 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001480v1_decoy (1430 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001481v1_decoy (1429 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001482v1_decoy (1429 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001483v1_decoy (1429 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001484v1_decoy (1426 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001485v1_decoy (1426 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001486v1_decoy (1420 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001487v1_decoy (1416 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001488v1_decoy (1416 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001489v1_decoy (1415 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001490v1_decoy (1415 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001491v1_decoy (1414 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001492v1_decoy (1413 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001493v1_decoy (1410 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001494v1_decoy (1405 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001495v1_decoy (1402 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001496v1_decoy (1398 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001497v1_decoy (1397 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001498v1_decoy (1395 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001499v1_decoy (1392 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001500v1_decoy (1388 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001501v1_decoy (1386 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001502v1_decoy (1382 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001503v1_decoy (1381 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001504v1_decoy (1379 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001505v1_decoy (1376 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001506v1_decoy (1374 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001507v1_decoy (1374 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001508v1_decoy (1373 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001509v1_decoy (1373 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001510v1_decoy (1372 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001511v1_decoy (1370 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001512v1_decoy (1367 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001513v1_decoy (1365 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001514v1_decoy (1364 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001515v1_decoy (1361 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001516v1_decoy (1361 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001517v1_decoy (1355 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001518v1_decoy (1355 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001519v1_decoy (1354 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001520v1_decoy (1353 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001521v1_decoy (1349 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001522v1_decoy (1345 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001523v1_decoy (1344 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001524v1_decoy (1343 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001525v1_decoy (1338 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001526v1_decoy (1338 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001527v1_decoy (1338 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001528v1_decoy (1336 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001529v1_decoy (1333 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001530v1_decoy (1333 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001531v1_decoy (1332 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001532v1_decoy (1324 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001533v1_decoy (1323 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001534v1_decoy (1323 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001535v1_decoy (1320 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001536v1_decoy (1320 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001537v1_decoy (1317 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001538v1_decoy (1316 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001539v1_decoy (1304 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001540v1_decoy (1304 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001541v1_decoy (1303 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001542v1_decoy (1302 bp)\r\n",
      "01:28:29.399 DEBUG GenomeLocParser -  chrUn_JTFH01001543v1_decoy (1301 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001544v1_decoy (1300 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001545v1_decoy (1298 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001546v1_decoy (1297 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001547v1_decoy (1295 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001548v1_decoy (1284 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001549v1_decoy (1283 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001550v1_decoy (1283 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001551v1_decoy (1279 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001552v1_decoy (1278 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001553v1_decoy (1271 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001554v1_decoy (1271 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001555v1_decoy (1268 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001556v1_decoy (1264 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001557v1_decoy (1263 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001558v1_decoy (1262 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001559v1_decoy (1261 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001560v1_decoy (1260 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001561v1_decoy (1259 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001562v1_decoy (1259 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001563v1_decoy (1258 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001564v1_decoy (1256 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001565v1_decoy (1253 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001566v1_decoy (1248 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001567v1_decoy (1248 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001568v1_decoy (1246 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001569v1_decoy (1246 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001570v1_decoy (1244 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001571v1_decoy (1238 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001572v1_decoy (1238 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001573v1_decoy (1236 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001574v1_decoy (1234 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001575v1_decoy (1234 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001576v1_decoy (1231 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001577v1_decoy (1231 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001578v1_decoy (1230 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001579v1_decoy (1230 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001580v1_decoy (1228 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001581v1_decoy (1227 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001582v1_decoy (1222 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001583v1_decoy (1222 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001584v1_decoy (1221 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001585v1_decoy (1221 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001586v1_decoy (1220 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001587v1_decoy (1218 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001588v1_decoy (1218 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001589v1_decoy (1216 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001590v1_decoy (1216 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001591v1_decoy (1212 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001592v1_decoy (1210 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001593v1_decoy (1209 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001594v1_decoy (1208 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001595v1_decoy (1208 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001596v1_decoy (1206 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001597v1_decoy (1205 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001598v1_decoy (1205 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001599v1_decoy (1202 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001600v1_decoy (1200 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001601v1_decoy (1199 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001602v1_decoy (1198 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001603v1_decoy (1198 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001604v1_decoy (1198 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001605v1_decoy (1195 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001606v1_decoy (1194 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001607v1_decoy (1191 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001608v1_decoy (1189 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001609v1_decoy (1188 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001610v1_decoy (1180 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001611v1_decoy (1180 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001612v1_decoy (1179 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001613v1_decoy (1172 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001614v1_decoy (1168 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001615v1_decoy (1166 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001616v1_decoy (1157 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001617v1_decoy (1156 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001618v1_decoy (1156 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001619v1_decoy (1155 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001620v1_decoy (1154 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001621v1_decoy (1154 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001622v1_decoy (1149 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001623v1_decoy (1143 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001624v1_decoy (1143 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001625v1_decoy (1140 bp)\r\n",
      "01:28:29.400 DEBUG GenomeLocParser -  chrUn_JTFH01001626v1_decoy (1137 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001627v1_decoy (1135 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001628v1_decoy (1135 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001629v1_decoy (1135 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001630v1_decoy (1127 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001631v1_decoy (1127 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001632v1_decoy (1126 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001633v1_decoy (1123 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001634v1_decoy (1123 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001635v1_decoy (1123 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001636v1_decoy (1122 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001637v1_decoy (1122 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001638v1_decoy (1121 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001639v1_decoy (1121 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001640v1_decoy (1119 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001641v1_decoy (1119 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001642v1_decoy (1119 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001643v1_decoy (1118 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001644v1_decoy (1115 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001645v1_decoy (1106 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001646v1_decoy (1106 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001647v1_decoy (1104 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001648v1_decoy (1102 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001649v1_decoy (1101 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001650v1_decoy (1098 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001651v1_decoy (1098 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001652v1_decoy (1096 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001653v1_decoy (1096 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001654v1_decoy (1095 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001655v1_decoy (1093 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001656v1_decoy (1090 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001657v1_decoy (1089 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001658v1_decoy (1087 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001659v1_decoy (1087 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001660v1_decoy (1085 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001661v1_decoy (1085 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001662v1_decoy (1085 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001663v1_decoy (1083 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001664v1_decoy (1080 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001665v1_decoy (1080 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001666v1_decoy (1079 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001667v1_decoy (1079 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001668v1_decoy (1079 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001669v1_decoy (1075 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001670v1_decoy (1074 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001671v1_decoy (1073 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001672v1_decoy (1070 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001673v1_decoy (1068 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001674v1_decoy (1067 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001675v1_decoy (1066 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001676v1_decoy (1066 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001677v1_decoy (1066 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001678v1_decoy (1063 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001679v1_decoy (1063 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001680v1_decoy (1063 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001681v1_decoy (1062 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001682v1_decoy (1058 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001683v1_decoy (1056 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001684v1_decoy (1052 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001685v1_decoy (1051 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001686v1_decoy (1051 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001687v1_decoy (1050 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001688v1_decoy (1048 bp)\r\n",
      "01:28:29.401 DEBUG GenomeLocParser -  chrUn_JTFH01001689v1_decoy (1046 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001690v1_decoy (1046 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001691v1_decoy (1045 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001692v1_decoy (1043 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001693v1_decoy (1038 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001694v1_decoy (1036 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001695v1_decoy (1035 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001696v1_decoy (1035 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001697v1_decoy (1035 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001698v1_decoy (1033 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001699v1_decoy (1032 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001700v1_decoy (1031 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001701v1_decoy (1026 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001702v1_decoy (1026 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001703v1_decoy (1026 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001704v1_decoy (1023 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001705v1_decoy (1022 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001706v1_decoy (1020 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001707v1_decoy (1020 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001708v1_decoy (1020 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001709v1_decoy (1019 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001710v1_decoy (1018 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001711v1_decoy (1018 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001712v1_decoy (1017 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001713v1_decoy (1015 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001714v1_decoy (1015 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001715v1_decoy (1015 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001716v1_decoy (1014 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001717v1_decoy (1014 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001718v1_decoy (1013 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001719v1_decoy (1013 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001720v1_decoy (1013 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001721v1_decoy (1012 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001722v1_decoy (1011 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001723v1_decoy (1011 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001724v1_decoy (1009 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001725v1_decoy (1008 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001726v1_decoy (1008 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001727v1_decoy (1007 bp)\r\n",
      "01:28:29.402 DEBUG GenomeLocParser -  chrUn_JTFH01001728v1_decoy (1007 bp)\r\n",
      "01:28:29.409 DEBUG GenomeLocParser -  chrUn_JTFH01001729v1_decoy (1007 bp)\r\n",
      "01:28:29.411 DEBUG GenomeLocParser -  chrUn_JTFH01001730v1_decoy (1006 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001731v1_decoy (1005 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001732v1_decoy (1003 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001733v1_decoy (1001 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001734v1_decoy (1000 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001735v1_decoy (19311 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001736v1_decoy (11713 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001737v1_decoy (11263 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001738v1_decoy (9779 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001739v1_decoy (9568 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001740v1_decoy (9344 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001741v1_decoy (9188 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001742v1_decoy (9100 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001743v1_decoy (8771 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001744v1_decoy (8690 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001745v1_decoy (8566 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001746v1_decoy (8058 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001747v1_decoy (7759 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001748v1_decoy (7585 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001749v1_decoy (7471 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001750v1_decoy (7461 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001751v1_decoy (7342 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001752v1_decoy (7223 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001753v1_decoy (7064 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001754v1_decoy (6916 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001755v1_decoy (6897 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001756v1_decoy (6880 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001757v1_decoy (6857 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001758v1_decoy (6840 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001759v1_decoy (6728 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001760v1_decoy (6688 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001761v1_decoy (6553 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001762v1_decoy (6396 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001763v1_decoy (6345 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001764v1_decoy (6295 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001765v1_decoy (6266 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001766v1_decoy (6173 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001767v1_decoy (6171 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001768v1_decoy (6120 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001769v1_decoy (6105 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001770v1_decoy (6099 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001771v1_decoy (5893 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001772v1_decoy (5829 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001773v1_decoy (5793 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001774v1_decoy (5776 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001775v1_decoy (5759 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001776v1_decoy (5716 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001777v1_decoy (5708 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001778v1_decoy (5590 bp)\r\n",
      "01:28:29.412 DEBUG GenomeLocParser -  chrUn_JTFH01001779v1_decoy (5566 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001780v1_decoy (5558 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001781v1_decoy (5418 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001782v1_decoy (5375 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001783v1_decoy (5300 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001784v1_decoy (5255 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001785v1_decoy (5157 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001786v1_decoy (5130 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001787v1_decoy (4978 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001788v1_decoy (4957 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001789v1_decoy (4947 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001790v1_decoy (4897 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001791v1_decoy (4867 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001792v1_decoy (4845 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001793v1_decoy (4678 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001794v1_decoy (4641 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001795v1_decoy (4592 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001796v1_decoy (4543 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001797v1_decoy (4532 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001798v1_decoy (4503 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001799v1_decoy (4495 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001800v1_decoy (4444 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001801v1_decoy (4414 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001802v1_decoy (4409 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001803v1_decoy (4302 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001804v1_decoy (4300 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001805v1_decoy (4277 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001806v1_decoy (4173 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001807v1_decoy (4169 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001808v1_decoy (4136 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001809v1_decoy (4101 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001810v1_decoy (4089 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001811v1_decoy (4015 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001812v1_decoy (4000 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001813v1_decoy (3973 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001814v1_decoy (3732 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001815v1_decoy (3709 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001816v1_decoy (3686 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001817v1_decoy (3676 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001818v1_decoy (3673 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001819v1_decoy (3672 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001820v1_decoy (3633 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001821v1_decoy (3633 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001822v1_decoy (3613 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001823v1_decoy (3605 bp)\r\n",
      "01:28:29.413 DEBUG GenomeLocParser -  chrUn_JTFH01001824v1_decoy (3592 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001825v1_decoy (3586 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001826v1_decoy (3584 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001827v1_decoy (3577 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001828v1_decoy (3537 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001829v1_decoy (3510 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001830v1_decoy (3509 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001831v1_decoy (3488 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001832v1_decoy (3473 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001833v1_decoy (3445 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001834v1_decoy (3427 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001835v1_decoy (3395 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001836v1_decoy (3367 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001837v1_decoy (3337 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001838v1_decoy (3324 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001839v1_decoy (3315 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001840v1_decoy (3313 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001841v1_decoy (3283 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001842v1_decoy (3250 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001843v1_decoy (3247 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001844v1_decoy (3237 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001845v1_decoy (3235 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001846v1_decoy (3200 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001847v1_decoy (3195 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001848v1_decoy (3175 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001849v1_decoy (3158 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001850v1_decoy (3143 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001851v1_decoy (3139 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001852v1_decoy (3138 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001853v1_decoy (3136 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001854v1_decoy (3132 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001855v1_decoy (3132 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001856v1_decoy (3095 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001857v1_decoy (3094 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001858v1_decoy (3093 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001859v1_decoy (3059 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001860v1_decoy (2985 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001861v1_decoy (2975 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001862v1_decoy (2967 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001863v1_decoy (2961 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001864v1_decoy (2955 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001865v1_decoy (2935 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001866v1_decoy (2933 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001867v1_decoy (2909 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001868v1_decoy (2904 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001869v1_decoy (2892 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001870v1_decoy (2886 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001871v1_decoy (2885 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001872v1_decoy (2878 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001873v1_decoy (2875 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001874v1_decoy (2861 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001875v1_decoy (2856 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001876v1_decoy (2838 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001877v1_decoy (2801 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001878v1_decoy (2797 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001879v1_decoy (2788 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001880v1_decoy (2773 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001881v1_decoy (2755 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001882v1_decoy (2754 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001883v1_decoy (2743 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001884v1_decoy (2725 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001885v1_decoy (2722 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001886v1_decoy (2682 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001887v1_decoy (2669 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001888v1_decoy (2663 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001889v1_decoy (2652 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001890v1_decoy (2647 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001891v1_decoy (2635 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001892v1_decoy (2633 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001893v1_decoy (2629 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001894v1_decoy (2612 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001895v1_decoy (2599 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001896v1_decoy (2566 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001897v1_decoy (2556 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001898v1_decoy (2551 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001899v1_decoy (2551 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001900v1_decoy (2538 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001901v1_decoy (2538 bp)\r\n",
      "01:28:29.414 DEBUG GenomeLocParser -  chrUn_JTFH01001902v1_decoy (2525 bp)\r\n",
      "01:28:29.415 DEBUG GenomeLocParser -  chrUn_JTFH01001903v1_decoy (2498 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001904v1_decoy (2496 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001905v1_decoy (2483 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001906v1_decoy (2475 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001907v1_decoy (2469 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001908v1_decoy (2455 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001909v1_decoy (2444 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001910v1_decoy (2437 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001911v1_decoy (2435 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001912v1_decoy (2427 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001913v1_decoy (2419 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001914v1_decoy (2413 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001915v1_decoy (2412 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001916v1_decoy (2400 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001917v1_decoy (2399 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001918v1_decoy (2396 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001919v1_decoy (2393 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001920v1_decoy (2386 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001921v1_decoy (2384 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001922v1_decoy (2382 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001923v1_decoy (2382 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001924v1_decoy (2367 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001925v1_decoy (2366 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001926v1_decoy (2362 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001927v1_decoy (2361 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001928v1_decoy (2353 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001929v1_decoy (2349 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001930v1_decoy (2348 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001931v1_decoy (2340 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001932v1_decoy (2339 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001933v1_decoy (2336 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001934v1_decoy (2333 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001935v1_decoy (2330 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001936v1_decoy (2327 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001937v1_decoy (2318 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001938v1_decoy (2293 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001939v1_decoy (2292 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001940v1_decoy (2287 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001941v1_decoy (2274 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001942v1_decoy (2274 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001943v1_decoy (2267 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001944v1_decoy (2260 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001945v1_decoy (2257 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001946v1_decoy (2240 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001947v1_decoy (2239 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001948v1_decoy (2232 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001949v1_decoy (2230 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001950v1_decoy (2230 bp)\r\n",
      "01:28:29.416 DEBUG GenomeLocParser -  chrUn_JTFH01001951v1_decoy (2222 bp)\r\n",
      "01:28:29.417 DEBUG GenomeLocParser -  chrUn_JTFH01001952v1_decoy (2216 bp)\r\n",
      "01:28:29.417 DEBUG GenomeLocParser -  chrUn_JTFH01001953v1_decoy (2214 bp)\r\n",
      "01:28:29.417 DEBUG GenomeLocParser -  chrUn_JTFH01001954v1_decoy (2210 bp)\r\n",
      "01:28:29.417 DEBUG GenomeLocParser -  chrUn_JTFH01001955v1_decoy (2203 bp)\r\n",
      "01:28:29.417 DEBUG GenomeLocParser -  chrUn_JTFH01001956v1_decoy (2197 bp)\r\n",
      "01:28:29.417 DEBUG GenomeLocParser -  chrUn_JTFH01001957v1_decoy (2196 bp)\r\n",
      "01:28:29.417 DEBUG GenomeLocParser -  chrUn_JTFH01001958v1_decoy (2196 bp)\r\n",
      "01:28:29.417 DEBUG GenomeLocParser -  chrUn_JTFH01001959v1_decoy (2179 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001960v1_decoy (2178 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001961v1_decoy (2178 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001962v1_decoy (2172 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001963v1_decoy (2170 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001964v1_decoy (2167 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001965v1_decoy (2167 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001966v1_decoy (2157 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001967v1_decoy (2153 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001968v1_decoy (2151 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001969v1_decoy (2147 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001970v1_decoy (2145 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001971v1_decoy (2142 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001972v1_decoy (2142 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001973v1_decoy (2136 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001974v1_decoy (2130 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001975v1_decoy (2128 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001976v1_decoy (2126 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001977v1_decoy (2126 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001978v1_decoy (2119 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001979v1_decoy (2107 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001980v1_decoy (2091 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001981v1_decoy (2087 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001982v1_decoy (2086 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001983v1_decoy (2083 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001984v1_decoy (2075 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001985v1_decoy (2075 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001986v1_decoy (2072 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001987v1_decoy (2068 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001988v1_decoy (2067 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001989v1_decoy (2055 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001990v1_decoy (2051 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001991v1_decoy (2050 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001992v1_decoy (2033 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001993v1_decoy (2024 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001994v1_decoy (2016 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001995v1_decoy (2011 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001996v1_decoy (2009 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001997v1_decoy (2003 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  chrUn_JTFH01001998v1_decoy (2001 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*01:01:01:01 (3503 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*01:01:01:02N (3291 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*01:01:38L (3374 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*01:02 (3374 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*01:03 (3503 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*01:04N (3136 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*01:09 (3105 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*01:11N (3374 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*01:14 (3095 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*01:16N (2985 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*01:20 (3105 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:01:01:01 (3517 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:01:01:02L (3287 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:01:01:03 (3023 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:01:01:04 (3516 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:02:01 (2917 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:03:01 (3517 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:03:03 (3148 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:05:01 (3517 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:06:01 (3517 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:07:01 (3517 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:10 (3517 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:251 (3517 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:259 (2978 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:264 (3002 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:265 (3148 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:266 (3084 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:269 (3101 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:279 (3103 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:32N (3517 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:376 (3104 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:43N (3218 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:455 (3118 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:48 (3517 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:51 (3109 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:533 (3217 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:53N (3305 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:57 (3054 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:60:01 (3112 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:65 (3387 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:68 (3109 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:77 (3371 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:81 (3309 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:89 (3371 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*02:95 (3388 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*03:01:01:01 (3502 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*03:01:01:02N (3373 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*03:01:01:03 (3094 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*03:02:01 (3502 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*03:11N (3404 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*03:21N (3095 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*03:36N (3142 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*11:01:01 (3503 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*11:01:18 (3503 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*11:02:01 (3503 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*11:05 (3373 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*11:110 (2903 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*11:25 (3073 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*11:50Q (3362 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*11:60 (3241 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*11:69N (3500 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*11:74 (3227 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*11:75 (3184 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*11:77 (3233 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*23:01:01 (3502 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*23:09 (3104 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*23:38N (3020 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:02:01:01 (3502 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:02:01:02L (3502 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:02:01:03 (3075 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:02:03Q (3247 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:02:10 (3356 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:03:01 (3502 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:07:01 (3502 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:08 (3502 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:09N (3502 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:10:01 (3502 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:11N (3503 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:152 (3176 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:20 (3502 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:215 (3116 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:61 (3043 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*24:86N (3415 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*25:01:01 (2917 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*26:01:01 (3517 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*26:11N (3091 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*26:15 (3217 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*26:50 (3141 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*29:01:01:01 (3518 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*29:01:01:02N (3303 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*29:02:01:01 (3518 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*29:02:01:02 (3518 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*29:46 (3310 bp)\r\n",
      "01:28:29.418 DEBUG GenomeLocParser -  HLA-A*30:01:01 (3503 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*30:02:01:01 (2903 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*30:02:01:02 (3374 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*30:04:01 (3503 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*30:89 (2903 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*31:01:02 (3518 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*31:01:23 (2918 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*31:04 (2918 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*31:14N (3090 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*31:46 (3075 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*32:01:01 (3518 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*32:06 (3389 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*33:01:01 (3518 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*33:03:01 (3518 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*33:07 (3389 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*34:01:01 (3517 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*34:02:01 (3096 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*36:01 (2903 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*43:01 (3388 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*66:01:01 (3517 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*66:17 (3075 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:01:01:01 (2930 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:01:01:02 (3517 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:01:02:01 (3517 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:01:02:02 (3388 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:02:01:01 (3517 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:02:01:02 (3506 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:02:01:03 (2909 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:02:02 (2916 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:03:01 (2917 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:08:01 (3120 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:113 (3070 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:17 (3134 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:18N (3237 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:22 (3119 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*68:71 (3198 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*69:01 (2917 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*74:01 (2918 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*74:02:01:01 (2918 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*74:02:01:02 (3518 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*80:01:01:01 (3263 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-A*80:01:01:02 (3055 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*07:02:01 (3323 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*07:05:01 (2676 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*07:06 (2676 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*07:156 (2967 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*07:33:01 (3239 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*07:41 (3266 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*07:44 (3270 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*07:50 (3323 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*08:01:01 (3322 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*08:08N (3035 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*08:132 (2675 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*08:134 (2959 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*08:19N (3322 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*08:20 (3322 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*08:33 (3322 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*08:79 (2676 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*13:01:01 (3324 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*13:02:01 (3324 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*13:02:03 (3323 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*13:02:09 (2919 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*13:08 (3324 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*13:15 (3323 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*13:25 (2689 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*14:01:01 (3312 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*14:02:01 (3312 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*14:07N (3255 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:01:01:01 (3336 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:01:01:02N (1208 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:01:01:03 (3026 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:02:01 (3335 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:03:01 (2689 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:04:01 (3052 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:07:01 (3336 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:108 (3283 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:10:01 (2689 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:11:01 (3336 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:13:01 (2688 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:16:01 (2688 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:17:01:01 (3051 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:17:01:02 (3051 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:18:01 (3336 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:220 (2878 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:25:01 (3335 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:27:01 (2689 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:32:01 (3336 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:42 (3333 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:58 (3336 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:66 (2902 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:77 (3336 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*15:83 (3337 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*18:01:01:01 (3323 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*18:01:01:02 (3323 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*18:02 (2686 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*18:03 (3323 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*18:17N (2979 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*18:26 (3323 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*18:94N (2970 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*27:04:01 (3325 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*27:05:02 (3325 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*27:05:18 (3321 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*27:06 (3325 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*27:07:01 (2677 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*27:131 (3325 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*27:24 (2677 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*27:25 (2677 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*27:32 (3325 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*35:01:01:01 (3327 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*35:01:01:02 (3327 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*35:01:22 (2806 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*35:02:01 (3327 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*35:03:01 (2689 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*35:05:01 (2690 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*35:08:01 (2689 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*35:14:02 (3327 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*35:241 (3042 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*35:41 (3327 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*37:01:01 (3324 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*37:01:05 (2687 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*38:01:01 (3312 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*38:02:01 (3312 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*38:14 (2738 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*39:01:01:01 (3155 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*39:01:01:02L (3153 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*39:01:01:03 (3312 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*39:01:03 (3155 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*39:01:16 (3155 bp)\r\n",
      "01:28:29.419 DEBUG GenomeLocParser -  HLA-B*39:01:21 (3312 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*39:05:01 (2675 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*39:06:02 (2674 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*39:10:01 (3027 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*39:13:02 (3255 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*39:14 (2765 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*39:34 (3254 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*39:38Q (2675 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*40:01:01 (2676 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*40:01:02 (3323 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*40:02:01 (3258 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*40:03 (2677 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*40:06:01:01 (3325 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*40:06:01:02 (3299 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*40:10:01 (3304 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*40:150 (2800 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*40:40 (2677 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*40:72:01 (3283 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*40:79 (3257 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*41:01:01 (3322 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*41:02:01 (3322 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*42:01:01 (3322 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*42:02 (2675 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*42:08 (3165 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*44:02:01:01 (3323 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*44:02:01:02S (3152 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*44:02:01:03 (3152 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*44:02:17 (3323 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*44:02:27 (2872 bp)\r\n",
      "01:28:29.420 DEBUG GenomeLocParser -  HLA-B*44:03:01 (3323 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*44:03:02 (2676 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*44:04 (3239 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*44:09 (3317 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*44:138Q (3043 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*44:150 (2676 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*44:23N (3323 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*44:26 (2804 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*44:46 (3323 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*44:49 (3039 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*44:56N (2676 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*45:01:01 (3338 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*45:04 (3339 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*46:01:01 (3336 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*46:01:05 (2891 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*47:01:01:01 (3041 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*47:01:01:02 (3041 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*48:01:01 (3323 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*48:03:01 (2676 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*48:04 (2676 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*48:08 (3323 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*49:01:01 (3340 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*49:32 (3340 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*50:01:01 (3340 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*51:01:01 (3327 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*51:01:02 (3043 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*51:02:01 (3327 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*51:07:01 (3327 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*51:42 (2962 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*52:01:01:01 (3327 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*52:01:01:02 (3327 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*52:01:01:03 (3327 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*52:01:02 (3327 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*53:01:01 (3327 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*53:11 (3274 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*54:01:01 (3332 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*54:18 (2813 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*55:01:01 (3332 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*55:01:03 (3332 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*55:02:01 (3333 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*55:12 (3332 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*55:24 (3332 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*55:48 (2980 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*56:01:01 (2688 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*56:03 (2688 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*56:04 (2688 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*57:01:01 (3337 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*57:03:01 (2689 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*57:06 (3284 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*57:11 (3306 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*57:29 (3337 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*58:01:01 (3336 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*58:31N (3004 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*59:01:01:01 (3333 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*59:01:01:02 (3332 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*67:01:01 (3312 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*67:01:02 (2675 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*67:02 (3307 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*73:01 (3323 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*78:01:01 (3327 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*81:01 (2676 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-B*82:02:01 (3050 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-C*01:02:01 (3349 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-C*01:02:11 (3057 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-C*01:02:29 (3349 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-C*01:02:30 (3333 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-C*01:03 (3349 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-C*01:06 (2895 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-C*01:08 (3349 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-C*01:14 (2895 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-C*01:21 (2895 bp)\r\n",
      "01:28:29.421 DEBUG GenomeLocParser -  HLA-C*01:30 (3349 bp)\r\n",
      "01:28:29.422 DEBUG GenomeLocParser -  HLA-C*01:40 (2968 bp)\r\n",
      "01:28:29.422 DEBUG GenomeLocParser -  HLA-C*02:02:02:01 (3347 bp)\r\n",
      "01:28:29.422 DEBUG GenomeLocParser -  HLA-C*02:02:02:02 (3347 bp)\r\n",
      "01:28:29.422 DEBUG GenomeLocParser -  HLA-C*02:10 (2893 bp)\r\n",
      "01:28:29.422 DEBUG GenomeLocParser -  HLA-C*02:11 (3320 bp)\r\n",
      "01:28:29.422 DEBUG GenomeLocParser -  HLA-C*02:16:02 (3029 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*02:69 (2933 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*02:85 (3347 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*02:86 (3347 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*02:87 (3064 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:02:01 (2894 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:02:02:01 (3348 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:02:02:02 (2896 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:02:02:03 (3348 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:03:01 (3348 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:04:01:01 (3348 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:04:01:02 (3348 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:04:02 (2877 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:04:04 (2966 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:05 (2894 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:06 (2894 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:100 (3034 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:13:01 (3065 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:20N (3321 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:219 (3070 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:261 (3348 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:40:01 (2894 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:41:02 (3328 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:46 (2997 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*03:61 (2894 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:01:01:01 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:01:01:02 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:01:01:03 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:01:01:04 (3012 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:01:01:05 (2931 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:01:62 (3329 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:03:01 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:06 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:09N (2991 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:128 (3086 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:161 (3237 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:177 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:70 (3058 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*04:71 (3086 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*05:01:01:01 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*05:01:01:02 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*05:08 (3059 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*05:09:01 (3322 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*05:93 (2946 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*06:02:01:01 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*06:02:01:02 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*06:02:01:03 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*06:23 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*06:24 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*06:46N (2987 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:01:01:01 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:01:01:02 (3093 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:01:02 (3352 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:01:19 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:01:27 (3195 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:01:45 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:02:01:01 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:02:01:02 (3074 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:02:01:03 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:02:01:04 (3353 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:02:01:05 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:02:05 (2903 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:02:06 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:02:64 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:04:01 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:04:02 (3343 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:06 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:149 (3098 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:18 (3353 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:19 (3222 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:26 (3069 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:30 (2903 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:32N (3334 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:384 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:385 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:386 (3183 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:391 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:392 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:49 (2935 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:56:02 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:66 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*07:67 (3354 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*08:01:01 (3349 bp)\r\n",
      "01:28:29.423 DEBUG GenomeLocParser -  HLA-C*08:01:03 (2998 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:02:01:01 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:02:01:02 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:03:01 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:04:01 (2895 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:112 (3178 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:20 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:21 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:22 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:24 (2895 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:27 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:36N (3097 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:40 (2978 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:41 (3019 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*08:62 (3086 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*12:02:02 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*12:03:01:01 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*12:03:01:02 (3348 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*12:08 (3066 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*12:13 (3058 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*12:19 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*12:22 (2895 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*12:99 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*14:02:01 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*14:03 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*14:21N (3099 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*14:23 (2976 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*15:02:01 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*15:05:01 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*15:05:02 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*15:13 (2895 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*15:16 (3066 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*15:17 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*15:96Q (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*16:01:01 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*16:02:01 (2895 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*16:04:01 (3349 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*17:01:01:01 (3368 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*17:01:01:02 (3368 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*17:01:01:03 (3368 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*17:03 (3197 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-C*18:01 (3346 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:01:02 (6489 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:02:01:01 (6484 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:02:01:02 (6485 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:02:01:03 (6485 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:02:01:04 (6492 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:03:01:01 (6485 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:03:01:02 (6492 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:04:01:01 (6484 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:04:01:02 (6485 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:05:01 (6485 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:07 (5959 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:10 (5790 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*01:11 (5926 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*02:01 (6403 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*03:01:01 (6437 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*03:02 (6437 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*03:03:01 (6437 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*04:01:02:01 (5853 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*04:01:02:02 (5666 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*04:02 (6210 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*05:01:01:01 (5806 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*05:01:01:02 (6529 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*05:03 (6121 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*05:05:01:01 (6593 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*05:05:01:02 (6597 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*05:05:01:03 (6393 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*05:11 (6589 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQA1*06:01:01 (5878 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*02:01:01 (7480 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*02:02:01 (7471 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*03:01:01:01 (7231 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*03:01:01:02 (7230 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*03:01:01:03 (7231 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*03:02:01 (7126 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*03:03:02:01 (7126 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*03:03:02:02 (7126 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*03:03:02:03 (6800 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*03:05:01 (6934 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*05:01:01:01 (7090 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*05:01:01:02 (7090 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*05:03:01:01 (7089 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*05:03:01:02 (7089 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*06:01:01 (7111 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*06:02:01 (7102 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*06:03:01 (7103 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DQB1*06:09:01 (7102 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*01:01:01 (10741 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*01:02:01 (11229 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*03:01:01:01 (13908 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*03:01:01:02 (13426 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*04:03:01 (15246 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*07:01:01:01 (16110 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*07:01:01:02 (16120 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*08:03:02 (13562 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*09:21 (16039 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*10:01:01 (13501 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*11:01:01 (13921 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*11:01:02 (13931 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*11:04:01 (13919 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*12:01:01 (13404 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*12:17 (11260 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*13:01:01 (13935 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*13:02:01 (13941 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*14:05:01 (13933 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*14:54:01 (13936 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*15:01:01:01 (11080 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*15:01:01:02 (11571 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*15:01:01:03 (11056 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*15:01:01:04 (11056 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*15:02:01 (10313 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*15:03:01:01 (11567 bp)\r\n",
      "01:28:29.424 DEBUG GenomeLocParser -  HLA-DRB1*15:03:01:02 (11569 bp)\r\n",
      "01:28:29.425 DEBUG GenomeLocParser -  HLA-DRB1*16:02:01 (11005 bp)\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:28:29.605 INFO  FeatureManager - Using codec IntervalListCodec to read file file:///home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.cohort.gc.filtered.interval_list\n",
      "01:28:30.210 DEBUG FeatureDataSource - Cache statistics for FeatureInput /home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.cohort.gc.filtered.interval_list:/home/jupyter/notebooks/germline-cnv/sandbox/chr20Y.cohort.gc.filtered.interval_list:\n",
      "01:28:30.216 DEBUG FeatureCache - Cache hit rate  was 0.00% (0 out of 0 total queries)\n",
      "01:28:30.247 INFO  IntervalArgumentCollection - Processing 62398000 bp from intervals\n",
      "01:28:30.544 INFO  DetermineGermlineContigPloidy - Validating and aggregating coverage per contig from input read-count files...\n",
      "01:28:30.554 DEBUG GenomeLocParser - Prepared reference sequence contig dictionary\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr1 (248956422 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr2 (242193529 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr3 (198295559 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr4 (190214555 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr5 (181538259 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr6 (170805979 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr7 (159345973 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr8 (145138636 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr9 (138394717 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr10 (133797422 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr11 (135086622 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr12 (133275309 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr13 (114364328 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr14 (107043718 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr15 (101991189 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr16 (90338345 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr17 (83257441 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr18 (80373285 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr19 (58617616 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr20 (64444167 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr21 (46709983 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chr22 (50818468 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chrX (156040895 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chrY (57227415 bp)\n",
      "01:28:30.555 DEBUG GenomeLocParser -  chrM (16569 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr1_KI270706v1_random (175055 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr1_KI270707v1_random (32032 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr1_KI270708v1_random (127682 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr1_KI270709v1_random (66860 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr1_KI270710v1_random (40176 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr1_KI270711v1_random (42210 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr1_KI270712v1_random (176043 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr1_KI270713v1_random (40745 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr1_KI270714v1_random (41717 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr2_KI270715v1_random (161471 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr2_KI270716v1_random (153799 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr3_GL000221v1_random (155397 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr4_GL000008v2_random (209709 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr5_GL000208v1_random (92689 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr9_KI270717v1_random (40062 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr9_KI270718v1_random (38054 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr9_KI270719v1_random (176845 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr9_KI270720v1_random (39050 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr11_KI270721v1_random (100316 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr14_GL000009v2_random (201709 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr14_GL000225v1_random (211173 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr14_KI270722v1_random (194050 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr14_GL000194v1_random (191469 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr14_KI270723v1_random (38115 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr14_KI270724v1_random (39555 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr14_KI270725v1_random (172810 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr14_KI270726v1_random (43739 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr15_KI270727v1_random (448248 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr16_KI270728v1_random (1872759 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr17_GL000205v2_random (185591 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr17_KI270729v1_random (280839 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr17_KI270730v1_random (112551 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr22_KI270731v1_random (150754 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr22_KI270732v1_random (41543 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr22_KI270733v1_random (179772 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr22_KI270734v1_random (165050 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr22_KI270735v1_random (42811 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr22_KI270736v1_random (181920 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr22_KI270737v1_random (103838 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr22_KI270738v1_random (99375 bp)\n",
      "01:28:30.556 DEBUG GenomeLocParser -  chr22_KI270739v1_random (73985 bp)\n",
      "01:28:30.557 DEBUG GenomeLocParser -  chrY_KI270740v1_random (37240 bp)\n",
      "01:28:30.558 DEBUG GenomeLocParser -  chrUn_KI270302v1 (2274 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270304v1 (2165 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270303v1 (1942 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270305v1 (1472 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270322v1 (21476 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270320v1 (4416 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270310v1 (1201 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270316v1 (1444 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270315v1 (2276 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270312v1 (998 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270311v1 (12399 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270317v1 (37690 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270412v1 (1179 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270411v1 (2646 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270414v1 (2489 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270419v1 (1029 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270418v1 (2145 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270420v1 (2321 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270424v1 (2140 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270417v1 (2043 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270422v1 (1445 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270423v1 (981 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270425v1 (1884 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270429v1 (1361 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270442v1 (392061 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270466v1 (1233 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270465v1 (1774 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270467v1 (3920 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270435v1 (92983 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270438v1 (112505 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270468v1 (4055 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270510v1 (2415 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270509v1 (2318 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270518v1 (2186 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270508v1 (1951 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270516v1 (1300 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270512v1 (22689 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270519v1 (138126 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270522v1 (5674 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270511v1 (8127 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270515v1 (6361 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270507v1 (5353 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270517v1 (3253 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270529v1 (1899 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270528v1 (2983 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270530v1 (2168 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270539v1 (993 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270538v1 (91309 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270544v1 (1202 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270548v1 (1599 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270583v1 (1400 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270587v1 (2969 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270580v1 (1553 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270581v1 (7046 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270579v1 (31033 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270589v1 (44474 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270590v1 (4685 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270584v1 (4513 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270582v1 (6504 bp)\n",
      "01:28:30.559 DEBUG GenomeLocParser -  chrUn_KI270588v1 (6158 bp)\n",
      "01:28:30.560 DEBUG GenomeLocParser -  chrUn_KI270593v1 (3041 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270591v1 (5796 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270330v1 (1652 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270329v1 (1040 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270334v1 (1368 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270333v1 (2699 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270335v1 (1048 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270338v1 (1428 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270340v1 (1428 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270336v1 (1026 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270337v1 (1121 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270363v1 (1803 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270364v1 (2855 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270362v1 (3530 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270366v1 (8320 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270378v1 (1048 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270379v1 (1045 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270389v1 (1298 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270390v1 (2387 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270387v1 (1537 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270395v1 (1143 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270396v1 (1880 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270388v1 (1216 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270394v1 (970 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270386v1 (1788 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270391v1 (1484 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270383v1 (1750 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270393v1 (1308 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270384v1 (1658 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270392v1 (971 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270381v1 (1930 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270385v1 (990 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270382v1 (4215 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270376v1 (1136 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270374v1 (2656 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270372v1 (1650 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270373v1 (1451 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270375v1 (2378 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270371v1 (2805 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270448v1 (7992 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270521v1 (7642 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_GL000195v1 (182896 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_GL000219v1 (179198 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_GL000220v1 (161802 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_GL000224v1 (179693 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270741v1 (157432 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_GL000226v1 (15008 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_GL000213v1 (164239 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270743v1 (210658 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270744v1 (168472 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270745v1 (41891 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270746v1 (66486 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270747v1 (198735 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270748v1 (93321 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270749v1 (158759 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270750v1 (148850 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270751v1 (150742 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270752v1 (27745 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270753v1 (62944 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270754v1 (40191 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270755v1 (36723 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270756v1 (79590 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270757v1 (71251 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_GL000214v1 (137718 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_KI270742v1 (186739 bp)\n",
      "01:28:30.561 DEBUG GenomeLocParser -  chrUn_GL000216v2 (176608 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chrUn_GL000218v1 (161147 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr1_KI270762v1_alt (354444 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr1_KI270766v1_alt (256271 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr1_KI270760v1_alt (109528 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr1_KI270765v1_alt (185285 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr1_GL383518v1_alt (182439 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr1_GL383519v1_alt (110268 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr1_GL383520v2_alt (366580 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr1_KI270764v1_alt (50258 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr1_KI270763v1_alt (911658 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr1_KI270759v1_alt (425601 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr1_KI270761v1_alt (165834 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_KI270770v1_alt (136240 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_KI270773v1_alt (70887 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_KI270774v1_alt (223625 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_KI270769v1_alt (120616 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_GL383521v1_alt (143390 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_KI270772v1_alt (133041 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_KI270775v1_alt (138019 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_KI270771v1_alt (110395 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_KI270768v1_alt (110099 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_GL582966v2_alt (96131 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_GL383522v1_alt (123821 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_KI270776v1_alt (174166 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr2_KI270767v1_alt (161578 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr3_JH636055v2_alt (173151 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr3_KI270783v1_alt (109187 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr3_KI270780v1_alt (224108 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr3_GL383526v1_alt (180671 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr3_KI270777v1_alt (173649 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr3_KI270778v1_alt (248252 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr3_KI270781v1_alt (113034 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr3_KI270779v1_alt (205312 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr3_KI270782v1_alt (162429 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr3_KI270784v1_alt (184404 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr4_KI270790v1_alt (220246 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr4_GL383528v1_alt (376187 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr4_KI270787v1_alt (111943 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr4_GL000257v2_alt (586476 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr4_KI270788v1_alt (158965 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr4_GL383527v1_alt (164536 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr4_KI270785v1_alt (119912 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr4_KI270789v1_alt (205944 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr4_KI270786v1_alt (244096 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr5_KI270793v1_alt (126136 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr5_KI270792v1_alt (179043 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr5_KI270791v1_alt (195710 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr5_GL383532v1_alt (82728 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr5_GL949742v1_alt (226852 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr5_KI270794v1_alt (164558 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr5_GL339449v2_alt (1612928 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr5_GL383530v1_alt (101241 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr5_KI270796v1_alt (172708 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr5_GL383531v1_alt (173459 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr5_KI270795v1_alt (131892 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr6_GL000250v2_alt (4672374 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr6_KI270800v1_alt (175808 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr6_KI270799v1_alt (152148 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr6_GL383533v1_alt (124736 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr6_KI270801v1_alt (870480 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr6_KI270802v1_alt (75005 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr6_KB021644v2_alt (185823 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr6_KI270797v1_alt (197536 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr6_KI270798v1_alt (271782 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr7_KI270804v1_alt (157952 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr7_KI270809v1_alt (209586 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr7_KI270806v1_alt (158166 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr7_GL383534v2_alt (119183 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr7_KI270803v1_alt (1111570 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr7_KI270808v1_alt (271455 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr7_KI270807v1_alt (126434 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr7_KI270805v1_alt (209988 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270818v1_alt (145606 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270812v1_alt (282736 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270811v1_alt (292436 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270821v1_alt (985506 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270813v1_alt (300230 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270822v1_alt (624492 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270814v1_alt (141812 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270810v1_alt (374415 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270819v1_alt (133535 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270820v1_alt (36640 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270817v1_alt (158983 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270816v1_alt (305841 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr8_KI270815v1_alt (132244 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr9_GL383539v1_alt (162988 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr9_GL383540v1_alt (71551 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr9_GL383541v1_alt (171286 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr9_GL383542v1_alt (60032 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr9_KI270823v1_alt (439082 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr10_GL383545v1_alt (179254 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr10_KI270824v1_alt (181496 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr10_GL383546v1_alt (309802 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr10_KI270825v1_alt (188315 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr11_KI270832v1_alt (210133 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr11_KI270830v1_alt (177092 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr11_KI270831v1_alt (296895 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr11_KI270829v1_alt (204059 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr11_GL383547v1_alt (154407 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr11_JH159136v1_alt (200998 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr11_JH159137v1_alt (191409 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr11_KI270827v1_alt (67707 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr11_KI270826v1_alt (186169 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr12_GL877875v1_alt (167313 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr12_GL877876v1_alt (408271 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr12_KI270837v1_alt (40090 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr12_GL383549v1_alt (120804 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr12_KI270835v1_alt (238139 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr12_GL383550v2_alt (169178 bp)\n",
      "01:28:30.562 DEBUG GenomeLocParser -  chr12_GL383552v1_alt (138655 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr12_GL383553v2_alt (152874 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr12_KI270834v1_alt (119498 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr12_GL383551v1_alt (184319 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr12_KI270833v1_alt (76061 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr12_KI270836v1_alt (56134 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr13_KI270840v1_alt (191684 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr13_KI270839v1_alt (180306 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr13_KI270843v1_alt (103832 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr13_KI270841v1_alt (169134 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr13_KI270838v1_alt (306913 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr13_KI270842v1_alt (37287 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr14_KI270844v1_alt (322166 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr14_KI270847v1_alt (1511111 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr14_KI270845v1_alt (180703 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr14_KI270846v1_alt (1351393 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr15_KI270852v1_alt (478999 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr15_KI270851v1_alt (263054 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr15_KI270848v1_alt (327382 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr15_GL383554v1_alt (296527 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr15_KI270849v1_alt (244917 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr15_GL383555v2_alt (388773 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr15_KI270850v1_alt (430880 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr16_KI270854v1_alt (134193 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr16_KI270856v1_alt (63982 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr16_KI270855v1_alt (232857 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr16_KI270853v1_alt (2659700 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr16_GL383556v1_alt (192462 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr16_GL383557v1_alt (89672 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_GL383563v3_alt (375691 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_KI270862v1_alt (391357 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_KI270861v1_alt (196688 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_KI270857v1_alt (2877074 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_JH159146v1_alt (278131 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_JH159147v1_alt (70345 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_GL383564v2_alt (133151 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_GL000258v2_alt (1821992 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_GL383565v1_alt (223995 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_KI270858v1_alt (235827 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_KI270859v1_alt (108763 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_GL383566v1_alt (90219 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_KI270860v1_alt (178921 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr18_KI270864v1_alt (111737 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr18_GL383567v1_alt (289831 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr18_GL383570v1_alt (164789 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr18_GL383571v1_alt (198278 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr18_GL383568v1_alt (104552 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr18_GL383569v1_alt (167950 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr18_GL383572v1_alt (159547 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr18_KI270863v1_alt (167999 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270868v1_alt (61734 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270865v1_alt (52969 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_GL383573v1_alt (385657 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_GL383575v2_alt (170222 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_GL383576v1_alt (188024 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_GL383574v1_alt (155864 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270866v1_alt (43156 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270867v1_alt (233762 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_GL949746v1_alt (987716 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr20_GL383577v2_alt (128386 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr20_KI270869v1_alt (118774 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr20_KI270871v1_alt (58661 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr20_KI270870v1_alt (183433 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr21_GL383578v2_alt (63917 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr21_KI270874v1_alt (166743 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr21_KI270873v1_alt (143900 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr21_GL383579v2_alt (201197 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr21_GL383580v2_alt (74653 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr21_GL383581v2_alt (116689 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr21_KI270872v1_alt (82692 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr22_KI270875v1_alt (259914 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr22_KI270878v1_alt (186262 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr22_KI270879v1_alt (304135 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr22_KI270876v1_alt (263666 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr22_KI270877v1_alt (101331 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr22_GL383583v2_alt (96924 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr22_GL383582v2_alt (162811 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chrX_KI270880v1_alt (284869 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chrX_KI270881v1_alt (144206 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270882v1_alt (248807 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270883v1_alt (170399 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270884v1_alt (157053 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270885v1_alt (171027 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270886v1_alt (204239 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270887v1_alt (209512 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270888v1_alt (155532 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270889v1_alt (170698 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270890v1_alt (184499 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr19_KI270891v1_alt (170680 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr1_KI270892v1_alt (162212 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr2_KI270894v1_alt (214158 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr2_KI270893v1_alt (161218 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr3_KI270895v1_alt (162896 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr4_KI270896v1_alt (378547 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr5_KI270897v1_alt (1144418 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr5_KI270898v1_alt (130957 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr6_GL000251v2_alt (4795265 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr7_KI270899v1_alt (190869 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr8_KI270901v1_alt (136959 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr8_KI270900v1_alt (318687 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr11_KI270902v1_alt (106711 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr11_KI270903v1_alt (214625 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr12_KI270904v1_alt (572349 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr15_KI270906v1_alt (196384 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr15_KI270905v1_alt (5161414 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_KI270907v1_alt (137721 bp)\n",
      "01:28:30.563 DEBUG GenomeLocParser -  chr17_KI270910v1_alt (157099 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr17_KI270909v1_alt (325800 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr17_JH159148v1_alt (88070 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr17_KI270908v1_alt (1423190 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr18_KI270912v1_alt (174061 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr18_KI270911v1_alt (157710 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_GL949747v2_alt (729520 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr22_KB663609v1_alt (74013 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrX_KI270913v1_alt (274009 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270914v1_alt (205194 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270915v1_alt (170665 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270916v1_alt (184516 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270917v1_alt (190932 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270918v1_alt (123111 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270919v1_alt (170701 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270920v1_alt (198005 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270921v1_alt (282224 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270922v1_alt (187935 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270923v1_alt (189352 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr3_KI270924v1_alt (166540 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr4_KI270925v1_alt (555799 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr6_GL000252v2_alt (4604811 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr8_KI270926v1_alt (229282 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr11_KI270927v1_alt (218612 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_GL949748v2_alt (1064304 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr22_KI270928v1_alt (176103 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270929v1_alt (186203 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270930v1_alt (200773 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270931v1_alt (170148 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270932v1_alt (215732 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270933v1_alt (170537 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_GL000209v2_alt (177381 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr3_KI270934v1_alt (163458 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr6_GL000253v2_alt (4677643 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_GL949749v2_alt (1091841 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr3_KI270935v1_alt (197351 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr6_GL000254v2_alt (4827813 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_GL949750v2_alt (1066390 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr3_KI270936v1_alt (164170 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr6_GL000255v2_alt (4606388 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_GL949751v2_alt (1002683 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr3_KI270937v1_alt (165607 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr6_GL000256v2_alt (4929269 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_GL949752v1_alt (987100 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr6_KI270758v1_alt (76752 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_GL949753v2_alt (796479 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chr19_KI270938v1_alt (1066800 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrEBV (171823 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707606v1_decoy (2200 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707607v1_decoy (3033 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707608v1_decoy (3112 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707609v1_decoy (1642 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707610v1_decoy (1393 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707611v1_decoy (1103 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707612v1_decoy (1039 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707613v1_decoy (1619 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707614v1_decoy (3122 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707615v1_decoy (1934 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707616v1_decoy (3111 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707617v1_decoy (2545 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707618v1_decoy (2295 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707619v1_decoy (1551 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707620v1_decoy (2046 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707621v1_decoy (1222 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707622v1_decoy (1535 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707623v1_decoy (3784 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707624v1_decoy (1329 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707625v1_decoy (1238 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707626v1_decoy (5623 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707627v1_decoy (5821 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707628v1_decoy (2960 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707629v1_decoy (1848 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707630v1_decoy (2315 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707631v1_decoy (1945 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707632v1_decoy (1424 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707633v1_decoy (1274 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707634v1_decoy (1007 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707635v1_decoy (1414 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707636v1_decoy (1725 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707637v1_decoy (5354 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707638v1_decoy (2189 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707639v1_decoy (1294 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707640v1_decoy (1831 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707641v1_decoy (1647 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707642v1_decoy (2943 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707643v1_decoy (2857 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707644v1_decoy (1030 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707645v1_decoy (1070 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707646v1_decoy (1735 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707647v1_decoy (1982 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707648v1_decoy (1564 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707649v1_decoy (1775 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707650v1_decoy (1540 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707651v1_decoy (2013 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707652v1_decoy (1176 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707653v1_decoy (1890 bp)\n",
      "01:28:30.564 DEBUG GenomeLocParser -  chrUn_KN707654v1_decoy (3644 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707655v1_decoy (2785 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707656v1_decoy (1017 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707657v1_decoy (1068 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707658v1_decoy (1007 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707659v1_decoy (2605 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707660v1_decoy (8410 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707661v1_decoy (5534 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707662v1_decoy (2173 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707663v1_decoy (1065 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707664v1_decoy (8683 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707665v1_decoy (2670 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707666v1_decoy (2420 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707667v1_decoy (2189 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707668v1_decoy (2093 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707669v1_decoy (1184 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707670v1_decoy (1205 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707671v1_decoy (2786 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707672v1_decoy (2794 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707673v1_decoy (19544 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707674v1_decoy (2848 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707675v1_decoy (10556 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707676v1_decoy (9066 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707677v1_decoy (7267 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707678v1_decoy (2462 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707679v1_decoy (1774 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707680v1_decoy (1297 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707681v1_decoy (4379 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707682v1_decoy (4208 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707683v1_decoy (4068 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707684v1_decoy (2940 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707685v1_decoy (3938 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707686v1_decoy (2072 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707687v1_decoy (1136 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707688v1_decoy (4248 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707689v1_decoy (5823 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707690v1_decoy (3715 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707691v1_decoy (4885 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707692v1_decoy (4813 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707693v1_decoy (2899 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707694v1_decoy (1228 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707695v1_decoy (3119 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707696v1_decoy (3828 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707697v1_decoy (1186 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707698v1_decoy (1908 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707699v1_decoy (2795 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707700v1_decoy (3703 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707701v1_decoy (6722 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707702v1_decoy (6466 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707703v1_decoy (2235 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707704v1_decoy (2871 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707705v1_decoy (4632 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707706v1_decoy (4225 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707707v1_decoy (4339 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707708v1_decoy (2305 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707709v1_decoy (3273 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707710v1_decoy (5701 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707711v1_decoy (4154 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707712v1_decoy (1243 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707713v1_decoy (1308 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707714v1_decoy (2922 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707715v1_decoy (3044 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707716v1_decoy (2888 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707717v1_decoy (1742 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707718v1_decoy (4969 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707719v1_decoy (3270 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707720v1_decoy (6028 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707721v1_decoy (1105 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707722v1_decoy (2884 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707723v1_decoy (1124 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707724v1_decoy (1454 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707725v1_decoy (2565 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707726v1_decoy (2149 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707727v1_decoy (2630 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707728v1_decoy (14625 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707729v1_decoy (7431 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707730v1_decoy (5776 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707731v1_decoy (4820 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707732v1_decoy (1227 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707733v1_decoy (7503 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707734v1_decoy (9652 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707735v1_decoy (1091 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707736v1_decoy (2467 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707737v1_decoy (1270 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707738v1_decoy (4365 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707739v1_decoy (4284 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707740v1_decoy (10282 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707741v1_decoy (5601 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707742v1_decoy (4758 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707743v1_decoy (1624 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707744v1_decoy (4024 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707745v1_decoy (1276 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707746v1_decoy (5083 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707747v1_decoy (2075 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707748v1_decoy (3553 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707749v1_decoy (7010 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707750v1_decoy (4718 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707751v1_decoy (3546 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707752v1_decoy (2873 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707753v1_decoy (2144 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707754v1_decoy (2243 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707755v1_decoy (5343 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707756v1_decoy (4877 bp)\n",
      "01:28:30.565 DEBUG GenomeLocParser -  chrUn_KN707757v1_decoy (3034 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707758v1_decoy (2826 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707759v1_decoy (1221 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707760v1_decoy (1169 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707761v1_decoy (2319 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707762v1_decoy (3450 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707763v1_decoy (2674 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707764v1_decoy (3912 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707765v1_decoy (6020 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707766v1_decoy (2303 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707767v1_decoy (2552 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707768v1_decoy (3656 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707769v1_decoy (1591 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707770v1_decoy (1209 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707771v1_decoy (3176 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707772v1_decoy (8915 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707773v1_decoy (4902 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707774v1_decoy (3324 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707775v1_decoy (5997 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707776v1_decoy (2618 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707777v1_decoy (10311 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707778v1_decoy (2440 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707779v1_decoy (12444 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707780v1_decoy (5691 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707781v1_decoy (2717 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707782v1_decoy (5277 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707783v1_decoy (4373 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707784v1_decoy (3224 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707785v1_decoy (2631 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707786v1_decoy (5385 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707787v1_decoy (3678 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707788v1_decoy (1412 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707789v1_decoy (1443 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707790v1_decoy (1098 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707791v1_decoy (3240 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707792v1_decoy (1915 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707793v1_decoy (4667 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707794v1_decoy (7219 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707795v1_decoy (3277 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707796v1_decoy (3473 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707797v1_decoy (4243 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707798v1_decoy (17599 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707799v1_decoy (5095 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707800v1_decoy (2237 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707801v1_decoy (2901 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707802v1_decoy (2666 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707803v1_decoy (5336 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707804v1_decoy (4383 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707805v1_decoy (5446 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707806v1_decoy (6252 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707807v1_decoy (4616 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707808v1_decoy (3021 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707809v1_decoy (3667 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707810v1_decoy (4563 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707811v1_decoy (1120 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707812v1_decoy (3845 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707813v1_decoy (2272 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707814v1_decoy (4764 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707815v1_decoy (5410 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707816v1_decoy (7150 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707817v1_decoy (1762 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707818v1_decoy (1207 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707819v1_decoy (1331 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707820v1_decoy (8307 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707821v1_decoy (2276 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707822v1_decoy (2575 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707823v1_decoy (3970 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707824v1_decoy (1352 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707825v1_decoy (3040 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707826v1_decoy (2070 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707827v1_decoy (2913 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707828v1_decoy (2389 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707829v1_decoy (1835 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707830v1_decoy (4807 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707831v1_decoy (2201 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707832v1_decoy (1265 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707833v1_decoy (1961 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707834v1_decoy (1064 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707835v1_decoy (1932 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707836v1_decoy (3213 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707837v1_decoy (1178 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707838v1_decoy (2926 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707839v1_decoy (1038 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707840v1_decoy (3298 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707841v1_decoy (8992 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707842v1_decoy (6698 bp)\n",
      "01:28:30.566 DEBUG GenomeLocParser -  chrUn_KN707843v1_decoy (4880 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707844v1_decoy (1766 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707845v1_decoy (3532 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707846v1_decoy (2297 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707847v1_decoy (1234 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707848v1_decoy (1205 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707849v1_decoy (2790 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707850v1_decoy (2006 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707851v1_decoy (4593 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707852v1_decoy (1579 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707853v1_decoy (9597 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707854v1_decoy (10451 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707855v1_decoy (3219 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707856v1_decoy (2300 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707857v1_decoy (5985 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707858v1_decoy (2959 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707859v1_decoy (1340 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707860v1_decoy (3148 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707861v1_decoy (2242 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707862v1_decoy (16513 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707863v1_decoy (7821 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707864v1_decoy (2159 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707865v1_decoy (2114 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707866v1_decoy (4109 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707867v1_decoy (1544 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707868v1_decoy (1005 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707869v1_decoy (8632 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707870v1_decoy (1012 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707871v1_decoy (4728 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707872v1_decoy (2165 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707873v1_decoy (7591 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707874v1_decoy (5202 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707875v1_decoy (4241 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707876v1_decoy (4131 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707877v1_decoy (2272 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707878v1_decoy (2085 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707879v1_decoy (4346 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707880v1_decoy (1208 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707881v1_decoy (4543 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707882v1_decoy (2772 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707883v1_decoy (2490 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707884v1_decoy (4568 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707885v1_decoy (1776 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707886v1_decoy (2699 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707887v1_decoy (3534 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707888v1_decoy (2424 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707889v1_decoy (1747 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707890v1_decoy (1088 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707891v1_decoy (1143 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707892v1_decoy (2530 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707893v1_decoy (8049 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707894v1_decoy (1366 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707895v1_decoy (4284 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707896v1_decoy (33125 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707897v1_decoy (2137 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707898v1_decoy (3840 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707899v1_decoy (3087 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707900v1_decoy (2041 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707901v1_decoy (3344 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707902v1_decoy (2921 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707903v1_decoy (6581 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707904v1_decoy (3968 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707905v1_decoy (2339 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707906v1_decoy (1243 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707907v1_decoy (7776 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707908v1_decoy (19837 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707909v1_decoy (1737 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707910v1_decoy (1098 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707911v1_decoy (1893 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707912v1_decoy (1281 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707913v1_decoy (1527 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707914v1_decoy (2055 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707915v1_decoy (2527 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707916v1_decoy (3275 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707917v1_decoy (1265 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707918v1_decoy (2623 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707919v1_decoy (4850 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707920v1_decoy (3584 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707921v1_decoy (2561 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707922v1_decoy (3041 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707923v1_decoy (1409 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707924v1_decoy (4596 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707925v1_decoy (11555 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707926v1_decoy (1266 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707927v1_decoy (1079 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707928v1_decoy (1087 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707929v1_decoy (1226 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707930v1_decoy (1131 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707931v1_decoy (1199 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707932v1_decoy (1084 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707933v1_decoy (2038 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707934v1_decoy (1070 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707935v1_decoy (1312 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707936v1_decoy (4031 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707937v1_decoy (7445 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707938v1_decoy (1770 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707939v1_decoy (5600 bp)\n",
      "01:28:30.567 DEBUG GenomeLocParser -  chrUn_KN707940v1_decoy (1882 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707941v1_decoy (1170 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707942v1_decoy (1300 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707943v1_decoy (5325 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707944v1_decoy (2043 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707945v1_decoy (1072 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707946v1_decoy (2463 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707947v1_decoy (1010 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707948v1_decoy (1432 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707949v1_decoy (1162 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707950v1_decoy (1095 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707951v1_decoy (1118 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707952v1_decoy (1383 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707953v1_decoy (2289 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707954v1_decoy (1648 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707955v1_decoy (2203 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707956v1_decoy (3270 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707957v1_decoy (11499 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707958v1_decoy (2474 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707959v1_decoy (2294 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707960v1_decoy (1238 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707961v1_decoy (3410 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707962v1_decoy (1523 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707963v1_decoy (62955 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707964v1_decoy (6282 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707965v1_decoy (3836 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707966v1_decoy (6486 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707967v1_decoy (15368 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707968v1_decoy (9572 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707969v1_decoy (6413 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707970v1_decoy (4104 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707971v1_decoy (12943 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707972v1_decoy (4650 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707973v1_decoy (3080 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707974v1_decoy (3134 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707975v1_decoy (6211 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707976v1_decoy (1126 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707977v1_decoy (1101 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707978v1_decoy (1101 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707979v1_decoy (2648 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707980v1_decoy (2973 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707981v1_decoy (2520 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707982v1_decoy (2318 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707983v1_decoy (2606 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707984v1_decoy (2205 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707985v1_decoy (2929 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707986v1_decoy (3869 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707987v1_decoy (1117 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707988v1_decoy (2960 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707989v1_decoy (1009 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707990v1_decoy (4048 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707991v1_decoy (2193 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_KN707992v1_decoy (1830 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000001v1_decoy (25139 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000002v1_decoy (18532 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000003v1_decoy (15240 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000004v1_decoy (13739 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000005v1_decoy (11297 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000006v1_decoy (10074 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000007v1_decoy (9891 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000008v1_decoy (9774 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000009v1_decoy (9727 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000010v1_decoy (9358 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000011v1_decoy (8920 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000012v1_decoy (8479 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000013v1_decoy (8312 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000014v1_decoy (8261 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000015v1_decoy (8131 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000016v1_decoy (8051 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000017v1_decoy (7832 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000018v1_decoy (7710 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000019v1_decoy (7702 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000020v1_decoy (7479 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000021v1_decoy (7368 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000022v1_decoy (7162 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000023v1_decoy (7065 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000024v1_decoy (7019 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000025v1_decoy (6997 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000026v1_decoy (6994 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000027v1_decoy (6979 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000028v1_decoy (6797 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000029v1_decoy (6525 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000030v1_decoy (6246 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000031v1_decoy (5926 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000032v1_decoy (5914 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000033v1_decoy (5898 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000034v1_decoy (5879 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000035v1_decoy (5834 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000036v1_decoy (5743 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000037v1_decoy (5577 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000038v1_decoy (5413 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000039v1_decoy (5250 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000040v1_decoy (5246 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000041v1_decoy (5118 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000042v1_decoy (5058 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000043v1_decoy (4959 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000044v1_decoy (4853 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000045v1_decoy (4828 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000046v1_decoy (4819 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000047v1_decoy (4809 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000048v1_decoy (4710 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000049v1_decoy (4680 bp)\n",
      "01:28:30.568 DEBUG GenomeLocParser -  chrUn_JTFH01000050v1_decoy (4645 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000051v1_decoy (4514 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000052v1_decoy (4439 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000053v1_decoy (4416 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000054v1_decoy (4409 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000055v1_decoy (4392 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000056v1_decoy (4359 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000057v1_decoy (4319 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000058v1_decoy (4290 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000059v1_decoy (4242 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000060v1_decoy (4228 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000061v1_decoy (4222 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000062v1_decoy (4216 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000063v1_decoy (4210 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000064v1_decoy (4206 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000065v1_decoy (4102 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000066v1_decoy (4101 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000067v1_decoy (4083 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000068v1_decoy (3967 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000069v1_decoy (3955 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000070v1_decoy (3945 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000071v1_decoy (3930 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000072v1_decoy (3929 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000073v1_decoy (3924 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000074v1_decoy (3919 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000075v1_decoy (3908 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000076v1_decoy (3892 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000077v1_decoy (3890 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000078v1_decoy (3859 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000079v1_decoy (3846 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000080v1_decoy (3835 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000081v1_decoy (3830 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000082v1_decoy (3828 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000083v1_decoy (3825 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000084v1_decoy (3821 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000085v1_decoy (3809 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000086v1_decoy (3801 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000087v1_decoy (3799 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000088v1_decoy (3737 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000089v1_decoy (3701 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000090v1_decoy (3698 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000091v1_decoy (3692 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000092v1_decoy (3686 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000093v1_decoy (3677 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000094v1_decoy (3664 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000095v1_decoy (3613 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000096v1_decoy (3611 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000097v1_decoy (3606 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000098v1_decoy (3584 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000099v1_decoy (3581 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000100v1_decoy (3543 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000101v1_decoy (3528 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000102v1_decoy (3527 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000103v1_decoy (3496 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000104v1_decoy (3493 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000105v1_decoy (3484 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000106v1_decoy (3435 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000107v1_decoy (3391 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000108v1_decoy (3374 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000109v1_decoy (3371 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000110v1_decoy (3361 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000111v1_decoy (3351 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000112v1_decoy (3340 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000113v1_decoy (3320 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000114v1_decoy (3282 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000115v1_decoy (3278 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000116v1_decoy (3260 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000117v1_decoy (3258 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000118v1_decoy (3253 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000119v1_decoy (3247 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000120v1_decoy (3230 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000121v1_decoy (3224 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000122v1_decoy (3216 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000123v1_decoy (3212 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000124v1_decoy (3194 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000125v1_decoy (3189 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000126v1_decoy (3177 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000127v1_decoy (3176 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000128v1_decoy (3173 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000129v1_decoy (3170 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000130v1_decoy (3166 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000131v1_decoy (3163 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000132v1_decoy (3143 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000133v1_decoy (3137 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000134v1_decoy (3116 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000135v1_decoy (3106 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000136v1_decoy (3093 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000137v1_decoy (3079 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000138v1_decoy (3053 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000139v1_decoy (3051 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000140v1_decoy (3015 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000141v1_decoy (3012 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000142v1_decoy (3009 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000143v1_decoy (2997 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000144v1_decoy (2997 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000145v1_decoy (2983 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000146v1_decoy (2979 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000147v1_decoy (2967 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000148v1_decoy (2967 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000149v1_decoy (2966 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000150v1_decoy (2954 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000151v1_decoy (2952 bp)\n",
      "01:28:30.569 DEBUG GenomeLocParser -  chrUn_JTFH01000152v1_decoy (2934 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000153v1_decoy (2918 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000154v1_decoy (2895 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000155v1_decoy (2887 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000156v1_decoy (2879 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000157v1_decoy (2878 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000158v1_decoy (2872 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000159v1_decoy (2868 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000160v1_decoy (2866 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000161v1_decoy (2865 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000162v1_decoy (2864 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000163v1_decoy (2859 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000164v1_decoy (2854 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000165v1_decoy (2830 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000166v1_decoy (2828 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000167v1_decoy (2824 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000168v1_decoy (2819 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000169v1_decoy (2813 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000170v1_decoy (2809 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000171v1_decoy (2802 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000172v1_decoy (2791 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000173v1_decoy (2783 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000174v1_decoy (2778 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000175v1_decoy (2777 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000176v1_decoy (2770 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000177v1_decoy (2769 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000178v1_decoy (2767 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000179v1_decoy (2763 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000180v1_decoy (2745 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000181v1_decoy (2742 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000182v1_decoy (2736 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000183v1_decoy (2729 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000184v1_decoy (2726 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000185v1_decoy (2719 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000186v1_decoy (2715 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000187v1_decoy (2708 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000188v1_decoy (2704 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000189v1_decoy (2692 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000190v1_decoy (2691 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000191v1_decoy (2690 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000192v1_decoy (2687 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000193v1_decoy (2677 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000194v1_decoy (2668 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000195v1_decoy (2668 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000196v1_decoy (2663 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000197v1_decoy (2655 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000198v1_decoy (2644 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000199v1_decoy (2642 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000200v1_decoy (2632 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000201v1_decoy (2632 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000202v1_decoy (2628 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000203v1_decoy (2623 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000204v1_decoy (2622 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000205v1_decoy (2619 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000206v1_decoy (2605 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000207v1_decoy (2603 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000208v1_decoy (2601 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000209v1_decoy (2598 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000210v1_decoy (2597 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000211v1_decoy (2596 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000212v1_decoy (2594 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000213v1_decoy (2586 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000214v1_decoy (2585 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000215v1_decoy (2583 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000216v1_decoy (2578 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000217v1_decoy (2569 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000218v1_decoy (2569 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000219v1_decoy (2551 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000220v1_decoy (2548 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000221v1_decoy (2548 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000222v1_decoy (2546 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000223v1_decoy (2545 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000224v1_decoy (2534 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000225v1_decoy (2533 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000226v1_decoy (2522 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000227v1_decoy (2522 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000228v1_decoy (2515 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000229v1_decoy (2513 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000230v1_decoy (2507 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000231v1_decoy (2504 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000232v1_decoy (2497 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000233v1_decoy (2471 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000234v1_decoy (2465 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000235v1_decoy (2464 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000236v1_decoy (2459 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000237v1_decoy (2457 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000238v1_decoy (2450 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000239v1_decoy (2435 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000240v1_decoy (2434 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000241v1_decoy (2432 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000242v1_decoy (2427 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000243v1_decoy (2421 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000244v1_decoy (2420 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000245v1_decoy (2414 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000246v1_decoy (2404 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000247v1_decoy (2403 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000248v1_decoy (2402 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000249v1_decoy (2397 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000250v1_decoy (2395 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000251v1_decoy (2394 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000252v1_decoy (2388 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000253v1_decoy (2382 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000254v1_decoy (2381 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000255v1_decoy (2380 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000256v1_decoy (2368 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000257v1_decoy (2364 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000258v1_decoy (2363 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000259v1_decoy (2348 bp)\n",
      "01:28:30.570 DEBUG GenomeLocParser -  chrUn_JTFH01000260v1_decoy (2339 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000261v1_decoy (2335 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000262v1_decoy (2332 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000263v1_decoy (2331 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000264v1_decoy (2330 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000265v1_decoy (2323 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000266v1_decoy (2319 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000267v1_decoy (2314 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000268v1_decoy (2308 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000269v1_decoy (2306 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000270v1_decoy (2296 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000271v1_decoy (2287 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000272v1_decoy (2279 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000273v1_decoy (2276 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000274v1_decoy (2273 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000275v1_decoy (2262 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000276v1_decoy (2254 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000277v1_decoy (2252 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000278v1_decoy (2245 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000279v1_decoy (2239 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000280v1_decoy (2223 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000281v1_decoy (2220 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000282v1_decoy (2218 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000283v1_decoy (2215 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000284v1_decoy (2213 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000285v1_decoy (2203 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000286v1_decoy (2200 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000287v1_decoy (2197 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000288v1_decoy (2194 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000289v1_decoy (2183 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000290v1_decoy (2179 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000291v1_decoy (2177 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000292v1_decoy (2177 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000293v1_decoy (2177 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000294v1_decoy (2168 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000295v1_decoy (2160 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000296v1_decoy (2155 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000297v1_decoy (2144 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000298v1_decoy (2143 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000299v1_decoy (2136 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000300v1_decoy (2134 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000301v1_decoy (2129 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000302v1_decoy (2128 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000303v1_decoy (2125 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000304v1_decoy (2125 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000305v1_decoy (2122 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000306v1_decoy (2111 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000307v1_decoy (2106 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000308v1_decoy (2094 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000309v1_decoy (2093 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000310v1_decoy (2088 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000311v1_decoy (2086 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000312v1_decoy (2086 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000313v1_decoy (2084 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000314v1_decoy (2080 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000315v1_decoy (2079 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000316v1_decoy (2076 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000317v1_decoy (2071 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000318v1_decoy (2066 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000319v1_decoy (2061 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000320v1_decoy (2055 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000321v1_decoy (2053 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000322v1_decoy (2040 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000323v1_decoy (2036 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000324v1_decoy (2035 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000325v1_decoy (2034 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000326v1_decoy (2032 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000327v1_decoy (2029 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000328v1_decoy (2025 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000329v1_decoy (2021 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000330v1_decoy (2018 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000331v1_decoy (2015 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000332v1_decoy (2009 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000333v1_decoy (2007 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000334v1_decoy (2005 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000335v1_decoy (2003 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000336v1_decoy (2001 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000337v1_decoy (2001 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000338v1_decoy (2000 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000339v1_decoy (1996 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000340v1_decoy (1992 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000341v1_decoy (1985 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000342v1_decoy (1981 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000343v1_decoy (1977 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000344v1_decoy (1971 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000345v1_decoy (1968 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000346v1_decoy (1962 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000347v1_decoy (1961 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000348v1_decoy (1960 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000349v1_decoy (1960 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000350v1_decoy (1954 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000351v1_decoy (1952 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000352v1_decoy (1947 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000353v1_decoy (1944 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000354v1_decoy (1943 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000355v1_decoy (1941 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000356v1_decoy (1937 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000357v1_decoy (1934 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000358v1_decoy (1929 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000359v1_decoy (1924 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000360v1_decoy (1924 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000361v1_decoy (1923 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000362v1_decoy (1921 bp)\n",
      "01:28:30.571 DEBUG GenomeLocParser -  chrUn_JTFH01000363v1_decoy (1918 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000364v1_decoy (1915 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000365v1_decoy (1915 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000366v1_decoy (1914 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000367v1_decoy (1912 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000368v1_decoy (1910 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000369v1_decoy (1907 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000370v1_decoy (1904 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000371v1_decoy (1897 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000372v1_decoy (1891 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000373v1_decoy (1890 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000374v1_decoy (1888 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000375v1_decoy (1888 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000376v1_decoy (1885 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000377v1_decoy (1881 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000378v1_decoy (1879 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000379v1_decoy (1877 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000380v1_decoy (1876 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000381v1_decoy (1876 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000382v1_decoy (1874 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000383v1_decoy (1872 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000384v1_decoy (1869 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000385v1_decoy (1866 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000386v1_decoy (1865 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000387v1_decoy (1865 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000388v1_decoy (1865 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000389v1_decoy (1862 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000390v1_decoy (1862 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000391v1_decoy (1859 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000392v1_decoy (1856 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000393v1_decoy (1856 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000394v1_decoy (1854 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000395v1_decoy (1850 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000396v1_decoy (1849 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000397v1_decoy (1849 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000398v1_decoy (1847 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000399v1_decoy (1839 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000400v1_decoy (1834 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000401v1_decoy (1821 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000402v1_decoy (1815 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000403v1_decoy (1811 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000404v1_decoy (1808 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000405v1_decoy (1808 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000406v1_decoy (1807 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000407v1_decoy (1807 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000408v1_decoy (1802 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000409v1_decoy (1801 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000410v1_decoy (1800 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000411v1_decoy (1795 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000412v1_decoy (1794 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000413v1_decoy (1792 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000414v1_decoy (1788 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000415v1_decoy (1786 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000416v1_decoy (1782 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000417v1_decoy (1782 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000418v1_decoy (1781 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000419v1_decoy (1781 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000420v1_decoy (1779 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000421v1_decoy (1777 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000422v1_decoy (1764 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000423v1_decoy (1762 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000424v1_decoy (1755 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000425v1_decoy (1749 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000426v1_decoy (1747 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000427v1_decoy (1746 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000428v1_decoy (1745 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000429v1_decoy (1744 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000430v1_decoy (1742 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000431v1_decoy (1740 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000432v1_decoy (1740 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000433v1_decoy (1736 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000434v1_decoy (1735 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000435v1_decoy (1732 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000436v1_decoy (1732 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000437v1_decoy (1730 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000438v1_decoy (1727 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000439v1_decoy (1722 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000440v1_decoy (1718 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000441v1_decoy (1716 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000442v1_decoy (1710 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000443v1_decoy (1708 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000444v1_decoy (1707 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000445v1_decoy (1706 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000446v1_decoy (1705 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000447v1_decoy (1704 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000448v1_decoy (1699 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000449v1_decoy (1698 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000450v1_decoy (1697 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000451v1_decoy (1697 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000452v1_decoy (1695 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000453v1_decoy (1695 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000454v1_decoy (1693 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000455v1_decoy (1687 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000456v1_decoy (1686 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000457v1_decoy (1680 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000458v1_decoy (1679 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000459v1_decoy (1679 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000460v1_decoy (1678 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000461v1_decoy (1674 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000462v1_decoy (1674 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000463v1_decoy (1671 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000464v1_decoy (1669 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000465v1_decoy (1665 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000466v1_decoy (1663 bp)\n",
      "01:28:30.572 DEBUG GenomeLocParser -  chrUn_JTFH01000467v1_decoy (1657 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000468v1_decoy (1653 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000469v1_decoy (1652 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000470v1_decoy (1650 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000471v1_decoy (1649 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000472v1_decoy (1649 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000473v1_decoy (1640 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000474v1_decoy (1638 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000475v1_decoy (1636 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000476v1_decoy (1632 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000477v1_decoy (1631 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000478v1_decoy (1630 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000479v1_decoy (1627 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000480v1_decoy (1624 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000481v1_decoy (1617 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000482v1_decoy (1616 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000483v1_decoy (1615 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000484v1_decoy (1611 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000485v1_decoy (1611 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000486v1_decoy (1606 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000487v1_decoy (1605 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000488v1_decoy (1605 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000489v1_decoy (1600 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000490v1_decoy (1598 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000491v1_decoy (1598 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000492v1_decoy (1597 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000493v1_decoy (1596 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000494v1_decoy (1595 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000495v1_decoy (1592 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000496v1_decoy (1589 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000497v1_decoy (1585 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000498v1_decoy (1579 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000499v1_decoy (1578 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000500v1_decoy (1577 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000501v1_decoy (1577 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000502v1_decoy (1577 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000503v1_decoy (1576 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000504v1_decoy (1575 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000505v1_decoy (1574 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000506v1_decoy (1572 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000507v1_decoy (1571 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000508v1_decoy (1563 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000509v1_decoy (1561 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000510v1_decoy (1561 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000511v1_decoy (1560 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000512v1_decoy (1560 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000513v1_decoy (1554 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000514v1_decoy (1552 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000515v1_decoy (1548 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000516v1_decoy (1546 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000517v1_decoy (1541 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000518v1_decoy (1536 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000519v1_decoy (1533 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000520v1_decoy (1532 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000521v1_decoy (1532 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000522v1_decoy (1530 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000523v1_decoy (1527 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000524v1_decoy (1526 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000525v1_decoy (1524 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000526v1_decoy (1523 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000527v1_decoy (1523 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000528v1_decoy (1522 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000529v1_decoy (1522 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000530v1_decoy (1519 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000531v1_decoy (1513 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000532v1_decoy (1508 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000533v1_decoy (1508 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000534v1_decoy (1505 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000535v1_decoy (1503 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000536v1_decoy (1496 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000537v1_decoy (1491 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000538v1_decoy (1490 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000539v1_decoy (1490 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000540v1_decoy (1487 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000541v1_decoy (1486 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000542v1_decoy (1485 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000543v1_decoy (1484 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000544v1_decoy (1483 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000545v1_decoy (1479 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000546v1_decoy (1479 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000547v1_decoy (1476 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000548v1_decoy (1475 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000549v1_decoy (1472 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000550v1_decoy (1469 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000551v1_decoy (1468 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000552v1_decoy (1467 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000553v1_decoy (1465 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000554v1_decoy (1464 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000555v1_decoy (1463 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000556v1_decoy (1463 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000557v1_decoy (1459 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000558v1_decoy (1459 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000559v1_decoy (1458 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000560v1_decoy (1458 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000561v1_decoy (1454 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000562v1_decoy (1449 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000563v1_decoy (1449 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000564v1_decoy (1448 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000565v1_decoy (1446 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000566v1_decoy (1442 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000567v1_decoy (1441 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000568v1_decoy (1440 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000569v1_decoy (1439 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000570v1_decoy (1437 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000571v1_decoy (1436 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000572v1_decoy (1429 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000573v1_decoy (1429 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000574v1_decoy (1427 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000575v1_decoy (1426 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000576v1_decoy (1425 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000577v1_decoy (1424 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000578v1_decoy (1424 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000579v1_decoy (1423 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000580v1_decoy (1423 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000581v1_decoy (1423 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000582v1_decoy (1414 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000583v1_decoy (1414 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000584v1_decoy (1413 bp)\n",
      "01:28:30.573 DEBUG GenomeLocParser -  chrUn_JTFH01000585v1_decoy (1413 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000586v1_decoy (1410 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000587v1_decoy (1409 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000588v1_decoy (1409 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000589v1_decoy (1406 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000590v1_decoy (1405 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000591v1_decoy (1405 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000592v1_decoy (1404 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000593v1_decoy (1404 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000594v1_decoy (1402 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000595v1_decoy (1402 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000596v1_decoy (1402 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000597v1_decoy (1402 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000598v1_decoy (1400 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000599v1_decoy (1398 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000600v1_decoy (1396 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000601v1_decoy (1395 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000602v1_decoy (1394 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000603v1_decoy (1393 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000604v1_decoy (1391 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000605v1_decoy (1389 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000606v1_decoy (1389 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000607v1_decoy (1388 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000608v1_decoy (1387 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000609v1_decoy (1384 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000610v1_decoy (1381 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000611v1_decoy (1381 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000612v1_decoy (1379 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000613v1_decoy (1377 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000614v1_decoy (1376 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000615v1_decoy (1376 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000616v1_decoy (1375 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000617v1_decoy (1374 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000618v1_decoy (1372 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000619v1_decoy (1371 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000620v1_decoy (1370 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000621v1_decoy (1370 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000622v1_decoy (1366 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000623v1_decoy (1363 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000624v1_decoy (1360 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000625v1_decoy (1356 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000626v1_decoy (1355 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000627v1_decoy (1355 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000628v1_decoy (1352 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000629v1_decoy (1345 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000630v1_decoy (1344 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000631v1_decoy (1344 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000632v1_decoy (1342 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000633v1_decoy (1342 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000634v1_decoy (1336 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000635v1_decoy (1334 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000636v1_decoy (1334 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000637v1_decoy (1333 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000638v1_decoy (1332 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000639v1_decoy (1328 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000640v1_decoy (1328 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000641v1_decoy (1328 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000642v1_decoy (1327 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000643v1_decoy (1325 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000644v1_decoy (1322 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000645v1_decoy (1320 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000646v1_decoy (1319 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000647v1_decoy (1318 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000648v1_decoy (1315 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000649v1_decoy (1314 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000650v1_decoy (1313 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000651v1_decoy (1313 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000652v1_decoy (1312 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000653v1_decoy (1310 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000654v1_decoy (1309 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000655v1_decoy (1309 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000656v1_decoy (1307 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000657v1_decoy (1307 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000658v1_decoy (1305 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000659v1_decoy (1304 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000660v1_decoy (1303 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000661v1_decoy (1302 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000662v1_decoy (1302 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000663v1_decoy (1301 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000664v1_decoy (1301 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000665v1_decoy (1300 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000666v1_decoy (1299 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000667v1_decoy (1297 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000668v1_decoy (1295 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000669v1_decoy (1294 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000670v1_decoy (1293 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000671v1_decoy (1291 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000672v1_decoy (1291 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000673v1_decoy (1289 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000674v1_decoy (1288 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000675v1_decoy (1288 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000676v1_decoy (1287 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000677v1_decoy (1287 bp)\n",
      "01:28:30.574 DEBUG GenomeLocParser -  chrUn_JTFH01000678v1_decoy (1287 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000679v1_decoy (1286 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000680v1_decoy (1283 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000681v1_decoy (1281 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000682v1_decoy (1277 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000683v1_decoy (1274 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000684v1_decoy (1270 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000685v1_decoy (1267 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000686v1_decoy (1266 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000687v1_decoy (1260 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000688v1_decoy (1259 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000689v1_decoy (1258 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000690v1_decoy (1258 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000691v1_decoy (1258 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000692v1_decoy (1256 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000693v1_decoy (1255 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000694v1_decoy (1254 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000695v1_decoy (1254 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000696v1_decoy (1253 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000697v1_decoy (1250 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000698v1_decoy (1249 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000699v1_decoy (1248 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000700v1_decoy (1248 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000701v1_decoy (1247 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000702v1_decoy (1242 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000703v1_decoy (1242 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000704v1_decoy (1241 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000705v1_decoy (1241 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000706v1_decoy (1241 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000707v1_decoy (1239 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000708v1_decoy (1238 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000709v1_decoy (1237 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000710v1_decoy (1236 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000711v1_decoy (1235 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000712v1_decoy (1234 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000713v1_decoy (1234 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000714v1_decoy (1234 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000715v1_decoy (1233 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000716v1_decoy (1232 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000717v1_decoy (1232 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000718v1_decoy (1231 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000719v1_decoy (1230 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000720v1_decoy (1228 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000721v1_decoy (1227 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000722v1_decoy (1227 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000723v1_decoy (1226 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000724v1_decoy (1224 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000725v1_decoy (1224 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000726v1_decoy (1220 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000727v1_decoy (1220 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000728v1_decoy (1219 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000729v1_decoy (1217 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000730v1_decoy (1216 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000731v1_decoy (1215 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000732v1_decoy (1214 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000733v1_decoy (1214 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000734v1_decoy (1214 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000735v1_decoy (1213 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000736v1_decoy (1212 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000737v1_decoy (1209 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000738v1_decoy (1208 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000739v1_decoy (1207 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000740v1_decoy (1207 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000741v1_decoy (1207 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000742v1_decoy (1206 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000743v1_decoy (1206 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000744v1_decoy (1205 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000745v1_decoy (1205 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000746v1_decoy (1204 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000747v1_decoy (1204 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000748v1_decoy (1204 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000749v1_decoy (1203 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000750v1_decoy (1201 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000751v1_decoy (1201 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000752v1_decoy (1200 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000753v1_decoy (1200 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000754v1_decoy (1199 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000755v1_decoy (1198 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000756v1_decoy (1197 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000757v1_decoy (1196 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000758v1_decoy (1195 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000759v1_decoy (1194 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000760v1_decoy (1194 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000761v1_decoy (1191 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000762v1_decoy (1189 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000763v1_decoy (1186 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000764v1_decoy (1186 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000765v1_decoy (1184 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000766v1_decoy (1183 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000767v1_decoy (1183 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000768v1_decoy (1182 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000769v1_decoy (1181 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000770v1_decoy (1181 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000771v1_decoy (1181 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000772v1_decoy (1181 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000773v1_decoy (1179 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000774v1_decoy (1178 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000775v1_decoy (1178 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000776v1_decoy (1177 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000777v1_decoy (1177 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000778v1_decoy (1171 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000779v1_decoy (1171 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000780v1_decoy (1171 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000781v1_decoy (1170 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000782v1_decoy (1170 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000783v1_decoy (1167 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000784v1_decoy (1167 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000785v1_decoy (1167 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000786v1_decoy (1165 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000787v1_decoy (1165 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000788v1_decoy (1162 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000789v1_decoy (1157 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000790v1_decoy (1156 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000791v1_decoy (1156 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000792v1_decoy (1154 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000793v1_decoy (1154 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000794v1_decoy (1151 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000795v1_decoy (1151 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000796v1_decoy (1150 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000797v1_decoy (1150 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000798v1_decoy (1147 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000799v1_decoy (1147 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000800v1_decoy (1146 bp)\n",
      "01:28:30.575 DEBUG GenomeLocParser -  chrUn_JTFH01000801v1_decoy (1144 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000802v1_decoy (1144 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000803v1_decoy (1143 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000804v1_decoy (1142 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000805v1_decoy (1141 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000806v1_decoy (1141 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000807v1_decoy (1140 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000808v1_decoy (1138 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000809v1_decoy (1134 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000810v1_decoy (1134 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000811v1_decoy (1132 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000812v1_decoy (1131 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000813v1_decoy (1131 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000814v1_decoy (1130 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000815v1_decoy (1127 bp)\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000816v1_decoy (1126 bp)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000817v1_decoy (1124 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000818v1_decoy (1122 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000819v1_decoy (1122 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000820v1_decoy (1121 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000821v1_decoy (1119 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000822v1_decoy (1119 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000823v1_decoy (1119 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000824v1_decoy (1119 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000825v1_decoy (1118 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000826v1_decoy (1116 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000827v1_decoy (1116 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000828v1_decoy (1115 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000829v1_decoy (1115 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000830v1_decoy (1115 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000831v1_decoy (1114 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000832v1_decoy (1113 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000833v1_decoy (1113 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000834v1_decoy (1110 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000835v1_decoy (1110 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000836v1_decoy (1109 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000837v1_decoy (1108 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000838v1_decoy (1107 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000839v1_decoy (1107 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000840v1_decoy (1107 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000841v1_decoy (1107 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000842v1_decoy (1106 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000843v1_decoy (1103 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000844v1_decoy (1103 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000845v1_decoy (1103 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000846v1_decoy (1100 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000847v1_decoy (1099 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000848v1_decoy (1098 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000849v1_decoy (1097 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000850v1_decoy (1096 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000851v1_decoy (1096 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000852v1_decoy (1094 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000853v1_decoy (1093 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000854v1_decoy (1090 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000855v1_decoy (1088 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000856v1_decoy (1087 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000857v1_decoy (1086 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000858v1_decoy (1085 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000859v1_decoy (1084 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000860v1_decoy (1084 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000861v1_decoy (1084 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000862v1_decoy (1084 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000863v1_decoy (1083 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000864v1_decoy (1083 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000865v1_decoy (1082 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000866v1_decoy (1082 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000867v1_decoy (1081 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000868v1_decoy (1081 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000869v1_decoy (1079 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000870v1_decoy (1076 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000871v1_decoy (1074 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000872v1_decoy (1073 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000873v1_decoy (1073 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000874v1_decoy (1071 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000875v1_decoy (1069 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000876v1_decoy (1067 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000877v1_decoy (1067 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000878v1_decoy (1067 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000879v1_decoy (1066 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000880v1_decoy (1065 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000881v1_decoy (1065 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000882v1_decoy (1065 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000883v1_decoy (1065 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000884v1_decoy (1065 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000885v1_decoy (1064 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000886v1_decoy (1064 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000887v1_decoy (1064 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000888v1_decoy (1063 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000889v1_decoy (1062 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000890v1_decoy (1062 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000891v1_decoy (1062 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000892v1_decoy (1061 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000893v1_decoy (1060 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000894v1_decoy (1057 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000895v1_decoy (1057 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000896v1_decoy (1056 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000897v1_decoy (1055 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000898v1_decoy (1055 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000899v1_decoy (1055 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000900v1_decoy (1055 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000901v1_decoy (1054 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000902v1_decoy (1051 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000903v1_decoy (1050 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000904v1_decoy (1050 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000905v1_decoy (1049 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000906v1_decoy (1048 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000907v1_decoy (1047 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000908v1_decoy (1046 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000909v1_decoy (1046 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000910v1_decoy (1046 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000911v1_decoy (1045 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000912v1_decoy (1045 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000913v1_decoy (1045 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000914v1_decoy (1044 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000915v1_decoy (1042 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000916v1_decoy (1041 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000917v1_decoy (1039 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000918v1_decoy (1039 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000919v1_decoy (1038 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000920v1_decoy (1036 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000921v1_decoy (1036 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000922v1_decoy (1035 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000923v1_decoy (1035 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000924v1_decoy (1033 bp)\r\n",
      "01:28:30.576 DEBUG GenomeLocParser -  chrUn_JTFH01000925v1_decoy (1032 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000926v1_decoy (1031 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000927v1_decoy (1031 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000928v1_decoy (1031 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000929v1_decoy (1027 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000930v1_decoy (1027 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000931v1_decoy (1026 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000932v1_decoy (1026 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000933v1_decoy (1024 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000934v1_decoy (1024 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000935v1_decoy (1022 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000936v1_decoy (1022 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000937v1_decoy (1021 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000938v1_decoy (1020 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000939v1_decoy (1019 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000940v1_decoy (1018 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000941v1_decoy (1018 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000942v1_decoy (1018 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000943v1_decoy (1016 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000944v1_decoy (1010 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000945v1_decoy (1010 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000946v1_decoy (1009 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000947v1_decoy (1008 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000948v1_decoy (1007 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000949v1_decoy (1006 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000950v1_decoy (1005 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000951v1_decoy (1005 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000952v1_decoy (1004 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000953v1_decoy (1004 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000954v1_decoy (1003 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000955v1_decoy (1003 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000956v1_decoy (1003 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000957v1_decoy (1003 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000958v1_decoy (1002 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000959v1_decoy (1002 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000960v1_decoy (1000 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000961v1_decoy (1000 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000962v1_decoy (8358 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000963v1_decoy (7932 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000964v1_decoy (6846 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000965v1_decoy (4591 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000966v1_decoy (4041 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000967v1_decoy (3841 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000968v1_decoy (3754 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000969v1_decoy (3743 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000970v1_decoy (3702 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000971v1_decoy (3625 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000972v1_decoy (3529 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000973v1_decoy (3508 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000974v1_decoy (3359 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000975v1_decoy (3320 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000976v1_decoy (3231 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000977v1_decoy (3220 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000978v1_decoy (3212 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000979v1_decoy (3192 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000980v1_decoy (3092 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000981v1_decoy (3087 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000982v1_decoy (3048 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000983v1_decoy (3005 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000984v1_decoy (3004 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000985v1_decoy (2959 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000986v1_decoy (2934 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000987v1_decoy (2933 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000988v1_decoy (2827 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000989v1_decoy (2794 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000990v1_decoy (2749 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000991v1_decoy (2745 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000992v1_decoy (2733 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000993v1_decoy (2698 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000994v1_decoy (2665 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000995v1_decoy (2634 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000996v1_decoy (2492 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000997v1_decoy (2489 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000998v1_decoy (2468 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01000999v1_decoy (2414 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001000v1_decoy (2395 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001001v1_decoy (2356 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001002v1_decoy (2339 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001003v1_decoy (2310 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001004v1_decoy (2288 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001005v1_decoy (2285 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001006v1_decoy (2269 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001007v1_decoy (2253 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001008v1_decoy (2203 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001009v1_decoy (2176 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001010v1_decoy (2159 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001011v1_decoy (2155 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001012v1_decoy (2149 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001013v1_decoy (2129 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001014v1_decoy (2116 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001015v1_decoy (2113 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001016v1_decoy (2098 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001017v1_decoy (2066 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001018v1_decoy (2066 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001019v1_decoy (2059 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001020v1_decoy (2047 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001021v1_decoy (2040 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001022v1_decoy (2030 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001023v1_decoy (2024 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001024v1_decoy (2001 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001025v1_decoy (1992 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001026v1_decoy (1981 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001027v1_decoy (1979 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001028v1_decoy (1957 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001029v1_decoy (1953 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001030v1_decoy (1944 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001031v1_decoy (1936 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001032v1_decoy (1932 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001033v1_decoy (1882 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001034v1_decoy (1878 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001035v1_decoy (1870 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001036v1_decoy (1821 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001037v1_decoy (1813 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001038v1_decoy (1809 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001039v1_decoy (1804 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001040v1_decoy (1797 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001041v1_decoy (1791 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001042v1_decoy (1781 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001043v1_decoy (1766 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001044v1_decoy (1764 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001045v1_decoy (1743 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001046v1_decoy (1741 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001047v1_decoy (1709 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001048v1_decoy (1706 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001049v1_decoy (1701 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001050v1_decoy (1689 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001051v1_decoy (1646 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001052v1_decoy (1641 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001053v1_decoy (1639 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001054v1_decoy (1636 bp)\r\n",
      "01:28:30.577 DEBUG GenomeLocParser -  chrUn_JTFH01001055v1_decoy (1632 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001056v1_decoy (1629 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001057v1_decoy (1623 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001058v1_decoy (1622 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001059v1_decoy (1622 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001060v1_decoy (1619 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001061v1_decoy (1606 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001062v1_decoy (1593 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001063v1_decoy (1592 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001064v1_decoy (1558 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001065v1_decoy (1545 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001066v1_decoy (1542 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001067v1_decoy (1540 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001068v1_decoy (1529 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001069v1_decoy (1518 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001070v1_decoy (1515 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001071v1_decoy (1513 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001072v1_decoy (1507 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001073v1_decoy (1504 bp)\r\n",
      "01:28:30.580 DEBUG GenomeLocParser -  chrUn_JTFH01001074v1_decoy (1499 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001075v1_decoy (1495 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001076v1_decoy (1495 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001077v1_decoy (1492 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001078v1_decoy (1492 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001079v1_decoy (1489 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001080v1_decoy (1485 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001081v1_decoy (1483 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001082v1_decoy (1473 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001083v1_decoy (1470 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001084v1_decoy (1463 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001085v1_decoy (1460 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001086v1_decoy (1458 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001087v1_decoy (1456 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001088v1_decoy (1453 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001089v1_decoy (1443 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001090v1_decoy (1441 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001091v1_decoy (1426 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001092v1_decoy (1425 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001093v1_decoy (1418 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001094v1_decoy (1413 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001095v1_decoy (1413 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001096v1_decoy (1412 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001097v1_decoy (1407 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001098v1_decoy (1406 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001099v1_decoy (1396 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001100v1_decoy (1390 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001101v1_decoy (1382 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001102v1_decoy (1376 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001103v1_decoy (1375 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001104v1_decoy (1371 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001105v1_decoy (1367 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001106v1_decoy (1364 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001107v1_decoy (1356 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001108v1_decoy (1355 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001109v1_decoy (1352 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001110v1_decoy (1350 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001111v1_decoy (1346 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001112v1_decoy (1345 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001113v1_decoy (1340 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001114v1_decoy (1330 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001115v1_decoy (1329 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001116v1_decoy (1324 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001117v1_decoy (1316 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001118v1_decoy (1307 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001119v1_decoy (1304 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001120v1_decoy (1304 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001121v1_decoy (1303 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001122v1_decoy (1301 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001123v1_decoy (1300 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001124v1_decoy (1297 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001125v1_decoy (1296 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001126v1_decoy (1290 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001127v1_decoy (1284 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001128v1_decoy (1282 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001129v1_decoy (1281 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001130v1_decoy (1280 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001131v1_decoy (1279 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001132v1_decoy (1272 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001133v1_decoy (1267 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001134v1_decoy (1267 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001135v1_decoy (1266 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001136v1_decoy (1264 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001137v1_decoy (1264 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001138v1_decoy (1264 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001139v1_decoy (1263 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001140v1_decoy (1249 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001141v1_decoy (1240 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001142v1_decoy (1239 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001143v1_decoy (1235 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001144v1_decoy (1235 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001145v1_decoy (1233 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001146v1_decoy (1232 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001147v1_decoy (1230 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001148v1_decoy (1226 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001149v1_decoy (1223 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001150v1_decoy (1214 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001151v1_decoy (1213 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001152v1_decoy (1211 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001153v1_decoy (1209 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001154v1_decoy (1202 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001155v1_decoy (1199 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001156v1_decoy (1197 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001157v1_decoy (1193 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001158v1_decoy (1191 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001159v1_decoy (1187 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001160v1_decoy (1186 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001161v1_decoy (1184 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001162v1_decoy (1184 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001163v1_decoy (1182 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001164v1_decoy (1179 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001165v1_decoy (1173 bp)\r\n",
      "01:28:30.581 DEBUG GenomeLocParser -  chrUn_JTFH01001166v1_decoy (1169 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001167v1_decoy (1167 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001168v1_decoy (1166 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001169v1_decoy (1165 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001170v1_decoy (1164 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001171v1_decoy (1163 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001172v1_decoy (1158 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001173v1_decoy (1158 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001174v1_decoy (1157 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001175v1_decoy (1157 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001176v1_decoy (1157 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001177v1_decoy (1155 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001178v1_decoy (1154 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001179v1_decoy (1149 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001180v1_decoy (1148 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001181v1_decoy (1148 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001182v1_decoy (1146 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001183v1_decoy (1144 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001184v1_decoy (1140 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001185v1_decoy (1136 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001186v1_decoy (1134 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001187v1_decoy (1133 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001188v1_decoy (1129 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001189v1_decoy (1127 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001190v1_decoy (1127 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001191v1_decoy (1118 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001192v1_decoy (1110 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001193v1_decoy (1104 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001194v1_decoy (1104 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001195v1_decoy (1101 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001196v1_decoy (1098 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001197v1_decoy (1096 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001198v1_decoy (1094 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001199v1_decoy (1091 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001200v1_decoy (1089 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001201v1_decoy (1086 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001202v1_decoy (1085 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001203v1_decoy (1084 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001204v1_decoy (1083 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001205v1_decoy (1083 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001206v1_decoy (1079 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001207v1_decoy (1076 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001208v1_decoy (1069 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001209v1_decoy (1068 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001210v1_decoy (1067 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001211v1_decoy (1067 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001212v1_decoy (1067 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001213v1_decoy (1063 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001214v1_decoy (1062 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001215v1_decoy (1059 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001216v1_decoy (1058 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001217v1_decoy (1058 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001218v1_decoy (1055 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001219v1_decoy (1054 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001220v1_decoy (1054 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001221v1_decoy (1053 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001222v1_decoy (1053 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001223v1_decoy (1052 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001224v1_decoy (1051 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001225v1_decoy (1049 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001226v1_decoy (1047 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001227v1_decoy (1044 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001228v1_decoy (1043 bp)\r\n",
      "01:28:30.582 DEBUG GenomeLocParser -  chrUn_JTFH01001229v1_decoy (1043 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001230v1_decoy (1042 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001231v1_decoy (1042 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001232v1_decoy (1041 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001233v1_decoy (1040 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001234v1_decoy (1039 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001235v1_decoy (1038 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001236v1_decoy (1037 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001237v1_decoy (1037 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001238v1_decoy (1035 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001239v1_decoy (1027 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001240v1_decoy (1021 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001241v1_decoy (1021 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001242v1_decoy (1019 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001243v1_decoy (1019 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001244v1_decoy (1016 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001245v1_decoy (1014 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001246v1_decoy (1013 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001247v1_decoy (1009 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001248v1_decoy (1008 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001249v1_decoy (1007 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001250v1_decoy (1004 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001251v1_decoy (1004 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001252v1_decoy (1003 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001253v1_decoy (1001 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001254v1_decoy (1000 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001255v1_decoy (1000 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001256v1_decoy (1000 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001257v1_decoy (17929 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001258v1_decoy (9749 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001259v1_decoy (8053 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001260v1_decoy (7826 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001261v1_decoy (7768 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001262v1_decoy (5691 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001263v1_decoy (5444 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001264v1_decoy (5077 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001265v1_decoy (4990 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001266v1_decoy (4545 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001267v1_decoy (4544 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001268v1_decoy (4202 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001269v1_decoy (4195 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001270v1_decoy (3807 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001271v1_decoy (3741 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001272v1_decoy (3699 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001273v1_decoy (3640 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001274v1_decoy (3531 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001275v1_decoy (3455 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001276v1_decoy (3411 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001277v1_decoy (3387 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001278v1_decoy (3358 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001279v1_decoy (3285 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001280v1_decoy (3273 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001281v1_decoy (3262 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001282v1_decoy (3259 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001283v1_decoy (3222 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001284v1_decoy (3127 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001285v1_decoy (3110 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001286v1_decoy (3104 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001287v1_decoy (3071 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001288v1_decoy (3063 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001289v1_decoy (3059 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001290v1_decoy (2990 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001291v1_decoy (2986 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001292v1_decoy (2928 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001293v1_decoy (2922 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001294v1_decoy (2875 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001295v1_decoy (2859 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001296v1_decoy (2850 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001297v1_decoy (2813 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001298v1_decoy (2785 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001299v1_decoy (2736 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001300v1_decoy (2688 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001301v1_decoy (2658 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001302v1_decoy (2643 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001303v1_decoy (2618 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001304v1_decoy (2605 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001305v1_decoy (2583 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001306v1_decoy (2534 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001307v1_decoy (2512 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001308v1_decoy (2500 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001309v1_decoy (2481 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001310v1_decoy (2478 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001311v1_decoy (2473 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001312v1_decoy (2467 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001313v1_decoy (2442 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001314v1_decoy (2430 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001315v1_decoy (2417 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001316v1_decoy (2408 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001317v1_decoy (2395 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001318v1_decoy (2352 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001319v1_decoy (2337 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001320v1_decoy (2322 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001321v1_decoy (2307 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001322v1_decoy (2306 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001323v1_decoy (2292 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001324v1_decoy (2271 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001325v1_decoy (2265 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001326v1_decoy (2260 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001327v1_decoy (2240 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001328v1_decoy (2238 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001329v1_decoy (2228 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001330v1_decoy (2215 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001331v1_decoy (2205 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001332v1_decoy (2191 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001333v1_decoy (2191 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001334v1_decoy (2190 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001335v1_decoy (2184 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001336v1_decoy (2166 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001337v1_decoy (2165 bp)\r\n",
      "01:28:30.583 DEBUG GenomeLocParser -  chrUn_JTFH01001338v1_decoy (2162 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001339v1_decoy (2146 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001340v1_decoy (2116 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001341v1_decoy (2112 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001342v1_decoy (2108 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001343v1_decoy (2106 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001344v1_decoy (2106 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001345v1_decoy (2106 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001346v1_decoy (2097 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001347v1_decoy (2081 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001348v1_decoy (2058 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001349v1_decoy (2055 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001350v1_decoy (2054 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001351v1_decoy (2037 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001352v1_decoy (2032 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001353v1_decoy (2032 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001354v1_decoy (2020 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001355v1_decoy (2018 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001356v1_decoy (2014 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001357v1_decoy (2001 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001358v1_decoy (2001 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001359v1_decoy (1991 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001360v1_decoy (1990 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001361v1_decoy (1983 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001362v1_decoy (1981 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001363v1_decoy (1981 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001364v1_decoy (1979 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001365v1_decoy (1963 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001366v1_decoy (1932 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001367v1_decoy (1929 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001368v1_decoy (1881 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001369v1_decoy (1874 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001370v1_decoy (1849 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001371v1_decoy (1849 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001372v1_decoy (1833 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001373v1_decoy (1832 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001374v1_decoy (1826 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001375v1_decoy (1814 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001376v1_decoy (1814 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001377v1_decoy (1791 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001378v1_decoy (1789 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001379v1_decoy (1786 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001380v1_decoy (1778 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001381v1_decoy (1776 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001382v1_decoy (1762 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001383v1_decoy (1758 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001384v1_decoy (1757 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001385v1_decoy (1754 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001386v1_decoy (1752 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001387v1_decoy (1751 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001388v1_decoy (1749 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001389v1_decoy (1738 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001390v1_decoy (1729 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001391v1_decoy (1726 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001392v1_decoy (1716 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001393v1_decoy (1712 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001394v1_decoy (1711 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001395v1_decoy (1703 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001396v1_decoy (1702 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001397v1_decoy (1699 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001398v1_decoy (1686 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001399v1_decoy (1684 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001400v1_decoy (1680 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001401v1_decoy (1678 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001402v1_decoy (1678 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001403v1_decoy (1677 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001404v1_decoy (1676 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001405v1_decoy (1672 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001406v1_decoy (1669 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001407v1_decoy (1668 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001408v1_decoy (1663 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001409v1_decoy (1660 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001410v1_decoy (1660 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001411v1_decoy (1658 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001412v1_decoy (1656 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001413v1_decoy (1656 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001414v1_decoy (1652 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001415v1_decoy (1647 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001416v1_decoy (1645 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001417v1_decoy (1641 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001418v1_decoy (1638 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001419v1_decoy (1633 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001420v1_decoy (1626 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001421v1_decoy (1614 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001422v1_decoy (1612 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001423v1_decoy (1605 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001424v1_decoy (1603 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001425v1_decoy (1599 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001426v1_decoy (1589 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001427v1_decoy (1588 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001428v1_decoy (1585 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001429v1_decoy (1584 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001430v1_decoy (1584 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001431v1_decoy (1580 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001432v1_decoy (1572 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001433v1_decoy (1570 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001434v1_decoy (1569 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001435v1_decoy (1568 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001436v1_decoy (1567 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001437v1_decoy (1565 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001438v1_decoy (1559 bp)\r\n",
      "01:28:30.584 DEBUG GenomeLocParser -  chrUn_JTFH01001439v1_decoy (1559 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001440v1_decoy (1556 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001441v1_decoy (1554 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001442v1_decoy (1549 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001443v1_decoy (1542 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001444v1_decoy (1541 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001445v1_decoy (1538 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001446v1_decoy (1537 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001447v1_decoy (1535 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001448v1_decoy (1530 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001449v1_decoy (1528 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001450v1_decoy (1522 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001451v1_decoy (1514 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001452v1_decoy (1509 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001453v1_decoy (1507 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001454v1_decoy (1500 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001455v1_decoy (1499 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001456v1_decoy (1499 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001457v1_decoy (1497 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001458v1_decoy (1496 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001459v1_decoy (1488 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001460v1_decoy (1486 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001461v1_decoy (1485 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001462v1_decoy (1481 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001463v1_decoy (1479 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001464v1_decoy (1472 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001465v1_decoy (1472 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001466v1_decoy (1470 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001467v1_decoy (1466 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001468v1_decoy (1465 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001469v1_decoy (1461 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001470v1_decoy (1458 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001471v1_decoy (1457 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001472v1_decoy (1448 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001473v1_decoy (1447 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001474v1_decoy (1444 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001475v1_decoy (1443 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001476v1_decoy (1443 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001477v1_decoy (1438 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001478v1_decoy (1432 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001479v1_decoy (1430 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001480v1_decoy (1430 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001481v1_decoy (1429 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001482v1_decoy (1429 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001483v1_decoy (1429 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001484v1_decoy (1426 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001485v1_decoy (1426 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001486v1_decoy (1420 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001487v1_decoy (1416 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001488v1_decoy (1416 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001489v1_decoy (1415 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001490v1_decoy (1415 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001491v1_decoy (1414 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001492v1_decoy (1413 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001493v1_decoy (1410 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001494v1_decoy (1405 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001495v1_decoy (1402 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001496v1_decoy (1398 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001497v1_decoy (1397 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001498v1_decoy (1395 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001499v1_decoy (1392 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001500v1_decoy (1388 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001501v1_decoy (1386 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001502v1_decoy (1382 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001503v1_decoy (1381 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001504v1_decoy (1379 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001505v1_decoy (1376 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001506v1_decoy (1374 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001507v1_decoy (1374 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001508v1_decoy (1373 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001509v1_decoy (1373 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001510v1_decoy (1372 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001511v1_decoy (1370 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001512v1_decoy (1367 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001513v1_decoy (1365 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001514v1_decoy (1364 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001515v1_decoy (1361 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001516v1_decoy (1361 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001517v1_decoy (1355 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001518v1_decoy (1355 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001519v1_decoy (1354 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001520v1_decoy (1353 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001521v1_decoy (1349 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001522v1_decoy (1345 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001523v1_decoy (1344 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001524v1_decoy (1343 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001525v1_decoy (1338 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001526v1_decoy (1338 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001527v1_decoy (1338 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001528v1_decoy (1336 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001529v1_decoy (1333 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001530v1_decoy (1333 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001531v1_decoy (1332 bp)\r\n",
      "01:28:30.585 DEBUG GenomeLocParser -  chrUn_JTFH01001532v1_decoy (1324 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001533v1_decoy (1323 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001534v1_decoy (1323 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001535v1_decoy (1320 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001536v1_decoy (1320 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001537v1_decoy (1317 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001538v1_decoy (1316 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001539v1_decoy (1304 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001540v1_decoy (1304 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001541v1_decoy (1303 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001542v1_decoy (1302 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001543v1_decoy (1301 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001544v1_decoy (1300 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001545v1_decoy (1298 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001546v1_decoy (1297 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001547v1_decoy (1295 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001548v1_decoy (1284 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001549v1_decoy (1283 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001550v1_decoy (1283 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001551v1_decoy (1279 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001552v1_decoy (1278 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001553v1_decoy (1271 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001554v1_decoy (1271 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001555v1_decoy (1268 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001556v1_decoy (1264 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001557v1_decoy (1263 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001558v1_decoy (1262 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001559v1_decoy (1261 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001560v1_decoy (1260 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001561v1_decoy (1259 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001562v1_decoy (1259 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001563v1_decoy (1258 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001564v1_decoy (1256 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001565v1_decoy (1253 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001566v1_decoy (1248 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001567v1_decoy (1248 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001568v1_decoy (1246 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001569v1_decoy (1246 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001570v1_decoy (1244 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001571v1_decoy (1238 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001572v1_decoy (1238 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001573v1_decoy (1236 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001574v1_decoy (1234 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001575v1_decoy (1234 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001576v1_decoy (1231 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001577v1_decoy (1231 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001578v1_decoy (1230 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001579v1_decoy (1230 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001580v1_decoy (1228 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001581v1_decoy (1227 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001582v1_decoy (1222 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001583v1_decoy (1222 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001584v1_decoy (1221 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001585v1_decoy (1221 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001586v1_decoy (1220 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001587v1_decoy (1218 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001588v1_decoy (1218 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001589v1_decoy (1216 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001590v1_decoy (1216 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001591v1_decoy (1212 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001592v1_decoy (1210 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001593v1_decoy (1209 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001594v1_decoy (1208 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001595v1_decoy (1208 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001596v1_decoy (1206 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001597v1_decoy (1205 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001598v1_decoy (1205 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001599v1_decoy (1202 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001600v1_decoy (1200 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001601v1_decoy (1199 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001602v1_decoy (1198 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001603v1_decoy (1198 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001604v1_decoy (1198 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001605v1_decoy (1195 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001606v1_decoy (1194 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001607v1_decoy (1191 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001608v1_decoy (1189 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001609v1_decoy (1188 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001610v1_decoy (1180 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001611v1_decoy (1180 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001612v1_decoy (1179 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001613v1_decoy (1172 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001614v1_decoy (1168 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001615v1_decoy (1166 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001616v1_decoy (1157 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001617v1_decoy (1156 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001618v1_decoy (1156 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001619v1_decoy (1155 bp)\r\n",
      "01:28:30.586 DEBUG GenomeLocParser -  chrUn_JTFH01001620v1_decoy (1154 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001621v1_decoy (1154 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001622v1_decoy (1149 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001623v1_decoy (1143 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001624v1_decoy (1143 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001625v1_decoy (1140 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001626v1_decoy (1137 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001627v1_decoy (1135 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001628v1_decoy (1135 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001629v1_decoy (1135 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001630v1_decoy (1127 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001631v1_decoy (1127 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001632v1_decoy (1126 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001633v1_decoy (1123 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001634v1_decoy (1123 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001635v1_decoy (1123 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001636v1_decoy (1122 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001637v1_decoy (1122 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001638v1_decoy (1121 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001639v1_decoy (1121 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001640v1_decoy (1119 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001641v1_decoy (1119 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001642v1_decoy (1119 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001643v1_decoy (1118 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001644v1_decoy (1115 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001645v1_decoy (1106 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001646v1_decoy (1106 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001647v1_decoy (1104 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001648v1_decoy (1102 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001649v1_decoy (1101 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001650v1_decoy (1098 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001651v1_decoy (1098 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001652v1_decoy (1096 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001653v1_decoy (1096 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001654v1_decoy (1095 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001655v1_decoy (1093 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001656v1_decoy (1090 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001657v1_decoy (1089 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001658v1_decoy (1087 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001659v1_decoy (1087 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001660v1_decoy (1085 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001661v1_decoy (1085 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001662v1_decoy (1085 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001663v1_decoy (1083 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001664v1_decoy (1080 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001665v1_decoy (1080 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001666v1_decoy (1079 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001667v1_decoy (1079 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001668v1_decoy (1079 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001669v1_decoy (1075 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001670v1_decoy (1074 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001671v1_decoy (1073 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001672v1_decoy (1070 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001673v1_decoy (1068 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001674v1_decoy (1067 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001675v1_decoy (1066 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001676v1_decoy (1066 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001677v1_decoy (1066 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001678v1_decoy (1063 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001679v1_decoy (1063 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001680v1_decoy (1063 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001681v1_decoy (1062 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001682v1_decoy (1058 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001683v1_decoy (1056 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001684v1_decoy (1052 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001685v1_decoy (1051 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001686v1_decoy (1051 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001687v1_decoy (1050 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001688v1_decoy (1048 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001689v1_decoy (1046 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001690v1_decoy (1046 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001691v1_decoy (1045 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001692v1_decoy (1043 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001693v1_decoy (1038 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001694v1_decoy (1036 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001695v1_decoy (1035 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001696v1_decoy (1035 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001697v1_decoy (1035 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001698v1_decoy (1033 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001699v1_decoy (1032 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001700v1_decoy (1031 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001701v1_decoy (1026 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001702v1_decoy (1026 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001703v1_decoy (1026 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001704v1_decoy (1023 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001705v1_decoy (1022 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001706v1_decoy (1020 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001707v1_decoy (1020 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001708v1_decoy (1020 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001709v1_decoy (1019 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001710v1_decoy (1018 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001711v1_decoy (1018 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001712v1_decoy (1017 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001713v1_decoy (1015 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001714v1_decoy (1015 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001715v1_decoy (1015 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001716v1_decoy (1014 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001717v1_decoy (1014 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001718v1_decoy (1013 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001719v1_decoy (1013 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001720v1_decoy (1013 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001721v1_decoy (1012 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001722v1_decoy (1011 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001723v1_decoy (1011 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001724v1_decoy (1009 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001725v1_decoy (1008 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001726v1_decoy (1008 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001727v1_decoy (1007 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001728v1_decoy (1007 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001729v1_decoy (1007 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001730v1_decoy (1006 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001731v1_decoy (1005 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001732v1_decoy (1003 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001733v1_decoy (1001 bp)\r\n",
      "01:28:30.587 DEBUG GenomeLocParser -  chrUn_JTFH01001734v1_decoy (1000 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001735v1_decoy (19311 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001736v1_decoy (11713 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001737v1_decoy (11263 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001738v1_decoy (9779 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001739v1_decoy (9568 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001740v1_decoy (9344 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001741v1_decoy (9188 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001742v1_decoy (9100 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001743v1_decoy (8771 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001744v1_decoy (8690 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001745v1_decoy (8566 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001746v1_decoy (8058 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001747v1_decoy (7759 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001748v1_decoy (7585 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001749v1_decoy (7471 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001750v1_decoy (7461 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001751v1_decoy (7342 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001752v1_decoy (7223 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001753v1_decoy (7064 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001754v1_decoy (6916 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001755v1_decoy (6897 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001756v1_decoy (6880 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001757v1_decoy (6857 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001758v1_decoy (6840 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001759v1_decoy (6728 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001760v1_decoy (6688 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001761v1_decoy (6553 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001762v1_decoy (6396 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001763v1_decoy (6345 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001764v1_decoy (6295 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001765v1_decoy (6266 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001766v1_decoy (6173 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001767v1_decoy (6171 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001768v1_decoy (6120 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001769v1_decoy (6105 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001770v1_decoy (6099 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001771v1_decoy (5893 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001772v1_decoy (5829 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001773v1_decoy (5793 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001774v1_decoy (5776 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001775v1_decoy (5759 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001776v1_decoy (5716 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001777v1_decoy (5708 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001778v1_decoy (5590 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001779v1_decoy (5566 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001780v1_decoy (5558 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001781v1_decoy (5418 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001782v1_decoy (5375 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001783v1_decoy (5300 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001784v1_decoy (5255 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001785v1_decoy (5157 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001786v1_decoy (5130 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001787v1_decoy (4978 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001788v1_decoy (4957 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001789v1_decoy (4947 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001790v1_decoy (4897 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001791v1_decoy (4867 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001792v1_decoy (4845 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001793v1_decoy (4678 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001794v1_decoy (4641 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001795v1_decoy (4592 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001796v1_decoy (4543 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001797v1_decoy (4532 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001798v1_decoy (4503 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001799v1_decoy (4495 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001800v1_decoy (4444 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001801v1_decoy (4414 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001802v1_decoy (4409 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001803v1_decoy (4302 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001804v1_decoy (4300 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001805v1_decoy (4277 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001806v1_decoy (4173 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001807v1_decoy (4169 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001808v1_decoy (4136 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001809v1_decoy (4101 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001810v1_decoy (4089 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001811v1_decoy (4015 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001812v1_decoy (4000 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001813v1_decoy (3973 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001814v1_decoy (3732 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001815v1_decoy (3709 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001816v1_decoy (3686 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001817v1_decoy (3676 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001818v1_decoy (3673 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001819v1_decoy (3672 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001820v1_decoy (3633 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001821v1_decoy (3633 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001822v1_decoy (3613 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001823v1_decoy (3605 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001824v1_decoy (3592 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001825v1_decoy (3586 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001826v1_decoy (3584 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001827v1_decoy (3577 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001828v1_decoy (3537 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001829v1_decoy (3510 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001830v1_decoy (3509 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001831v1_decoy (3488 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001832v1_decoy (3473 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001833v1_decoy (3445 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001834v1_decoy (3427 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001835v1_decoy (3395 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001836v1_decoy (3367 bp)\r\n",
      "01:28:30.588 DEBUG GenomeLocParser -  chrUn_JTFH01001837v1_decoy (3337 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001838v1_decoy (3324 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001839v1_decoy (3315 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001840v1_decoy (3313 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001841v1_decoy (3283 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001842v1_decoy (3250 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001843v1_decoy (3247 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001844v1_decoy (3237 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001845v1_decoy (3235 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001846v1_decoy (3200 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001847v1_decoy (3195 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001848v1_decoy (3175 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001849v1_decoy (3158 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001850v1_decoy (3143 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001851v1_decoy (3139 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001852v1_decoy (3138 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001853v1_decoy (3136 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001854v1_decoy (3132 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001855v1_decoy (3132 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001856v1_decoy (3095 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001857v1_decoy (3094 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001858v1_decoy (3093 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001859v1_decoy (3059 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001860v1_decoy (2985 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001861v1_decoy (2975 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001862v1_decoy (2967 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001863v1_decoy (2961 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001864v1_decoy (2955 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001865v1_decoy (2935 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001866v1_decoy (2933 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001867v1_decoy (2909 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001868v1_decoy (2904 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001869v1_decoy (2892 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001870v1_decoy (2886 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001871v1_decoy (2885 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001872v1_decoy (2878 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001873v1_decoy (2875 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001874v1_decoy (2861 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001875v1_decoy (2856 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001876v1_decoy (2838 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001877v1_decoy (2801 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001878v1_decoy (2797 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001879v1_decoy (2788 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001880v1_decoy (2773 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001881v1_decoy (2755 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001882v1_decoy (2754 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001883v1_decoy (2743 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001884v1_decoy (2725 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001885v1_decoy (2722 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001886v1_decoy (2682 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001887v1_decoy (2669 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001888v1_decoy (2663 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001889v1_decoy (2652 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001890v1_decoy (2647 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001891v1_decoy (2635 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001892v1_decoy (2633 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001893v1_decoy (2629 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001894v1_decoy (2612 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001895v1_decoy (2599 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001896v1_decoy (2566 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001897v1_decoy (2556 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001898v1_decoy (2551 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001899v1_decoy (2551 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001900v1_decoy (2538 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001901v1_decoy (2538 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001902v1_decoy (2525 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001903v1_decoy (2498 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001904v1_decoy (2496 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001905v1_decoy (2483 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001906v1_decoy (2475 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001907v1_decoy (2469 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001908v1_decoy (2455 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001909v1_decoy (2444 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001910v1_decoy (2437 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001911v1_decoy (2435 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001912v1_decoy (2427 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001913v1_decoy (2419 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001914v1_decoy (2413 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001915v1_decoy (2412 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001916v1_decoy (2400 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001917v1_decoy (2399 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001918v1_decoy (2396 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001919v1_decoy (2393 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001920v1_decoy (2386 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001921v1_decoy (2384 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001922v1_decoy (2382 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001923v1_decoy (2382 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001924v1_decoy (2367 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001925v1_decoy (2366 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001926v1_decoy (2362 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001927v1_decoy (2361 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001928v1_decoy (2353 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001929v1_decoy (2349 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001930v1_decoy (2348 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001931v1_decoy (2340 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001932v1_decoy (2339 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001933v1_decoy (2336 bp)\r\n",
      "01:28:30.589 DEBUG GenomeLocParser -  chrUn_JTFH01001934v1_decoy (2333 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001935v1_decoy (2330 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001936v1_decoy (2327 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001937v1_decoy (2318 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001938v1_decoy (2293 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001939v1_decoy (2292 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001940v1_decoy (2287 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001941v1_decoy (2274 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001942v1_decoy (2274 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001943v1_decoy (2267 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001944v1_decoy (2260 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001945v1_decoy (2257 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001946v1_decoy (2240 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001947v1_decoy (2239 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001948v1_decoy (2232 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001949v1_decoy (2230 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001950v1_decoy (2230 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001951v1_decoy (2222 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001952v1_decoy (2216 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001953v1_decoy (2214 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001954v1_decoy (2210 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001955v1_decoy (2203 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001956v1_decoy (2197 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001957v1_decoy (2196 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001958v1_decoy (2196 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001959v1_decoy (2179 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001960v1_decoy (2178 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001961v1_decoy (2178 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001962v1_decoy (2172 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001963v1_decoy (2170 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001964v1_decoy (2167 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001965v1_decoy (2167 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001966v1_decoy (2157 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001967v1_decoy (2153 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001968v1_decoy (2151 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001969v1_decoy (2147 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001970v1_decoy (2145 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001971v1_decoy (2142 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001972v1_decoy (2142 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001973v1_decoy (2136 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001974v1_decoy (2130 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001975v1_decoy (2128 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001976v1_decoy (2126 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001977v1_decoy (2126 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001978v1_decoy (2119 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001979v1_decoy (2107 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001980v1_decoy (2091 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001981v1_decoy (2087 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001982v1_decoy (2086 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001983v1_decoy (2083 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001984v1_decoy (2075 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001985v1_decoy (2075 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001986v1_decoy (2072 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001987v1_decoy (2068 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001988v1_decoy (2067 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001989v1_decoy (2055 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001990v1_decoy (2051 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001991v1_decoy (2050 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001992v1_decoy (2033 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001993v1_decoy (2024 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001994v1_decoy (2016 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001995v1_decoy (2011 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001996v1_decoy (2009 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001997v1_decoy (2003 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  chrUn_JTFH01001998v1_decoy (2001 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*01:01:01:01 (3503 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*01:01:01:02N (3291 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*01:01:38L (3374 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*01:02 (3374 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*01:03 (3503 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*01:04N (3136 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*01:09 (3105 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*01:11N (3374 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*01:14 (3095 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*01:16N (2985 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*01:20 (3105 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:01:01:01 (3517 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:01:01:02L (3287 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:01:01:03 (3023 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:01:01:04 (3516 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:02:01 (2917 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:03:01 (3517 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:03:03 (3148 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:05:01 (3517 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:06:01 (3517 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:07:01 (3517 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:10 (3517 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:251 (3517 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:259 (2978 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:264 (3002 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:265 (3148 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:266 (3084 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:269 (3101 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:279 (3103 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:32N (3517 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:376 (3104 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:43N (3218 bp)\r\n",
      "01:28:30.590 DEBUG GenomeLocParser -  HLA-A*02:455 (3118 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*02:48 (3517 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*02:51 (3109 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*02:533 (3217 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*02:53N (3305 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*02:57 (3054 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*02:60:01 (3112 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*02:65 (3387 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*02:68 (3109 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*02:77 (3371 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*02:81 (3309 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*02:89 (3371 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*02:95 (3388 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*03:01:01:01 (3502 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*03:01:01:02N (3373 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*03:01:01:03 (3094 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*03:02:01 (3502 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*03:11N (3404 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*03:21N (3095 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*03:36N (3142 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*11:01:01 (3503 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*11:01:18 (3503 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*11:02:01 (3503 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*11:05 (3373 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*11:110 (2903 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*11:25 (3073 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*11:50Q (3362 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*11:60 (3241 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*11:69N (3500 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*11:74 (3227 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*11:75 (3184 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*11:77 (3233 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*23:01:01 (3502 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*23:09 (3104 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*23:38N (3020 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:02:01:01 (3502 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:02:01:02L (3502 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:02:01:03 (3075 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:02:03Q (3247 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:02:10 (3356 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:03:01 (3502 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:07:01 (3502 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:08 (3502 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:09N (3502 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:10:01 (3502 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:11N (3503 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:152 (3176 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:20 (3502 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:215 (3116 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:61 (3043 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*24:86N (3415 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*25:01:01 (2917 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*26:01:01 (3517 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*26:11N (3091 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*26:15 (3217 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*26:50 (3141 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*29:01:01:01 (3518 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*29:01:01:02N (3303 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*29:02:01:01 (3518 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*29:02:01:02 (3518 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*29:46 (3310 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*30:01:01 (3503 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*30:02:01:01 (2903 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*30:02:01:02 (3374 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*30:04:01 (3503 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*30:89 (2903 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*31:01:02 (3518 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*31:01:23 (2918 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*31:04 (2918 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*31:14N (3090 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*31:46 (3075 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*32:01:01 (3518 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*32:06 (3389 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*33:01:01 (3518 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*33:03:01 (3518 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*33:07 (3389 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*34:01:01 (3517 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*34:02:01 (3096 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*36:01 (2903 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*43:01 (3388 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*66:01:01 (3517 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*66:17 (3075 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:01:01:01 (2930 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:01:01:02 (3517 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:01:02:01 (3517 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:01:02:02 (3388 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:02:01:01 (3517 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:02:01:02 (3506 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:02:01:03 (2909 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:02:02 (2916 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:03:01 (2917 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:08:01 (3120 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:113 (3070 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:17 (3134 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:18N (3237 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:22 (3119 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*68:71 (3198 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*69:01 (2917 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*74:01 (2918 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*74:02:01:01 (2918 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*74:02:01:02 (3518 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*80:01:01:01 (3263 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-A*80:01:01:02 (3055 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-B*07:02:01 (3323 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-B*07:05:01 (2676 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-B*07:06 (2676 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-B*07:156 (2967 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-B*07:33:01 (3239 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-B*07:41 (3266 bp)\r\n",
      "01:28:30.591 DEBUG GenomeLocParser -  HLA-B*07:44 (3270 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*07:50 (3323 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*08:01:01 (3322 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*08:08N (3035 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*08:132 (2675 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*08:134 (2959 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*08:19N (3322 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*08:20 (3322 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*08:33 (3322 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*08:79 (2676 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*13:01:01 (3324 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*13:02:01 (3324 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*13:02:03 (3323 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*13:02:09 (2919 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*13:08 (3324 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*13:15 (3323 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*13:25 (2689 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*14:01:01 (3312 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*14:02:01 (3312 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*14:07N (3255 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:01:01:01 (3336 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:01:01:02N (1208 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:01:01:03 (3026 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:02:01 (3335 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:03:01 (2689 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:04:01 (3052 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:07:01 (3336 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:108 (3283 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:10:01 (2689 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:11:01 (3336 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:13:01 (2688 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:16:01 (2688 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:17:01:01 (3051 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:17:01:02 (3051 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:18:01 (3336 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:220 (2878 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:25:01 (3335 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:27:01 (2689 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:32:01 (3336 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:42 (3333 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:58 (3336 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:66 (2902 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:77 (3336 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*15:83 (3337 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*18:01:01:01 (3323 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*18:01:01:02 (3323 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*18:02 (2686 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*18:03 (3323 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*18:17N (2979 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*18:26 (3323 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*18:94N (2970 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*27:04:01 (3325 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*27:05:02 (3325 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*27:05:18 (3321 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*27:06 (3325 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*27:07:01 (2677 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*27:131 (3325 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*27:24 (2677 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*27:25 (2677 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*27:32 (3325 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*35:01:01:01 (3327 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*35:01:01:02 (3327 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*35:01:22 (2806 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*35:02:01 (3327 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*35:03:01 (2689 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*35:05:01 (2690 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*35:08:01 (2689 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*35:14:02 (3327 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*35:241 (3042 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*35:41 (3327 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*37:01:01 (3324 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*37:01:05 (2687 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*38:01:01 (3312 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*38:02:01 (3312 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*38:14 (2738 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:01:01:01 (3155 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:01:01:02L (3153 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:01:01:03 (3312 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:01:03 (3155 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:01:16 (3155 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:01:21 (3312 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:05:01 (2675 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:06:02 (2674 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:10:01 (3027 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:13:02 (3255 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:14 (2765 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:34 (3254 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*39:38Q (2675 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*40:01:01 (2676 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*40:01:02 (3323 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*40:02:01 (3258 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*40:03 (2677 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*40:06:01:01 (3325 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*40:06:01:02 (3299 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*40:10:01 (3304 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*40:150 (2800 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*40:40 (2677 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*40:72:01 (3283 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*40:79 (3257 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*41:01:01 (3322 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*41:02:01 (3322 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*42:01:01 (3322 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*42:02 (2675 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*42:08 (3165 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*44:02:01:01 (3323 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*44:02:01:02S (3152 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*44:02:01:03 (3152 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*44:02:17 (3323 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*44:02:27 (2872 bp)\r\n",
      "01:28:30.592 DEBUG GenomeLocParser -  HLA-B*44:03:01 (3323 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*44:03:02 (2676 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*44:04 (3239 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*44:09 (3317 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*44:138Q (3043 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*44:150 (2676 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*44:23N (3323 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*44:26 (2804 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*44:46 (3323 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*44:49 (3039 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*44:56N (2676 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*45:01:01 (3338 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*45:04 (3339 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*46:01:01 (3336 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*46:01:05 (2891 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*47:01:01:01 (3041 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*47:01:01:02 (3041 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*48:01:01 (3323 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*48:03:01 (2676 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*48:04 (2676 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*48:08 (3323 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*49:01:01 (3340 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*49:32 (3340 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*50:01:01 (3340 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*51:01:01 (3327 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*51:01:02 (3043 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*51:02:01 (3327 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*51:07:01 (3327 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*51:42 (2962 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*52:01:01:01 (3327 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*52:01:01:02 (3327 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*52:01:01:03 (3327 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*52:01:02 (3327 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*53:01:01 (3327 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*53:11 (3274 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*54:01:01 (3332 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*54:18 (2813 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*55:01:01 (3332 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*55:01:03 (3332 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*55:02:01 (3333 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*55:12 (3332 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*55:24 (3332 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*55:48 (2980 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*56:01:01 (2688 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*56:03 (2688 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*56:04 (2688 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*57:01:01 (3337 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*57:03:01 (2689 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*57:06 (3284 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*57:11 (3306 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*57:29 (3337 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*58:01:01 (3336 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*58:31N (3004 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*59:01:01:01 (3333 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*59:01:01:02 (3332 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*67:01:01 (3312 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*67:01:02 (2675 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*67:02 (3307 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*73:01 (3323 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*78:01:01 (3327 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*81:01 (2676 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-B*82:02:01 (3050 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*01:02:01 (3349 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*01:02:11 (3057 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*01:02:29 (3349 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*01:02:30 (3333 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*01:03 (3349 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*01:06 (2895 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*01:08 (3349 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*01:14 (2895 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*01:21 (2895 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*01:30 (3349 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*01:40 (2968 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*02:02:02:01 (3347 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*02:02:02:02 (3347 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*02:10 (2893 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*02:11 (3320 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*02:16:02 (3029 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*02:69 (2933 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*02:85 (3347 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*02:86 (3347 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*02:87 (3064 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:02:01 (2894 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:02:02:01 (3348 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:02:02:02 (2896 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:02:02:03 (3348 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:03:01 (3348 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:04:01:01 (3348 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:04:01:02 (3348 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:04:02 (2877 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:04:04 (2966 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:05 (2894 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:06 (2894 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:100 (3034 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:13:01 (3065 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:20N (3321 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:219 (3070 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:261 (3348 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:40:01 (2894 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:41:02 (3328 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:46 (2997 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*03:61 (2894 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*04:01:01:01 (3349 bp)\r\n",
      "01:28:30.593 DEBUG GenomeLocParser -  HLA-C*04:01:01:02 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*04:01:01:03 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*04:01:01:04 (3012 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*04:01:01:05 (2931 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*04:01:62 (3329 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*04:03:01 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*04:06 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*04:09N (2991 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*04:128 (3086 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*04:161 (3237 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*04:177 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*04:70 (3058 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*04:71 (3086 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*05:01:01:01 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*05:01:01:02 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*05:08 (3059 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*05:09:01 (3322 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*05:93 (2946 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*06:02:01:01 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*06:02:01:02 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*06:02:01:03 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*06:23 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*06:24 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*06:46N (2987 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:01:01:01 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:01:01:02 (3093 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:01:02 (3352 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:01:19 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:01:27 (3195 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:01:45 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:02:01:01 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:02:01:02 (3074 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:02:01:03 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:02:01:04 (3353 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:02:01:05 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:02:05 (2903 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:02:06 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:02:64 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:04:01 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:04:02 (3343 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:06 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:149 (3098 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:18 (3353 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:19 (3222 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:26 (3069 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:30 (2903 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:32N (3334 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:384 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:385 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:386 (3183 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:391 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:392 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:49 (2935 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:56:02 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:66 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*07:67 (3354 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:01:01 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:01:03 (2998 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:02:01:01 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:02:01:02 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:03:01 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:04:01 (2895 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:112 (3178 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:20 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:21 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:22 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:24 (2895 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:27 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:36N (3097 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:40 (2978 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:41 (3019 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*08:62 (3086 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*12:02:02 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*12:03:01:01 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*12:03:01:02 (3348 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*12:08 (3066 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*12:13 (3058 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*12:19 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*12:22 (2895 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*12:99 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*14:02:01 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*14:03 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*14:21N (3099 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*14:23 (2976 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*15:02:01 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*15:05:01 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*15:05:02 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*15:13 (2895 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*15:16 (3066 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*15:17 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*15:96Q (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*16:01:01 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*16:02:01 (2895 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*16:04:01 (3349 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*17:01:01:01 (3368 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*17:01:01:02 (3368 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*17:01:01:03 (3368 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*17:03 (3197 bp)\r\n",
      "01:28:30.594 DEBUG GenomeLocParser -  HLA-C*18:01 (3346 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:01:02 (6489 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:02:01:01 (6484 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:02:01:02 (6485 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:02:01:03 (6485 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:02:01:04 (6492 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:03:01:01 (6485 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:03:01:02 (6492 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:04:01:01 (6484 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:04:01:02 (6485 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:05:01 (6485 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:07 (5959 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:10 (5790 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*01:11 (5926 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*02:01 (6403 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*03:01:01 (6437 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*03:02 (6437 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*03:03:01 (6437 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*04:01:02:01 (5853 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*04:01:02:02 (5666 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*04:02 (6210 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*05:01:01:01 (5806 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*05:01:01:02 (6529 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*05:03 (6121 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*05:05:01:01 (6593 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*05:05:01:02 (6597 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*05:05:01:03 (6393 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*05:11 (6589 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQA1*06:01:01 (5878 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*02:01:01 (7480 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*02:02:01 (7471 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*03:01:01:01 (7231 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*03:01:01:02 (7230 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*03:01:01:03 (7231 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*03:02:01 (7126 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*03:03:02:01 (7126 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*03:03:02:02 (7126 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*03:03:02:03 (6800 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*03:05:01 (6934 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*05:01:01:01 (7090 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*05:01:01:02 (7090 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*05:03:01:01 (7089 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*05:03:01:02 (7089 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*06:01:01 (7111 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*06:02:01 (7102 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*06:03:01 (7103 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DQB1*06:09:01 (7102 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*01:01:01 (10741 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*01:02:01 (11229 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*03:01:01:01 (13908 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*03:01:01:02 (13426 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*04:03:01 (15246 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*07:01:01:01 (16110 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*07:01:01:02 (16120 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*08:03:02 (13562 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*09:21 (16039 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*10:01:01 (13501 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*11:01:01 (13921 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*11:01:02 (13931 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*11:04:01 (13919 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*12:01:01 (13404 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*12:17 (11260 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*13:01:01 (13935 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*13:02:01 (13941 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*14:05:01 (13933 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*14:54:01 (13936 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*15:01:01:01 (11080 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*15:01:01:02 (11571 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*15:01:01:03 (11056 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*15:01:01:04 (11056 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*15:02:01 (10313 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*15:03:01:01 (11567 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*15:03:01:02 (11569 bp)\r\n",
      "01:28:30.595 DEBUG GenomeLocParser -  HLA-DRB1*16:02:01 (11005 bp)\r\n",
      "01:28:30.625 INFO  DetermineGermlineContigPloidy - Aggregating read-count file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA18525.tsv (1 / 7)\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:28:31.017 INFO  DetermineGermlineContigPloidy - Aggregating read-count file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA18939.tsv (2 / 7)\n",
      "01:28:31.370 INFO  DetermineGermlineContigPloidy - Aggregating read-count file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA19017.tsv (3 / 7)\n",
      "01:28:31.704 INFO  DetermineGermlineContigPloidy - Aggregating read-count file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA19625.tsv (4 / 7)\n",
      "01:28:31.967 INFO  DetermineGermlineContigPloidy - Aggregating read-count file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA19648.tsv (5 / 7)\n",
      "01:28:32.237 INFO  DetermineGermlineContigPloidy - Aggregating read-count file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA20502.tsv (6 / 7)\n",
      "01:28:32.586 INFO  DetermineGermlineContigPloidy - Aggregating read-count file /home/jupyter/notebooks/germline-cnv/data/Tutorial11684/cvg/NA20845.tsv (7 / 7)\n",
      "01:28:32.932 DEBUG ScriptExecutor - Executing:\n",
      "01:28:32.932 DEBUG ScriptExecutor -   python\n",
      "01:28:32.932 DEBUG ScriptExecutor -   /tmp/cohort_determine_ploidy_and_depth.6062301064784209829.py\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --sample_coverage_metadata=/tmp/samples-by-coverage-per-contig17412580304077095937.tsv\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --output_calls_path=/home/jupyter/notebooks/germline-cnv/sandbox/ploidy-cohort7-calls\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --mapping_error_rate=3.000000e-01\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --psi_s_scale=1.000000e-04\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --mean_bias_sd=1.000000e+00\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --psi_j_scale=1.000000e-03\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --learning_rate=5.000000e-02\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --adamax_beta1=9.000000e-01\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --adamax_beta2=9.990000e-01\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --log_emission_samples_per_round=2000\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --log_emission_sampling_rounds=100\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --log_emission_sampling_median_rel_error=5.000000e-04\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --max_advi_iter_first_epoch=1000\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --max_advi_iter_subsequent_epochs=500\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --min_training_epochs=20\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --max_training_epochs=20\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --initial_temperature=2.000000e+00\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --num_thermal_advi_iters=1000\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --convergence_snr_averaging_window=5000\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --convergence_snr_trigger_threshold=1.000000e-01\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --convergence_snr_countdown_window=10\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --max_calling_iters=1\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --caller_update_convergence_threshold=1.000000e-03\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --caller_internal_admixing_rate=7.500000e-01\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --caller_external_admixing_rate=7.500000e-01\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --disable_caller=false\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --disable_sampler=false\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --disable_annealing=false\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --interval_list=/tmp/intervals14242751390761912678.tsv\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --contig_ploidy_prior_table=/home/jupyter/notebooks/germline-cnv/data/Tutorial11684/chr20XY_contig_ploidy_priors.tsv\n",
      "01:28:32.932 DEBUG ScriptExecutor -   --output_model_path=/home/jupyter/notebooks/germline-cnv/sandbox/ploidy-cohort7-model\n",
      "01:29:10.870 INFO cohort_determine_ploidy_and_depth - THEANO_FLAGS environment variable has been set to: device=cpu,floatX=float64,optimizer=fast_run,compute_test_value=ignore,openmp=true,blas.ldflags=-lmkl_rt,openmp_elemwise_minsize=10\n",
      "01:29:11.156 INFO gcnvkernel.structs.metadata - Generating intervals metadata...\n",
      "01:29:11.180 INFO gcnvkernel.tasks.task_cohort_ploidy_determination - Instantiating the germline contig ploidy determination model...\n",
      "01:31:06.427 INFO gcnvkernel.tasks.task_cohort_ploidy_determination - Instantiating the ploidy emission sampler...\n",
      "01:31:06.428 INFO gcnvkernel.tasks.task_cohort_ploidy_determination - Instantiating the ploidy caller...\n",
      "01:31:21.759 INFO gcnvkernel.models.fancy_model - Global model variables: {'mean_bias_j_lowerbound__', 'psi_j_log__'}\n",
      "01:31:21.760 INFO gcnvkernel.models.fancy_model - Sample-specific model variables: {'psi_s_log__'}\n",
      "01:31:21.760 INFO gcnvkernel.tasks.inference_task_base - Instantiating the convergence tracker...\n",
      "01:31:21.760 INFO gcnvkernel.tasks.inference_task_base - Setting up DA-ADVI...\n",
      "01:33:54.740 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/1000 [00:00<?, ?it/s]\n",
      "01:33:54.843 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -248.608 +/- 53.228, SNR: 710.6, T: 1.99:   1%|1         | 14/1000 [00:00<00:07, 137.86it/s]\n",
      "01:33:54.947 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -210.657 +/- 46.529, SNR: 334.9, T: 1.97:   3%|3         | 32/1000 [00:00<00:06, 157.88it/s]\n",
      "01:33:55.055 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -185.890 +/- 48.231, SNR: 191.1, T: 1.95:   5%|4         | 48/1000 [00:00<00:06, 153.36it/s]\n",
      "01:33:55.155 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -151.401 +/- 51.712, SNR: 78.9, T: 1.91:   9%|8         | 86/1000 [00:00<00:03, 239.75it/s]\n",
      "01:33:55.256 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -139.080 +/- 41.010, SNR: 75.1, T: 1.86:  14%|#3        | 137/1000 [00:00<00:02, 334.55it/s]\n",
      "01:33:55.357 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -125.878 +/- 37.024, SNR: 61.1, T: 1.82:  18%|#7        | 179/1000 [00:00<00:02, 361.53it/s]\n",
      "01:33:55.458 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -113.116 +/- 34.873, SNR: 47.2, T: 1.78:  22%|##2       | 223/1000 [00:00<00:02, 385.67it/s]\n",
      "01:33:55.559 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -103.489 +/- 32.897, SNR: 38.3, T: 1.73:  27%|##7       | 271/1000 [00:00<00:01, 414.73it/s]\n",
      "01:33:55.660 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -91.562 +/- 30.692, SNR: 28.6, T: 1.65:  35%|###4      | 346/1000 [00:00<00:01, 516.50it/s]\n",
      "01:33:55.766 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -85.472 +/- 29.351, SNR: 24.4, T: 1.60:  40%|###9      | 398/1000 [00:01<00:01, 508.72it/s]\n",
      "01:33:55.867 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -78.778 +/- 28.038, SNR: 20.1, T: 1.54:  46%|####6     | 465/1000 [00:01<00:00, 555.69it/s]\n",
      "01:33:55.967 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -73.617 +/- 27.096, SNR: 17.0, T: 1.47:  53%|#####2    | 526/1000 [00:01<00:00, 571.67it/s]\n",
      "01:33:56.093 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -69.868 +/- 26.158, SNR: 15.1, T: 1.42:  58%|#####8    | 584/1000 [00:01<00:00, 531.63it/s]\n",
      "01:33:56.212 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -66.682 +/- 25.423, SNR: 13.5, T: 1.36:  64%|######3   | 638/1000 [00:01<00:00, 507.13it/s]\n",
      "01:33:56.344 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -63.982 +/- 24.781, SNR: 12.2, T: 1.31:  69%|######9   | 690/1000 [00:01<00:00, 468.85it/s]\n",
      "01:33:56.470 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -61.695 +/- 24.251, SNR: 11.2, T: 1.26:  74%|#######3  | 738/1000 [00:01<00:00, 441.08it/s]\n",
      "01:33:56.594 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -59.838 +/- 23.762, SNR: 10.4, T: 1.22:  78%|#######8  | 783/1000 [00:01<00:00, 417.40it/s]\n",
      "01:33:56.694 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -57.244 +/- 23.123, SNR: 9.3, T: 1.15:  85%|########5 | 850/1000 [00:01<00:00, 482.81it/s]\n",
      "01:33:56.809 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -55.470 +/- 22.695, SNR: 8.6, T: 1.10:  90%|######### | 900/1000 [00:02<00:00, 468.51it/s]\n",
      "01:33:56.927 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -53.925 +/- 22.300, SNR: 8.0, T: 1.05:  95%|#########4| 948/1000 [00:02<00:00, 450.26it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:33:57.047 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -52.542 +/- 21.948, SNR: 7.5, T: 1.01:  99%|#########9| 994/1000 [00:02<00:00, 429.06it/s]\n",
      "01:33:57.077 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 1) ELBO: -52.371 +/- 21.902, SNR: 7.4, T: 1.00: 100%|##########| 1000/1000 [00:02<00:00, 428.06it/s]\n",
      "01:33:57.079 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 1):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:33:57.276 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 1) relative error: 1.0000 +/- 0.0000:   1%|1         | 1/100 [00:00<00:19,  5.07it/s]\n",
      "01:33:57.394 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 1) relative error: 0.0008 +/- 0.0035:   2%|2         | 2/100 [00:00<00:14,  6.66it/s]\n",
      "01:33:57.472 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 1) relative error: 0.0004 +/- 0.0027:   2%|2         | 2/100 [00:00<00:19,  5.09it/s]\n",
      "01:33:57.473 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 1):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:33:57.479 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 1) ploidy update size: 0.107082: 100%|##########| 1/1 [00:00<00:00, 162.59it/s]\n",
      "01:33:57.479 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:33:57.580 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 2) ELBO: -50.967 +/- 21.491, SNR: 6.9, T: 1.00:  11%|#         | 54/500 [00:00<00:00, 539.09it/s]\n",
      "01:33:57.681 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 2) ELBO: -49.599 +/- 21.099, SNR: 6.5, T: 1.00:  22%|##2       | 110/500 [00:00<00:00, 548.60it/s]\n",
      "01:33:57.782 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 2) ELBO: -48.262 +/- 20.712, SNR: 6.0, T: 1.00:  34%|###3      | 169/500 [00:00<00:00, 565.24it/s]\n",
      "01:33:57.889 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 2) ELBO: -47.070 +/- 20.360, SNR: 5.6, T: 1.00:  45%|####5     | 226/500 [00:00<00:00, 550.26it/s]\n",
      "01:33:58.131 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 2) ELBO: -45.998 +/- 20.028, SNR: 5.3, T: 1.00:  56%|#####6    | 282/500 [00:00<00:00, 368.96it/s]\n",
      "01:33:58.310 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 2) ELBO: -45.194 +/- 19.783, SNR: 5.1, T: 1.00:  65%|######5   | 326/500 [00:00<00:00, 322.76it/s]\n",
      "01:33:58.441 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 2) ELBO: -44.529 +/- 19.580, SNR: 4.9, T: 1.00:  73%|#######2  | 364/500 [00:00<00:00, 313.74it/s]\n",
      "01:33:58.584 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 2) ELBO: -43.952 +/- 19.397, SNR: 4.7, T: 1.00:  80%|#######9  | 399/500 [00:01<00:00, 292.33it/s]\n",
      "01:33:58.750 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 2) ELBO: -43.445 +/- 19.234, SNR: 4.6, T: 1.00:  86%|########6 | 431/500 [00:01<00:00, 259.07it/s]\n",
      "01:33:58.851 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 2) ELBO: -43.004 +/- 19.098, SNR: 4.4, T: 1.00:  92%|#########1| 459/500 [00:01<00:00, 263.05it/s]\n",
      "01:33:58.962 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 2) ELBO: -42.577 +/- 18.965, SNR: 4.3, T: 1.00:  97%|#########7| 487/500 [00:01<00:00, 260.52it/s]\n",
      "01:33:59.026 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 2) ELBO: -42.391 +/- 18.902, SNR: 4.3, T: 1.00: 100%|##########| 500/500 [00:01<00:00, 323.23it/s]\n",
      "01:33:59.027 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 2):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:33:59.209 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 2) relative error: 0.0140 +/- 0.0761:   1%|1         | 1/100 [00:00<00:17,  5.51it/s]\n",
      "01:33:59.332 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 2) relative error: 0.0022 +/- 0.0049:   2%|2         | 2/100 [00:00<00:14,  6.77it/s]\n",
      "01:33:59.492 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 2) relative error: 0.0006 +/- 0.0020:   4%|4         | 4/100 [00:00<00:10,  9.52it/s]\n",
      "01:33:59.581 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 2) relative error: 0.0002 +/- 0.0019:   4%|4         | 4/100 [00:00<00:13,  7.23it/s]\n",
      "01:33:59.582 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 2):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:33:59.582 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 2) ploidy update size: 0.0499715: 100%|##########| 1/1 [00:00<00:00, 2565.32it/s]\n",
      "01:33:59.583 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:33:59.685 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -41.802 +/- 18.706, SNR: 4.1, T: 1.00:   8%|8         | 42/500 [00:00<00:01, 413.20it/s]\n",
      "01:33:59.793 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -41.208 +/- 18.508, SNR: 3.9, T: 1.00:  17%|#7        | 86/500 [00:00<00:01, 409.89it/s]\n",
      "01:33:59.896 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -40.678 +/- 18.323, SNR: 3.8, T: 1.00:  26%|##5       | 128/500 [00:00<00:00, 409.77it/s]\n",
      "01:33:59.996 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -40.153 +/- 18.140, SNR: 3.7, T: 1.00:  34%|###4      | 171/500 [00:00<00:00, 417.36it/s]\n",
      "01:34:00.163 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -39.662 +/- 17.967, SNR: 3.5, T: 1.00:  43%|####2     | 213/500 [00:00<00:00, 337.36it/s]\n",
      "01:34:00.306 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -39.254 +/- 17.824, SNR: 3.4, T: 1.00:  50%|####9     | 249/500 [00:00<00:00, 305.58it/s]\n",
      "01:34:00.492 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -38.894 +/- 17.695, SNR: 3.4, T: 1.00:  56%|#####6    | 282/500 [00:00<00:00, 253.79it/s]\n",
      "01:34:00.600 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -38.599 +/- 17.588, SNR: 3.3, T: 1.00:  62%|######2   | 310/500 [00:01<00:00, 255.17it/s]\n",
      "01:34:00.701 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -38.311 +/- 17.482, SNR: 3.2, T: 1.00:  68%|######7   | 338/500 [00:01<00:00, 260.87it/s]\n",
      "01:34:00.811 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -38.033 +/- 17.379, SNR: 3.1, T: 1.00:  73%|#######3  | 366/500 [00:01<00:00, 258.65it/s]\n",
      "01:34:00.928 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -37.774 +/- 17.280, SNR: 3.1, T: 1.00:  79%|#######8  | 393/500 [00:01<00:00, 250.98it/s]\n",
      "01:34:01.043 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -37.521 +/- 17.188, SNR: 3.0, T: 1.00:  84%|########3 | 419/500 [00:01<00:00, 243.05it/s]\n",
      "01:34:01.144 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -37.170 +/- 17.061, SNR: 2.9, T: 1.00:  91%|#########1| 456/500 [00:01<00:00, 276.37it/s]\n",
      "01:34:01.244 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -36.858 +/- 16.946, SNR: 2.9, T: 1.00:  98%|#########8| 490/500 [00:01<00:00, 293.91it/s]\n",
      "01:34:01.292 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 3) ELBO: -36.769 +/- 16.913, SNR: 2.8, T: 1.00: 100%|##########| 500/500 [00:01<00:00, 292.69it/s]\n",
      "01:34:01.292 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 3):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:01.569 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 3) relative error: 0.0415 +/- 0.0752:   1%|1         | 1/100 [00:00<00:27,  3.61it/s]\n",
      "01:34:01.692 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 3) relative error: 0.0005 +/- 0.0036:   1%|1         | 1/100 [00:00<00:39,  2.50it/s]\n",
      "01:34:01.693 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 3):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:01.694 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 3) ploidy update size: 0.0493655: 100%|##########| 1/1 [00:00<00:00, 6232.25it/s]\n",
      "01:34:01.694 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:01.795 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -36.450 +/- 16.785, SNR: 2.8, T: 1.00:   8%|7         | 38/500 [00:00<00:01, 374.84it/s]\n",
      "01:34:01.910 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -36.140 +/- 16.660, SNR: 2.7, T: 1.00:  15%|#5        | 76/500 [00:00<00:01, 348.77it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:34:02.023 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -35.863 +/- 16.548, SNR: 2.6, T: 1.00:  22%|##2       | 111/500 [00:00<00:01, 329.83it/s]\n",
      "01:34:02.123 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -35.592 +/- 16.438, SNR: 2.6, T: 1.00:  29%|##9       | 146/500 [00:00<00:01, 337.15it/s]\n",
      "01:34:02.225 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -35.339 +/- 16.333, SNR: 2.5, T: 1.00:  36%|###6      | 180/500 [00:00<00:00, 336.77it/s]\n",
      "01:34:02.387 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -35.096 +/- 16.229, SNR: 2.5, T: 1.00:  43%|####2     | 214/500 [00:00<00:01, 280.02it/s]\n",
      "01:34:02.604 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -34.887 +/- 16.139, SNR: 2.4, T: 1.00:  49%|####8     | 244/500 [00:00<00:01, 214.94it/s]\n",
      "01:34:02.783 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -34.716 +/- 16.066, SNR: 2.4, T: 1.00:  54%|#####3    | 269/500 [00:01<00:01, 188.84it/s]\n",
      "01:34:02.883 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -34.537 +/- 15.987, SNR: 2.4, T: 1.00:  59%|#####9    | 296/500 [00:01<00:00, 206.28it/s]\n",
      "01:34:02.987 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -34.340 +/- 15.902, SNR: 2.3, T: 1.00:  65%|######5   | 326/500 [00:01<00:00, 226.66it/s]\n",
      "01:34:03.091 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -34.177 +/- 15.832, SNR: 2.3, T: 1.00:  70%|#######   | 351/500 [00:01<00:00, 230.09it/s]\n",
      "01:34:03.193 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -34.010 +/- 15.760, SNR: 2.3, T: 1.00:  75%|#######5  | 377/500 [00:01<00:00, 236.56it/s]\n",
      "01:34:03.304 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -33.855 +/- 15.692, SNR: 2.2, T: 1.00:  80%|########  | 402/500 [00:01<00:00, 233.60it/s]\n",
      "01:34:03.421 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -33.699 +/- 15.625, SNR: 2.2, T: 1.00:  85%|########5 | 427/500 [00:01<00:00, 227.44it/s]\n",
      "01:34:03.522 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -33.547 +/- 15.560, SNR: 2.2, T: 1.00:  90%|######### | 452/500 [00:01<00:00, 232.95it/s]\n",
      "01:34:03.627 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -33.390 +/- 15.492, SNR: 2.1, T: 1.00:  96%|#########5| 478/500 [00:01<00:00, 237.13it/s]\n",
      "01:34:03.722 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 4) ELBO: -33.261 +/- 15.435, SNR: 2.1, T: 1.00: 100%|##########| 500/500 [00:02<00:00, 246.52it/s]\n",
      "01:34:03.723 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 4):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:03.991 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 4) relative error: 0.0500 +/- 0.1132:   1%|1         | 1/100 [00:00<00:26,  3.72it/s]\n",
      "01:34:04.095 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 4) relative error: 0.0011 +/- 0.0031:   2%|2         | 2/100 [00:00<00:16,  5.82it/s]\n",
      "01:34:04.184 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 4) relative error: 0.0004 +/- 0.0032:   2%|2         | 2/100 [00:00<00:22,  4.33it/s]\n",
      "01:34:04.186 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 4):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:04.187 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 4) ploidy update size: 0.0428752: 100%|##########| 1/1 [00:00<00:00, 1899.59it/s]\n",
      "01:34:04.188 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:04.290 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -33.102 +/- 15.364, SNR: 2.1, T: 1.00:   6%|5         | 28/500 [00:00<00:01, 276.44it/s]\n",
      "01:34:04.391 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -32.930 +/- 15.286, SNR: 2.0, T: 1.00:  12%|#1        | 59/500 [00:00<00:01, 294.00it/s]\n",
      "01:34:04.494 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -32.760 +/- 15.205, SNR: 2.0, T: 1.00:  18%|#8        | 91/500 [00:00<00:01, 301.70it/s]\n",
      "01:34:04.596 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -32.589 +/- 15.124, SNR: 2.0, T: 1.00:  25%|##4       | 124/500 [00:00<00:01, 310.03it/s]\n",
      "01:34:04.737 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -32.427 +/- 15.050, SNR: 1.9, T: 1.00:  31%|###1      | 155/500 [00:00<00:01, 270.59it/s]\n",
      "01:34:04.851 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -32.285 +/- 14.983, SNR: 1.9, T: 1.00:  37%|###6      | 183/500 [00:00<00:01, 261.95it/s]\n",
      "01:34:04.970 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -32.153 +/- 14.920, SNR: 1.9, T: 1.00:  42%|####2     | 210/500 [00:00<00:01, 250.60it/s]\n",
      "01:34:05.086 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -32.025 +/- 14.859, SNR: 1.9, T: 1.00:  47%|####7     | 236/500 [00:00<00:01, 241.76it/s]\n",
      "01:34:05.202 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -31.906 +/- 14.802, SNR: 1.8, T: 1.00:  52%|#####2    | 261/500 [00:01<00:01, 233.62it/s]\n",
      "01:34:05.332 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -31.792 +/- 14.748, SNR: 1.8, T: 1.00:  57%|#####6    | 285/500 [00:01<00:00, 217.20it/s]\n",
      "01:34:05.452 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -31.689 +/- 14.699, SNR: 1.8, T: 1.00:  61%|######1   | 307/500 [00:01<00:00, 207.14it/s]\n",
      "01:34:05.576 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -31.591 +/- 14.653, SNR: 1.8, T: 1.00:  66%|######5   | 328/500 [00:01<00:00, 195.70it/s]\n",
      "01:34:05.680 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -31.501 +/- 14.609, SNR: 1.8, T: 1.00:  70%|######9   | 348/500 [00:01<00:00, 194.71it/s]\n",
      "01:34:05.782 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -31.404 +/- 14.564, SNR: 1.8, T: 1.00:  74%|#######3  | 369/500 [00:01<00:00, 197.38it/s]\n",
      "01:34:05.918 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -31.315 +/- 14.521, SNR: 1.7, T: 1.00:  78%|#######7  | 389/500 [00:01<00:00, 179.99it/s]\n",
      "01:34:06.097 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -31.231 +/- 14.480, SNR: 1.7, T: 1.00:  82%|########1 | 408/500 [00:01<00:00, 151.07it/s]\n",
      "01:34:06.197 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -31.152 +/- 14.442, SNR: 1.7, T: 1.00:  85%|########5 | 426/500 [00:02<00:00, 157.91it/s]\n",
      "01:34:06.301 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -31.051 +/- 14.395, SNR: 1.7, T: 1.00:  90%|########9 | 449/500 [00:02<00:00, 174.66it/s]\n",
      "01:34:06.405 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -30.961 +/- 14.351, SNR: 1.7, T: 1.00:  94%|#########3| 470/500 [00:02<00:00, 182.37it/s]\n",
      "01:34:06.520 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -30.881 +/- 14.312, SNR: 1.7, T: 1.00:  98%|#########7| 489/500 [00:02<00:00, 177.20it/s]\n",
      "01:34:06.564 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 5) ELBO: -30.836 +/- 14.290, SNR: 1.7, T: 1.00: 100%|##########| 500/500 [00:02<00:00, 210.41it/s]\n",
      "01:34:06.565 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 5):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:06.775 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 5) relative error: 0.0337 +/- 0.1125:   1%|1         | 1/100 [00:00<00:20,  4.77it/s]\n",
      "01:34:06.880 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 5) relative error: 0.0011 +/- 0.0043:   2%|2         | 2/100 [00:00<00:14,  6.76it/s]\n",
      "01:34:07.040 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 5) relative error: 0.0002 +/- 0.0030:   3%|3         | 3/100 [00:00<00:15,  6.32it/s]\n",
      "01:34:07.040 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 5):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:07.041 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 5) ploidy update size: 0.0296127: 100%|##########| 1/1 [00:00<00:00, 4894.17it/s]\n",
      "01:34:07.041 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:34:07.143 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -30.763 +/- 14.253, SNR: 1.6, T: 1.00:   4%|3         | 18/500 [00:00<00:02, 176.93it/s]\n",
      "01:34:07.247 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -30.682 +/- 14.213, SNR: 1.6, T: 1.00:   8%|7         | 38/500 [00:00<00:02, 186.40it/s]\n",
      "01:34:07.348 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -30.570 +/- 14.157, SNR: 1.6, T: 1.00:  13%|#3        | 66/500 [00:00<00:01, 227.43it/s]\n",
      "01:34:07.457 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -30.480 +/- 14.112, SNR: 1.6, T: 1.00:  18%|#7        | 89/500 [00:00<00:01, 220.72it/s]\n",
      "01:34:07.558 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -30.391 +/- 14.067, SNR: 1.6, T: 1.00:  22%|##2       | 112/500 [00:00<00:01, 223.24it/s]\n",
      "01:34:07.678 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -30.302 +/- 14.022, SNR: 1.6, T: 1.00:  27%|##7       | 135/500 [00:00<00:01, 211.08it/s]\n",
      "01:34:07.788 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -30.219 +/- 13.980, SNR: 1.5, T: 1.00:  31%|###1      | 157/500 [00:00<00:01, 207.66it/s]\n",
      "01:34:07.898 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -30.140 +/- 13.940, SNR: 1.5, T: 1.00:  36%|###5      | 178/500 [00:00<00:01, 202.50it/s]\n",
      "01:34:08.018 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -30.062 +/- 13.901, SNR: 1.5, T: 1.00:  40%|###9      | 199/500 [00:00<00:01, 193.30it/s]\n",
      "01:34:08.123 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.981 +/- 13.860, SNR: 1.5, T: 1.00:  44%|####4     | 221/500 [00:01<00:01, 197.75it/s]\n",
      "01:34:08.228 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.897 +/- 13.818, SNR: 1.5, T: 1.00:  49%|####8     | 244/500 [00:01<00:01, 204.06it/s]\n",
      "01:34:08.330 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.818 +/- 13.777, SNR: 1.5, T: 1.00:  53%|#####3    | 266/500 [00:01<00:01, 207.55it/s]\n",
      "01:34:08.432 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.740 +/- 13.738, SNR: 1.5, T: 1.00:  58%|#####7    | 288/500 [00:01<00:01, 210.22it/s]\n",
      "01:34:08.541 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.662 +/- 13.698, SNR: 1.5, T: 1.00:  62%|######2   | 310/500 [00:01<00:00, 207.43it/s]\n",
      "01:34:08.643 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.584 +/- 13.660, SNR: 1.4, T: 1.00:  66%|######6   | 332/500 [00:01<00:00, 209.99it/s]\n",
      "01:34:08.744 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.504 +/- 13.619, SNR: 1.4, T: 1.00:  71%|#######1  | 355/500 [00:01<00:00, 215.05it/s]\n",
      "01:34:08.852 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.428 +/- 13.581, SNR: 1.4, T: 1.00:  75%|#######5  | 377/500 [00:01<00:00, 211.35it/s]\n",
      "01:34:08.958 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.354 +/- 13.543, SNR: 1.4, T: 1.00:  80%|#######9  | 399/500 [00:01<00:00, 210.39it/s]\n",
      "01:34:09.064 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.280 +/- 13.505, SNR: 1.4, T: 1.00:  84%|########4 | 421/500 [00:02<00:00, 209.27it/s]\n",
      "01:34:09.167 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.200 +/- 13.465, SNR: 1.4, T: 1.00:  89%|########9 | 445/500 [00:02<00:00, 216.57it/s]\n",
      "01:34:09.292 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.127 +/- 13.428, SNR: 1.4, T: 1.00:  93%|#########3| 467/500 [00:02<00:00, 202.59it/s]\n",
      "01:34:09.394 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.058 +/- 13.393, SNR: 1.4, T: 1.00:  98%|#########7| 488/500 [00:02<00:00, 203.73it/s]\n",
      "01:34:09.440 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 6) ELBO: -29.019 +/- 13.374, SNR: 1.3, T: 1.00: 100%|##########| 500/500 [00:02<00:00, 208.40it/s]\n",
      "01:34:09.441 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 6):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:09.601 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 6) relative error: 0.0145 +/- 0.0408:   1%|1         | 1/100 [00:00<00:15,  6.26it/s]\n",
      "01:34:09.758 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 6) relative error: 0.0011 +/- 0.0067:   2%|2         | 2/100 [00:00<00:15,  6.33it/s]\n",
      "01:34:09.908 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 6) relative error: 0.0005 +/- 0.0012:   4%|4         | 4/100 [00:00<00:10,  9.42it/s]\n",
      "01:34:10.043 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 6) relative error: 0.0003 +/- 0.0021:   5%|5         | 5/100 [00:00<00:11,  8.30it/s]\n",
      "01:34:10.044 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 6):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:10.045 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 6) ploidy update size: 0.0151366: 100%|##########| 1/1 [00:00<00:00, 3017.48it/s]\n",
      "01:34:10.045 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:10.149 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.952 +/- 13.339, SNR: 1.3, T: 1.00:   4%|4         | 21/500 [00:00<00:02, 202.63it/s]\n",
      "01:34:10.250 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.880 +/- 13.301, SNR: 1.3, T: 1.00:   9%|8         | 44/500 [00:00<00:02, 218.02it/s]\n",
      "01:34:10.350 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.812 +/- 13.266, SNR: 1.3, T: 1.00:  13%|#3        | 66/500 [00:00<00:01, 218.55it/s]\n",
      "01:34:10.469 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.743 +/- 13.231, SNR: 1.3, T: 1.00:  18%|#7        | 88/500 [00:00<00:02, 204.10it/s]\n",
      "01:34:10.580 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.679 +/- 13.197, SNR: 1.3, T: 1.00:  22%|##1       | 109/500 [00:00<00:01, 198.93it/s]\n",
      "01:34:10.717 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.617 +/- 13.166, SNR: 1.3, T: 1.00:  26%|##5       | 129/500 [00:00<00:02, 177.86it/s]\n",
      "01:34:10.843 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.560 +/- 13.136, SNR: 1.3, T: 1.00:  30%|##9       | 148/500 [00:00<00:02, 168.82it/s]\n",
      "01:34:10.944 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.503 +/- 13.107, SNR: 1.3, T: 1.00:  33%|###3      | 167/500 [00:00<00:01, 173.98it/s]\n",
      "01:34:11.049 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.444 +/- 13.076, SNR: 1.3, T: 1.00:  37%|###7      | 187/500 [00:01<00:01, 178.67it/s]\n",
      "01:34:11.150 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.382 +/- 13.044, SNR: 1.2, T: 1.00:  42%|####1     | 208/500 [00:01<00:01, 187.45it/s]\n",
      "01:34:11.250 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.324 +/- 13.013, SNR: 1.2, T: 1.00:  46%|####5     | 228/500 [00:01<00:01, 190.75it/s]\n",
      "01:34:11.361 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.267 +/- 12.983, SNR: 1.2, T: 1.00:  50%|####9     | 248/500 [00:01<00:01, 187.78it/s]\n",
      "01:34:11.466 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.212 +/- 12.955, SNR: 1.2, T: 1.00:  53%|#####3    | 267/500 [00:01<00:01, 185.72it/s]\n",
      "01:34:11.568 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.155 +/- 12.925, SNR: 1.2, T: 1.00:  57%|#####7    | 287/500 [00:01<00:01, 188.74it/s]\n",
      "01:34:11.669 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.099 +/- 12.896, SNR: 1.2, T: 1.00:  61%|######1   | 307/500 [00:01<00:01, 191.19it/s]\n",
      "01:34:11.796 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -28.044 +/- 12.866, SNR: 1.2, T: 1.00:  65%|######5   | 327/500 [00:01<00:00, 179.48it/s]\n",
      "01:34:11.897 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -27.989 +/- 12.837, SNR: 1.2, T: 1.00:  69%|######9   | 347/500 [00:01<00:00, 184.87it/s]\n",
      "01:34:12.002 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -27.937 +/- 12.810, SNR: 1.2, T: 1.00:  73%|#######3  | 366/500 [00:01<00:00, 183.69it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:34:12.106 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -27.886 +/- 12.782, SNR: 1.2, T: 1.00:  77%|#######7  | 385/500 [00:02<00:00, 183.60it/s]\n",
      "01:34:12.207 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -27.830 +/- 12.752, SNR: 1.2, T: 1.00:  81%|########1 | 406/500 [00:02<00:00, 190.73it/s]\n",
      "01:34:12.345 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -27.776 +/- 12.724, SNR: 1.2, T: 1.00:  85%|########5 | 426/500 [00:02<00:00, 174.21it/s]\n",
      "01:34:12.453 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -27.728 +/- 12.699, SNR: 1.1, T: 1.00:  89%|########8 | 444/500 [00:02<00:00, 171.89it/s]\n",
      "01:34:12.598 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -27.681 +/- 12.674, SNR: 1.1, T: 1.00:  92%|#########2| 462/500 [00:02<00:00, 154.78it/s]\n",
      "01:34:12.725 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -27.639 +/- 12.652, SNR: 1.1, T: 1.00:  96%|#########5| 478/500 [00:02<00:00, 146.05it/s]\n",
      "01:34:12.843 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -27.601 +/- 12.631, SNR: 1.1, T: 1.00:  99%|#########8| 493/500 [00:02<00:00, 140.66it/s]\n",
      "01:34:12.931 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 7) ELBO: -27.582 +/- 12.621, SNR: 1.1, T: 1.00: 100%|##########| 500/500 [00:02<00:00, 173.32it/s]\n",
      "01:34:12.934 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 7):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:13.199 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 7) relative error: 0.0119 +/- 0.0228:   1%|1         | 1/100 [00:00<00:26,  3.80it/s]\n",
      "01:34:13.349 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 7) relative error: 0.0006 +/- 0.0031:   3%|3         | 3/100 [00:00<00:11,  8.09it/s]\n",
      "01:34:13.511 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 7) relative error: 0.0003 +/- 0.0020:   4%|4         | 4/100 [00:00<00:13,  6.95it/s]\n",
      "01:34:13.511 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 7):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:13.512 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 7) ploidy update size: 0.00753085: 100%|##########| 1/1 [00:00<00:00, 5729.92it/s]\n",
      "01:34:13.512 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:13.619 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -27.524 +/- 12.590, SNR: 1.1, T: 1.00:   5%|4         | 23/500 [00:00<00:02, 216.66it/s]\n",
      "01:34:13.724 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -27.463 +/- 12.557, SNR: 1.1, T: 1.00:   9%|9         | 47/500 [00:00<00:02, 222.46it/s]\n",
      "01:34:13.894 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -27.405 +/- 12.526, SNR: 1.1, T: 1.00:  14%|#4        | 70/500 [00:00<00:02, 172.53it/s]\n",
      "01:34:14.000 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -27.358 +/- 12.501, SNR: 1.1, T: 1.00:  18%|#7        | 89/500 [00:00<00:02, 175.08it/s]\n",
      "01:34:14.110 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -27.311 +/- 12.476, SNR: 1.1, T: 1.00:  22%|##1       | 108/500 [00:00<00:02, 174.29it/s]\n",
      "01:34:14.213 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -27.267 +/- 12.452, SNR: 1.1, T: 1.00:  25%|##5       | 126/500 [00:00<00:02, 174.19it/s]\n",
      "01:34:14.316 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -27.223 +/- 12.428, SNR: 1.1, T: 1.00:  29%|##8       | 144/500 [00:00<00:02, 174.46it/s]\n",
      "01:34:14.424 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -27.179 +/- 12.405, SNR: 1.1, T: 1.00:  32%|###2      | 162/500 [00:00<00:01, 172.04it/s]\n",
      "01:34:14.528 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -27.127 +/- 12.376, SNR: 1.1, T: 1.00:  37%|###6      | 184/500 [00:01<00:01, 183.77it/s]\n",
      "01:34:14.634 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -27.081 +/- 12.352, SNR: 1.0, T: 1.00:  41%|####      | 203/500 [00:01<00:01, 182.72it/s]\n",
      "01:34:14.739 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -27.030 +/- 12.324, SNR: 1.0, T: 1.00:  45%|####5     | 225/500 [00:01<00:01, 190.90it/s]\n",
      "01:34:14.840 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.974 +/- 12.293, SNR: 1.0, T: 1.00:  50%|####9     | 249/500 [00:01<00:01, 204.40it/s]\n",
      "01:34:14.997 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.925 +/- 12.267, SNR: 1.0, T: 1.00:  54%|#####4    | 270/500 [00:01<00:01, 176.43it/s]\n",
      "01:34:15.137 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.880 +/- 12.243, SNR: 1.0, T: 1.00:  58%|#####7    | 289/500 [00:01<00:01, 162.71it/s]\n",
      "01:34:15.261 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.842 +/- 12.222, SNR: 1.0, T: 1.00:  61%|######1   | 306/500 [00:01<00:01, 155.29it/s]\n",
      "01:34:15.364 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.799 +/- 12.198, SNR: 1.0, T: 1.00:  65%|######5   | 325/500 [00:01<00:01, 162.72it/s]\n",
      "01:34:15.473 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.756 +/- 12.175, SNR: 1.0, T: 1.00:  69%|######8   | 344/500 [00:01<00:00, 166.27it/s]\n",
      "01:34:15.580 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.716 +/- 12.153, SNR: 1.0, T: 1.00:  72%|#######2  | 362/500 [00:02<00:00, 166.63it/s]\n",
      "01:34:15.691 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.679 +/- 12.132, SNR: 1.0, T: 1.00:  76%|#######5  | 379/500 [00:02<00:00, 162.77it/s]\n",
      "01:34:15.799 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.641 +/- 12.112, SNR: 1.0, T: 1.00:  79%|#######9  | 396/500 [00:02<00:00, 161.12it/s]\n",
      "01:34:15.963 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.603 +/- 12.091, SNR: 1.0, T: 1.00:  83%|########2 | 413/500 [00:02<00:00, 141.62it/s]\n",
      "01:34:16.139 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.571 +/- 12.073, SNR: 1.0, T: 1.00:  86%|########5 | 428/500 [00:02<00:00, 118.37it/s]\n",
      "01:34:16.242 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.538 +/- 12.055, SNR: 1.0, T: 1.00:  89%|########8 | 443/500 [00:02<00:00, 124.65it/s]\n",
      "01:34:16.343 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.492 +/- 12.031, SNR: 1.0, T: 1.00:  93%|#########2| 464/500 [00:02<00:00, 145.35it/s]\n",
      "01:34:16.444 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.456 +/- 12.010, SNR: 1.0, T: 1.00:  96%|#########6| 481/500 [00:02<00:00, 151.65it/s]\n",
      "01:34:16.546 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.421 +/- 11.992, SNR: 1.0, T: 1.00:  99%|#########9| 497/500 [00:03<00:00, 153.17it/s]\n",
      "01:34:16.561 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 8) ELBO: -26.415 +/- 11.988, SNR: 1.0, T: 1.00: 100%|##########| 500/500 [00:03<00:00, 163.98it/s]\n",
      "01:34:16.562 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 8):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:16.736 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 8) relative error: 0.0209 +/- 0.0255:   1%|1         | 1/100 [00:00<00:17,  5.76it/s]\n",
      "01:34:16.868 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 8) relative error: 0.0013 +/- 0.0054:   2%|2         | 2/100 [00:00<00:14,  6.70it/s]\n",
      "01:34:17.021 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 8) relative error: 0.0004 +/- 0.0023:   3%|3         | 3/100 [00:00<00:14,  6.54it/s]\n",
      "01:34:17.021 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 8):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:17.021 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 8) ploidy update size: 0.00361526: 100%|##########| 1/1 [00:00<00:00, 4100.00it/s]\n",
      "01:34:17.022 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:17.127 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -26.375 +/- 11.966, SNR: 0.9, T: 1.00:   4%|3         | 19/500 [00:00<00:02, 180.97it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:34:17.227 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -26.333 +/- 11.943, SNR: 0.9, T: 1.00:   8%|7         | 39/500 [00:00<00:02, 191.69it/s]\n",
      "01:34:17.327 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -26.288 +/- 11.918, SNR: 0.9, T: 1.00:  12%|#2        | 61/500 [00:00<00:02, 204.42it/s]\n",
      "01:34:17.436 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -26.241 +/- 11.891, SNR: 0.9, T: 1.00:  17%|#6        | 84/500 [00:00<00:02, 207.54it/s]\n",
      "01:34:17.543 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -26.198 +/- 11.867, SNR: 0.9, T: 1.00:  21%|##1       | 105/500 [00:00<00:01, 203.33it/s]\n",
      "01:34:17.679 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -26.156 +/- 11.844, SNR: 0.9, T: 1.00:  25%|##5       | 126/500 [00:00<00:02, 183.92it/s]\n",
      "01:34:17.881 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -26.118 +/- 11.822, SNR: 0.9, T: 1.00:  29%|##9       | 145/500 [00:00<00:02, 142.70it/s]\n",
      "01:34:18.025 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -26.086 +/- 11.804, SNR: 0.9, T: 1.00:  32%|###2      | 161/500 [00:01<00:02, 132.49it/s]\n",
      "01:34:18.129 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -26.054 +/- 11.787, SNR: 0.9, T: 1.00:  35%|###5      | 177/500 [00:01<00:02, 137.72it/s]\n",
      "01:34:18.230 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -26.023 +/- 11.769, SNR: 0.9, T: 1.00:  39%|###8      | 193/500 [00:01<00:02, 143.06it/s]\n",
      "01:34:18.334 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.982 +/- 11.746, SNR: 0.9, T: 1.00:  43%|####2     | 214/500 [00:01<00:01, 158.96it/s]\n",
      "01:34:18.438 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.945 +/- 11.725, SNR: 0.9, T: 1.00:  47%|####6     | 233/500 [00:01<00:01, 165.96it/s]\n",
      "01:34:18.576 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.910 +/- 11.706, SNR: 0.9, T: 1.00:  50%|#####     | 251/500 [00:01<00:01, 153.73it/s]\n",
      "01:34:18.682 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.879 +/- 11.688, SNR: 0.9, T: 1.00:  53%|#####3    | 267/500 [00:01<00:01, 152.91it/s]\n",
      "01:34:18.786 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.846 +/- 11.670, SNR: 0.9, T: 1.00:  57%|#####6    | 284/500 [00:01<00:01, 155.71it/s]\n",
      "01:34:18.891 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.812 +/- 11.651, SNR: 0.9, T: 1.00:  60%|######    | 302/500 [00:01<00:01, 160.20it/s]\n",
      "01:34:19.008 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.780 +/- 11.632, SNR: 0.9, T: 1.00:  64%|######3   | 319/500 [00:01<00:01, 155.55it/s]\n",
      "01:34:19.110 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.751 +/- 11.615, SNR: 0.9, T: 1.00:  67%|######7   | 335/500 [00:02<00:01, 155.82it/s]\n",
      "01:34:19.217 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.721 +/- 11.599, SNR: 0.9, T: 1.00:  70%|#######   | 351/500 [00:02<00:00, 154.05it/s]\n",
      "01:34:19.320 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.689 +/- 11.581, SNR: 0.9, T: 1.00:  74%|#######3  | 368/500 [00:02<00:00, 157.32it/s]\n",
      "01:34:19.433 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.660 +/- 11.564, SNR: 0.8, T: 1.00:  77%|#######6  | 384/500 [00:02<00:00, 152.35it/s]\n",
      "01:34:19.542 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.630 +/- 11.547, SNR: 0.8, T: 1.00:  80%|########  | 400/500 [00:02<00:00, 150.45it/s]\n",
      "01:34:19.647 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.601 +/- 11.531, SNR: 0.8, T: 1.00:  83%|########3 | 416/500 [00:02<00:00, 151.10it/s]\n",
      "01:34:19.754 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.572 +/- 11.514, SNR: 0.8, T: 1.00:  86%|########6 | 432/500 [00:02<00:00, 150.52it/s]\n",
      "01:34:19.862 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.540 +/- 11.496, SNR: 0.8, T: 1.00:  90%|######### | 450/500 [00:02<00:00, 155.67it/s]\n",
      "01:34:19.962 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.511 +/- 11.479, SNR: 0.8, T: 1.00:  93%|#########3| 466/500 [00:02<00:00, 156.89it/s]\n",
      "01:34:20.067 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.481 +/- 11.462, SNR: 0.8, T: 1.00:  97%|#########6| 483/500 [00:03<00:00, 158.29it/s]\n",
      "01:34:20.168 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.452 +/- 11.446, SNR: 0.8, T: 1.00: 100%|#########9| 499/500 [00:03<00:00, 158.45it/s]\n",
      "01:34:20.172 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 9) ELBO: -25.451 +/- 11.445, SNR: 0.8, T: 1.00: 100%|##########| 500/500 [00:03<00:00, 158.73it/s]\n",
      "01:34:20.172 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 9):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:20.342 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 9) relative error: 0.0154 +/- 0.0185:   1%|1         | 1/100 [00:00<00:16,  5.91it/s]\n",
      "01:34:20.462 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 9) relative error: 0.0015 +/- 0.0049:   2%|2         | 2/100 [00:00<00:13,  7.11it/s]\n",
      "01:34:20.547 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 9) relative error: 0.0003 +/- 0.0031:   2%|2         | 2/100 [00:00<00:18,  5.34it/s]\n",
      "01:34:20.547 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 9):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:20.548 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 9) ploidy update size: 0.00142852: 100%|##########| 1/1 [00:00<00:00, 1700.85it/s]\n",
      "01:34:20.549 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:20.650 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 10) ELBO: -21.295 +/- 4.942, SNR: 1.1, T: 1.00:  14%|#3        | 69/500 [00:00<00:00, 686.45it/s]\n",
      "01:34:20.751 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 10) ELBO: -19.724 +/- 2.511, SNR: 1.5, T: 1.00:  28%|##8       | 142/500 [00:00<00:00, 709.71it/s]\n",
      "01:34:20.857 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 10) ELBO: -18.993 +/- 1.940, SNR: 1.6, T: 1.00:  43%|####2     | 213/500 [00:00<00:00, 691.38it/s]\n",
      "01:34:20.957 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 10) ELBO: -18.395 +/- 1.391, SNR: 1.8, T: 1.00:  59%|#####8    | 294/500 [00:00<00:00, 736.69it/s]\n",
      "01:34:21.116 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 10) ELBO: -18.037 +/- 1.128, SNR: 1.9, T: 1.00:  74%|#######3  | 368/500 [00:00<00:00, 609.83it/s]\n",
      "01:34:21.259 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 10) ELBO: -17.806 +/- 0.999, SNR: 1.9, T: 1.00:  86%|########6 | 432/500 [00:00<00:00, 547.78it/s]\n",
      "01:34:21.353 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 10) ELBO: -17.565 +/- 0.815, SNR: 2.0, T: 1.00: 100%|##########| 500/500 [00:00<00:00, 622.20it/s]\n",
      "01:34:21.354 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 10):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:21.504 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 10) relative error: 0.0010 +/- 0.0047:   2%|2         | 2/100 [00:00<00:07, 13.30it/s]\n",
      "01:34:21.577 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 10) relative error: 0.0004 +/- 0.0024:   2%|2         | 2/100 [00:00<00:10,  8.95it/s]\n",
      "01:34:21.577 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 10):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:21.578 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 10) ploidy update size: 0.00118774: 100%|##########| 1/1 [00:00<00:00, 1937.32it/s]\n",
      "01:34:21.578 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:21.689 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 11) ELBO: -17.472 +/- 0.702, SNR: 2.2, T: 1.00:   6%|6         | 31/500 [00:00<00:01, 281.80it/s]\n",
      "01:34:21.790 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 11) ELBO: -17.331 +/- 0.611, SNR: 2.4, T: 1.00:  20%|#9        | 99/500 [00:00<00:00, 504.42it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:34:21.890 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 11) ELBO: -17.211 +/- 0.560, SNR: 2.4, T: 1.00:  34%|###3      | 168/500 [00:00<00:00, 585.82it/s]\n",
      "01:34:21.990 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 11) ELBO: -17.087 +/- 0.481, SNR: 2.5, T: 1.00:  47%|####7     | 237/500 [00:00<00:00, 625.40it/s]\n",
      "01:34:22.091 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 11) ELBO: -17.006 +/- 0.454, SNR: 2.5, T: 1.00:  62%|######1   | 308/500 [00:00<00:00, 655.12it/s]\n",
      "01:34:22.191 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 11) ELBO: -16.898 +/- 0.411, SNR: 2.5, T: 1.00:  78%|#######8  | 392/500 [00:00<00:00, 716.25it/s]\n",
      "01:34:22.292 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 11) ELBO: -16.784 +/- 0.356, SNR: 2.6, T: 1.00:  96%|#########5| 478/500 [00:00<00:00, 760.35it/s]\n",
      "01:34:22.316 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 11) ELBO: -16.757 +/- 0.342, SNR: 2.7, T: 1.00: 100%|##########| 500/500 [00:00<00:00, 677.64it/s]\n",
      "01:34:22.317 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 11):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:22.418 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 11) relative error: 0.0107 +/- 0.0340:   1%|1         | 1/100 [00:00<00:09,  9.93it/s]\n",
      "01:34:22.605 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 11) relative error: 0.0002 +/- 0.0029:   2%|2         | 2/100 [00:00<00:14,  6.95it/s]\n",
      "01:34:22.606 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 11):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:22.606 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 11) ploidy update size: 0.000865213:   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:22.606 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:22.708 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 12) ELBO: -16.701 +/- 0.334, SNR: 2.6, T: 1.00:  14%|#4        | 70/500 [00:00<00:00, 691.07it/s]\n",
      "01:34:22.808 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 12) ELBO: -16.628 +/- 0.315, SNR: 2.5, T: 1.00:  31%|###1      | 155/500 [00:00<00:00, 783.69it/s]\n",
      "01:34:22.916 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 12) ELBO: -16.566 +/- 0.300, SNR: 2.5, T: 1.00:  47%|####6     | 234/500 [00:00<00:00, 760.17it/s]\n",
      "01:34:23.029 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 12) ELBO: -16.511 +/- 0.290, SNR: 2.4, T: 1.00:  62%|######2   | 311/500 [00:00<00:00, 726.92it/s]\n",
      "01:34:23.129 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 12) ELBO: -16.455 +/- 0.281, SNR: 2.3, T: 1.00:  77%|#######7  | 387/500 [00:00<00:00, 737.11it/s]\n",
      "01:34:23.238 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 12) ELBO: -16.401 +/- 0.270, SNR: 2.2, T: 1.00:  92%|#########2| 461/500 [00:00<00:00, 717.37it/s]\n",
      "01:34:23.298 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 12) ELBO: -16.372 +/- 0.263, SNR: 2.2, T: 1.00: 100%|##########| 500/500 [00:00<00:00, 723.65it/s]\n",
      "01:34:23.298 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 12):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:23.444 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 12) relative error: 0.0007 +/- 0.0101:   2%|2         | 2/100 [00:00<00:07, 13.70it/s]\n",
      "01:34:23.584 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 12) relative error: 0.0006 +/- 0.0023:   4%|4         | 4/100 [00:00<00:06, 14.07it/s]\n",
      "01:34:23.675 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 12) relative error: 0.0003 +/- 0.0018:   4%|4         | 4/100 [00:00<00:09, 10.63it/s]\n",
      "01:34:23.675 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 12):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:23.675 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 12) ploidy update size: 0.000631456:   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:23.676 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:23.797 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 13) ELBO: -16.334 +/- 0.256, SNR: 2.1, T: 1.00:  12%|#1        | 58/500 [00:00<00:00, 480.45it/s]\n",
      "01:34:23.932 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 13) ELBO: -16.303 +/- 0.250, SNR: 2.0, T: 1.00:  21%|##1       | 107/500 [00:00<00:00, 408.76it/s]\n",
      "01:34:24.031 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 13) ELBO: -16.261 +/- 0.244, SNR: 1.9, T: 1.00:  35%|###5      | 175/500 [00:00<00:00, 515.58it/s]\n",
      "01:34:24.133 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 13) ELBO: -16.218 +/- 0.236, SNR: 1.8, T: 1.00:  49%|####9     | 245/500 [00:00<00:00, 579.43it/s]\n",
      "01:34:24.239 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 13) ELBO: -16.184 +/- 0.229, SNR: 1.7, T: 1.00:  61%|######1   | 305/500 [00:00<00:00, 574.71it/s]\n",
      "01:34:24.339 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 13) ELBO: -16.139 +/- 0.220, SNR: 1.6, T: 1.00:  77%|#######7  | 386/500 [00:00<00:00, 649.85it/s]\n",
      "01:34:24.440 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 13) ELBO: -16.086 +/- 0.201, SNR: 1.5, T: 1.00:  94%|#########3| 469/500 [00:00<00:00, 705.64it/s]\n",
      "01:34:24.471 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 13) ELBO: -16.067 +/- 0.194, SNR: 1.5, T: 1.00: 100%|##########| 500/500 [00:00<00:00, 628.87it/s]\n",
      "01:34:24.471 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 13):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:24.606 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 13) relative error: 0.0018 +/- 0.0083:   2%|2         | 2/100 [00:00<00:06, 14.90it/s]\n",
      "01:34:24.751 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 13) relative error: 0.0004 +/- 0.0025:   3%|3         | 3/100 [00:00<00:09, 10.71it/s]\n",
      "01:34:24.752 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 13):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:24.752 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 13) ploidy update size: 0.000404586:   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:24.752 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:24.853 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 14) ELBO: -16.035 +/- 0.198, SNR: 1.3, T: 1.00:  13%|#3        | 66/500 [00:00<00:00, 658.50it/s]\n",
      "01:34:24.959 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 14) ELBO: -16.004 +/- 0.189, SNR: 1.2, T: 1.00:  26%|##6       | 132/500 [00:00<00:00, 636.96it/s]\n",
      "01:34:25.160 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 14) ELBO: -15.978 +/- 0.184, SNR: 1.1, T: 1.00:  39%|###9      | 196/500 [00:00<00:00, 439.83it/s]\n",
      "01:34:25.270 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 14) ELBO: -15.958 +/- 0.179, SNR: 1.0, T: 1.00:  52%|#####1    | 258/500 [00:00<00:00, 479.32it/s]\n",
      "01:34:25.418 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 14) ELBO: -15.943 +/- 0.175, SNR: 0.9, T: 1.00:  62%|######2   | 310/500 [00:00<00:00, 430.87it/s]\n",
      "01:34:25.518 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 14) ELBO: -15.926 +/- 0.170, SNR: 0.9, T: 1.00:  73%|#######2  | 363/500 [00:00<00:00, 457.09it/s]\n",
      "01:34:25.619 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 14) ELBO: -15.905 +/- 0.164, SNR: 0.8, T: 1.00:  86%|########5 | 428/500 [00:00<00:00, 510.15it/s]\n",
      "01:34:25.717 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 14) ELBO: -15.880 +/- 0.156, SNR: 0.7, T: 1.00: 100%|##########| 500/500 [00:00<00:00, 518.48it/s]\n",
      "01:34:25.717 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 14):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:25.872 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 14) relative error: 0.0021 +/- 0.0053:   2%|2         | 2/100 [00:00<00:07, 12.98it/s]\n",
      "01:34:25.954 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 14) relative error: 0.0001 +/- 0.0023:   2%|2         | 2/100 [00:00<00:11,  8.45it/s]\n",
      "01:34:25.955 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 14):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:25.956 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 14) ploidy update size: 7.60277e-05:   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:25.957 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:34:26.059 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 15) ELBO: -15.862 +/- 0.150, SNR: 0.6, T: 1.00:  16%|#5        | 79/500 [00:00<00:00, 780.75it/s]\n",
      "01:34:26.165 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 15) ELBO: -15.848 +/- 0.146, SNR: 0.5, T: 1.00:  32%|###1      | 158/500 [00:00<00:00, 756.44it/s]\n",
      "01:34:26.279 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 15) ELBO: -15.835 +/- 0.142, SNR: 0.4, T: 1.00:  47%|####6     | 234/500 [00:00<00:00, 714.32it/s]\n",
      "01:34:26.384 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 15) ELBO: -15.826 +/- 0.140, SNR: 0.4, T: 1.00:  61%|######1   | 306/500 [00:00<00:00, 704.93it/s]\n",
      "01:34:26.484 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 15) ELBO: -15.814 +/- 0.136, SNR: 0.3, T: 1.00:  77%|#######7  | 385/500 [00:00<00:00, 733.94it/s]\n",
      "01:34:26.584 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 15) ELBO: -15.803 +/- 0.133, SNR: 0.2, T: 1.00:  93%|#########2| 463/500 [00:00<00:00, 748.31it/s]\n",
      "01:34:26.626 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 15) ELBO: -15.799 +/- 0.132, SNR: 0.2, T: 1.00: 100%|##########| 500/500 [00:00<00:00, 747.65it/s]\n",
      "01:34:26.626 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 15):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:26.785 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 15) relative error: 0.0009 +/- 0.0091:   2%|2         | 2/100 [00:00<00:07, 12.65it/s]\n",
      "01:34:26.930 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 15) relative error: 0.0003 +/- 0.0012:   3%|3         | 3/100 [00:00<00:09,  9.89it/s]\n",
      "01:34:26.930 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 15):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:26.931 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 15) ploidy update size: 0.000562073:   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:26.931 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:27.034 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 16) ELBO: -15.795 +/- 0.131, SNR: 0.1, T: 1.00:  15%|#5        | 75/500 [00:00<00:00, 726.37it/s]\n",
      "01:34:27.157 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 16) ELBO: -15.793 +/- 0.130, SNR: 0.1, T: 1.00:  30%|##9       | 148/500 [00:00<00:00, 644.55it/s]\n",
      "01:34:27.257 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 16) ELBO: -15.789 +/- 0.128, SNR: 0.1, T: 1.00:  45%|####5     | 225/500 [00:00<00:00, 696.21it/s]\n",
      "01:34:27.361 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 16) ELBO: -15.787 +/- 0.128, SNR: 0.1, T: 1.00:  59%|#####9    | 296/500 [00:00<00:00, 691.88it/s]\n",
      "01:34:27.482 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 16) ELBO: -15.783 +/- 0.127, SNR: 0.1, T: 1.00:  73%|#######3  | 366/500 [00:00<00:00, 646.80it/s]\n",
      "01:34:27.583 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 16) ELBO: -15.780 +/- 0.126, SNR: 0.1, T: 1.00:  87%|########7 | 437/500 [00:00<00:00, 665.89it/s]\n",
      "01:34:27.666 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 16) ELBO: -15.777 +/- 0.125, SNR: 0.0, T: 1.00: 100%|##########| 500/500 [00:00<00:00, 680.37it/s]\n",
      "01:34:27.666 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 16):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:27.809 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 16) relative error: 0.0011 +/- 0.0074:   2%|2         | 2/100 [00:00<00:06, 14.05it/s]\n",
      "01:34:27.897 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 16) relative error: 0.0004 +/- 0.0032:   2%|2         | 2/100 [00:00<00:11,  8.67it/s]\n",
      "01:34:27.898 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 16):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:27.899 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 16) ploidy update size: 0.000519708:   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:27.899 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:28.000 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 17) ELBO: -15.778 +/- 0.125, SNR: 0.0, T: 1.00:  13%|#2        | 63/500 [00:00<00:00, 626.43it/s]\n",
      "01:34:28.102 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 17) ELBO: -15.776 +/- 0.124, SNR: 0.0, T: 1.00:  28%|##8       | 141/500 [00:00<00:00, 709.64it/s]\n",
      "01:34:28.204 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 17) ELBO: -15.775 +/- 0.123, SNR: 0.0, T: 1.00:  42%|####2     | 212/500 [00:00<00:00, 702.68it/s]\n",
      "01:34:28.308 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 17) ELBO: -15.772 +/- 0.127, SNR: 0.0, T: 1.00:  58%|#####8    | 292/500 [00:00<00:00, 728.34it/s]\n",
      "01:34:28.408 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 17) ELBO: -15.771 +/- 0.129, SNR: 0.0, T: 1.00:  75%|#######4  | 373/500 [00:00<00:00, 756.86it/s]\n",
      "01:34:28.509 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 17) ELBO: -15.771 +/- 0.129, SNR: 0.0, T: 1.00:  90%|######### | 451/500 [00:00<00:00, 763.09it/s]\n",
      "01:34:28.584 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 17) ELBO: -15.771 +/- 0.129, SNR: 0.0, T: 1.00: 100%|##########| 500/500 [00:00<00:00, 730.49it/s]\n",
      "01:34:28.584 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 17):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:28.740 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 17) relative error: 0.0016 +/- 0.0046:   2%|2         | 2/100 [00:00<00:07, 12.87it/s]\n",
      "01:34:28.883 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 17) relative error: 0.0004 +/- 0.0017:   3%|3         | 3/100 [00:00<00:09, 10.05it/s]\n",
      "01:34:28.883 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 17):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:28.888 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 17) ploidy update size: 0.00101179: 100%|##########| 1/1 [00:00<00:00, 223.74it/s]\n",
      "01:34:28.888 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:28.990 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 18) ELBO: -15.773 +/- 0.129, SNR: 0.0, T: 1.00:  11%|#1        | 55/500 [00:00<00:00, 540.28it/s]\n",
      "01:34:29.090 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 18) ELBO: -15.773 +/- 0.128, SNR: 0.0, T: 1.00:  24%|##3       | 119/500 [00:00<00:00, 598.21it/s]\n",
      "01:34:29.191 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 18) ELBO: -15.773 +/- 0.128, SNR: 0.0, T: 1.00:  37%|###7      | 186/500 [00:00<00:00, 627.88it/s]\n",
      "01:34:29.291 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 18) ELBO: -15.774 +/- 0.127, SNR: 0.0, T: 1.00:  53%|#####2    | 263/500 [00:00<00:00, 682.97it/s]\n",
      "01:34:29.391 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 18) ELBO: -15.774 +/- 0.127, SNR: 0.0, T: 1.00:  68%|######7   | 339/500 [00:00<00:00, 710.27it/s]\n",
      "01:34:29.498 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 18) ELBO: -15.774 +/- 0.129, SNR: 0.0, T: 1.00:  82%|########2 | 411/500 [00:00<00:00, 698.13it/s]\n",
      "01:34:29.598 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 18) ELBO: -15.773 +/- 0.129, SNR: 0.0, T: 1.00:  98%|#########7| 489/500 [00:00<00:00, 723.38it/s]\n",
      "01:34:29.631 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 18) ELBO: -15.773 +/- 0.129, SNR: 0.0, T: 1.00: 100%|##########| 500/500 [00:00<00:00, 673.10it/s]\n",
      "01:34:29.631 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 18):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:29.759 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 18) relative error: 0.0011 +/- 0.0056:   2%|2         | 2/100 [00:00<00:06, 15.64it/s]\n",
      "01:34:29.910 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 18) relative error: 0.0004 +/- 0.0019:   3%|3         | 3/100 [00:00<00:09, 10.76it/s]\n",
      "01:34:29.911 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 18):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:29.911 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 18) ploidy update size: 0.000801414:   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:29.911 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01:34:30.014 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 19) ELBO: -15.773 +/- 0.128, SNR: 0.0, T: 1.00:  13%|#3        | 67/500 [00:00<00:00, 655.80it/s]\n",
      "01:34:30.128 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 19) ELBO: -15.775 +/- 0.128, SNR: 0.0, T: 1.00:  27%|##6       | 133/500 [00:00<00:00, 606.40it/s]\n",
      "01:34:30.229 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 19) ELBO: -15.775 +/- 0.129, SNR: 0.0, T: 1.00:  41%|####1     | 205/500 [00:00<00:00, 654.24it/s]\n",
      "01:34:30.331 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 19) ELBO: -15.775 +/- 0.128, SNR: 0.0, T: 1.00:  61%|######1   | 306/500 [00:00<00:00, 785.08it/s]\n",
      "01:34:30.440 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 19) ELBO: -15.775 +/- 0.129, SNR: 0.0, T: 1.00:  77%|#######7  | 386/500 [00:00<00:00, 766.32it/s]\n",
      "01:34:30.541 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 19) ELBO: -15.774 +/- 0.129, SNR: 0.0, T: 1.00:  93%|#########2| 464/500 [00:00<00:00, 767.01it/s]\n",
      "01:34:30.612 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 19) ELBO: -15.774 +/- 0.129, SNR: 0.0, T: 1.00: 100%|##########| 500/500 [00:00<00:00, 713.64it/s]\n",
      "01:34:30.613 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 19):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:30.758 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 19) relative error: 0.0006 +/- 0.0081:   2%|2         | 2/100 [00:00<00:07, 13.76it/s]\n",
      "01:34:30.838 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 19) relative error: 0.0005 +/- 0.0020:   2%|2         | 2/100 [00:00<00:11,  8.86it/s]\n",
      "01:34:30.839 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 19):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:30.839 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 19) ploidy update size: 0.000193127:   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:30.839 INFO gcnvkernel.tasks.inference_task_base - (denoising) starting...:   0%|          | 0/500 [00:00<?, ?it/s]\n",
      "01:34:30.854 INFO gcnvkernel.tasks.inference_task_base - (denoising epoch 20) ELBO: -15.774 +/- 0.129, SNR: 0.0, T: 1.00:   2%|1         | 9/500 [00:00<00:00, 612.00it/s]\n",
      "01:34:30.854 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 20):   0%|          | 0/100 [00:00<?, ?it/s]\n",
      "01:34:31.001 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 20) relative error: 0.0014 +/- 0.0068:   2%|2         | 2/100 [00:00<00:07, 13.66it/s]\n",
      "01:34:31.084 INFO gcnvkernel.tasks.inference_task_base - (sampling epoch 20) relative error: 0.0004 +/- 0.0029:   2%|2         | 2/100 [00:00<00:11,  8.72it/s]\n",
      "01:34:31.084 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 20):   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:31.085 INFO gcnvkernel.tasks.inference_task_base - (ploidy calling epoch 20) ploidy update size: 0.000111239:   0%|          | 0/1 [00:00<?, ?it/s]\n",
      "01:34:31.085 INFO gcnvkernel.tasks.inference_task_base - Inference task completed successfully and convergence achieved.\n",
      "01:34:32.569 INFO gcnvkernel.io.io_ploidy - Writing mean_bias_j_lowerbound__...\n",
      "01:34:32.615 INFO gcnvkernel.io.io_ploidy - Writing psi_j_log__...\n",
      "01:34:32.642 INFO gcnvkernel.io.io_ploidy - Saving posteriors for sample \"NA18525\" in \"/home/jupyter/notebooks/germline-cnv/sandbox/ploidy-cohort7-calls/SAMPLE_0\"...\n",
      "01:34:32.656 INFO gcnvkernel.io.io_ploidy - Saving posteriors for sample \"NA18939\" in \"/home/jupyter/notebooks/germline-cnv/sandbox/ploidy-cohort7-calls/SAMPLE_1\"...\n",
      "01:34:32.668 INFO gcnvkernel.io.io_ploidy - Saving posteriors for sample \"NA19017\" in \"/home/jupyter/notebooks/germline-cnv/sandbox/ploidy-cohort7-calls/SAMPLE_2\"...\n",
      "01:34:32.680 INFO gcnvkernel.io.io_ploidy - Saving posteriors for sample \"NA19625\" in \"/home/jupyter/notebooks/germline-cnv/sandbox/ploidy-cohort7-calls/SAMPLE_3\"...\n",
      "01:34:32.692 INFO gcnvkernel.io.io_ploidy - Saving posteriors for sample \"NA19648\" in \"/home/jupyter/notebooks/germline-cnv/sandbox/ploidy-cohort7-calls/SAMPLE_4\"...\n",
      "01:34:32.706 INFO gcnvkernel.io.io_ploidy - Saving posteriors for sample \"NA20502\" in \"/home/jupyter/notebooks/germline-cnv/sandbox/ploidy-cohort7-calls/SAMPLE_5\"...\n",
      "01:34:32.717 INFO gcnvkernel.io.io_ploidy - Saving posteriors for sample \"NA20845\" in \"/home/jupyter/notebooks/germline-cnv/sandbox/ploidy-cohort7-calls/SAMPLE_6\"...\n",
      "01:34:33.262 DEBUG ScriptExecutor - Result: 0\n",
      "01:34:33.268 INFO  DetermineGermlineContigPloidy - DetermineGermlineContigPloidy complete.\n",
      "01:34:33.268 INFO  DetermineGermlineContigPloidy - Shutting down engine\n",
      "[January 20, 2024 at 1:34:33 AM GMT] org.broadinstitute.hellbender.tools.copynumber.DetermineGermlineContigPloidy done. Elapsed time: 7.01 minutes.\n",
      "Runtime.totalMemory()=548405248\n"
     ]
    }
   ],
   "source": [
    "# --max-training-epochs 20 --num-thermal-advi-iters 1000 --max-advi-iter-subsequent-epochs 500 args only used \n",
    "# for tutorial to decreaase training time. Not recommeneded for use with full size data.\n",
    "\n",
    "! gatk DetermineGermlineContigPloidy \\\n",
    "        -L $WORKSPACE_LOCAL/sandbox/chr20Y.cohort.gc.filtered.interval_list \\\n",
    "        --interval-merging-rule OVERLAPPING_ONLY \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18525.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18939.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19017.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19625.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19648.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20502.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20845.tsv \\\n",
    "        --contig-ploidy-priors $TUTORIAL_DATA_PATH/chr20XY_contig_ploidy_priors.tsv \\\n",
    "        --max-training-epochs 20 --num-thermal-advi-iters 1000 --max-advi-iter-subsequent-epochs 500 \\\n",
    "        --output $WORKSPACE_LOCAL/sandbox \\\n",
    "        --output-prefix ploidy-cohort7 \\\n",
    "        --verbosity DEBUG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This produces a `<output-prefix>-calls` directory and a `<output-prefix>-model` directory.\n",
    "\n",
    "The ploidy-calls directory contains a folder of data for each sample in the cohort including the contig ploidy calls. Each sample directory, e.g. `ploidy-calls/SAMPLE_0`, contains five files:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls $WORKSPACE_LOCAL/sandbox/ploidy-cohort7-calls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls $WORKSPACE_LOCAL/sandbox/ploidy-cohort7-calls/SAMPLE_0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# let's look at the ploidy calls for NA19017\n",
    "! cat $WORKSPACE_LOCAL/sandbox/ploidy-cohort7-calls/SAMPLE_2/sample_name.txt\n",
    "! cat $WORKSPACE_LOCAL/sandbox/ploidy-cohort7-calls/SAMPLE_2/contig_ploidy.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Description of output files:\n",
    "- `contig_ploidy.tsv` notes the ploidy and genotype quality (GQ) of the ploidy call for each contig.\n",
    "- `global_read_depth.tsv` notes an average depth value and an average ploidy across all the intervals of the sample.\n",
    "- `mu_psi_s_log__.tsv` captures the posterior mean for all of the modeled parameters.\n",
    "- `sample_name.txt` contains the readgroup sample (RG SM) name.\n",
    "- `std_psi_s_log__.tsv` captures the standard deviation for all of the modeled paramters."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls $WORKSPACE_LOCAL/sandbox/ploidy-cohort7-model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The ploidy-model directory contains aggregated model data for the cohort. This is the model to provide to a case-mode _DetermineGermlineContigPloidy_ analysis and to _GermlineCNVCaller_. The tutorial ploidy-model directory contains the eight files as follows.\n",
    "\n",
    "- `contig_ploidy_prior.tsv` is a copy of the ploidy priors given to the tool.\n",
    "- `gcnvkernel_version.json` notes the version of the kernel.\n",
    "- `interval_list.tsv` recapitulates the intervals used, e.g. the filtered intervals.\n",
    "- `mu_mean_bias_j_lowerbound__.tsv` is the estimated contig-level capture bias (mean).\n",
    "- `mu_psi_j_log__.tsv` is the estimated contig-level capture noise (mean).\n",
    "- `ploidy_config.json` configuration of the ploidy model\n",
    "- `std_mean_bias_j_lowerbound__.tsv` is the estimated contg-level capture bias (std).\n",
    "- `std_psi_j_log__.tsv` is the estimated contig-level capture noise (std).   \n",
    "\n",
    "Note: The PyMC3/Theano model automatically generates mu_ and std_ files and may append transformations it performs to the file name, e.g. log or lowerbound as we see above. These are likely of interest only to advanced users."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In preparation for the next section (i.e. running the tool in case mode), let us run _DetermineGermlineContigPloidy_ one more time, but assuming that we didn't have `NA19017` sample in the original cohort:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# --max-training-epochs 20 --num-thermal-advi-iters 1000 --max-advi-iter-subsequent-epochs 500 args only used \n",
    "# for tutorial to decrease training time. Not recommeneded for use with full size data.\n",
    "\n",
    "! gatk DetermineGermlineContigPloidy \\\n",
    "        -L $WORKSPACE_LOCAL/sandbox/chr20Y.cohort.gc.filtered.interval_list \\\n",
    "        --interval-merging-rule OVERLAPPING_ONLY \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18525.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18939.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19625.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19648.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20502.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20845.tsv \\\n",
    "        --contig-ploidy-priors $TUTORIAL_DATA_PATH/chr20XY_contig_ploidy_priors.tsv \\\n",
    "        --max-training-epochs 20 --num-thermal-advi-iters 1000 --max-advi-iter-subsequent-epochs 500 \\\n",
    "        --output $WORKSPACE_LOCAL/sandbox \\\n",
    "        --output-prefix ploidy-cohort6 \\\n",
    "        --verbosity DEBUG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## _DetermineGermlineContigPloidy_ in case mode\n",
    "\n",
    "The case mode calls contig ploidies for each sample against the ploidy model given by `--model`. The following command runs sample NA19017 against the 23-sample cohort model we built above:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# --max-training-epochs 20 --num-thermal-advi-iters 1000 --max-advi-iter-subsequent-epochs 500 args only used \n",
    "# for tutorial to decrease training time. Not recommeneded for use with full size data.\n",
    "\n",
    "! gatk DetermineGermlineContigPloidy \\\n",
    "        --model $WORKSPACE_LOCAL/sandbox/ploidy-cohort6-model \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19017.tsv \\\n",
    "        -O $WORKSPACE_LOCAL/sandbox \\\n",
    "        --output-prefix ploidy-cohort6-case \\\n",
    "        --max-training-epochs 20 --num-thermal-advi-iters 1000 --max-advi-iter-subsequent-epochs 500 \\\n",
    "        --verbosity DEBUG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This produces a ploidy-case-calls directory, which in turn contains a directory of sample data, SAMPLE_0. A list of the five resulting files is some paragraphs above:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls $WORKSPACE_LOCAL/sandbox/ploidy-cohort6-case-calls/SAMPLE_0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# let's look at the ploidy calls for the case sample (NA19017)\n",
    "! cat $WORKSPACE_LOCAL/sandbox/ploidy-cohort6-case-calls/SAMPLE_0/sample_name.txt\n",
    "! cat $WORKSPACE_LOCAL/sandbox/ploidy-cohort6-case-calls/SAMPLE_0/contig_ploidy.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Comments on select parameters of _DetermineGermlineContigPloidy_\n",
    "\n",
    "- It is possible to analyze multiple samples simultaneously in a case mode command. Provide each sample with `-I`.\n",
    "- For the `-L` intervals, supply the most processed intervals list. For the tutorial, this is the filtered intervals. Note the case mode does not require explicit intervals because the ploidy model provides them.\n",
    "- Provide a `--contig-ploidy-priors` table containing the per-contig prior probabilities for integer ploidy state. Again, the case mode does not require an explicit priors file as the ploidy model provides them. Tool index describes this resource in detail.\n",
    "- Optionally provide intervals to exclude from analysis with `--exclude-intervals` or `-XL`, e.g. [pseudoautosomal (PAR) regions](https://gatk.zendesk.com/hc/en-us/articles/360035891071), which can skew the results on sex chromosomes."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Call copy-number variants with _GermlineCNVCaller_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "_GermlineCNVCaller_ learns a denoising model per scattered shard while consistently calling CNVs across the shards. The tool models systematic biases and CNVs simultaneously, which allows for sensitive detection of both rare and common CNVs. As the tool index states under Important Remarks (v4.1.0.0), the tool should see data from a large enough genomic region so as to be exposed to diverse genomic features. The current recommendation is to provide at least ~10–50Mbp genomic coverage per scatter. This applies to exomes or WGS. This allows reliable inference of bias factors including GC bias. The limitation of analyzing larger regions is available memory. As an analysis covers more data, memory requirements increase.\n",
    "\n",
    "For expediency, the tutorial commands below analyze small data, specifically the 1400 bins in `twelveregions.cohort.gc.filtered.interval_list` and use default parameters. The tutorial splits the 1400 bins into two shards with 700 bins each to illustrate scattering. This results in ~0.7Mbp genomic coverage per shard. See section 5.2.3 for how to split interval lists by a given number of intervals. Default inference parameters are conservatively set for efficient run times."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## _GermlineCNVCaller_ in cohort mode\n",
    "\n",
    "This cell produces per-interval gCNV calls for each of the cohort samples and a gCNV model for the cohort. Each command produces three directories within `gcnv-cohort24-twelve`:\n",
    "- a `gcnv-cohort24-twelve-1of2-calls` folder of per sample gCNV call results,\n",
    "- a `gcnv-cohort24-twelve-1of2-model` folder of cohort model data,\n",
    "- a `gcnv-cohort24-twelve-1of2-tracking` folder of data that tracks model fitting.\n",
    "\n",
    "Note: Each shard should finish under 10 minutes in this Terra VM (4 vCPUs).  When running the default parameters of the v4.1.0.0 WDL cohort-mode workflow on the cloud, the majority of the shard analyses complete in half an hour."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**First shard (24 samples)**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! gatk GermlineCNVCaller \\\n",
    "        --run-mode COHORT \\\n",
    "        -L $TUTORIAL_DATA_PATH/scatter-sm/twelve_1of2.interval_list \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18525.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18939.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19017.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19625.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19648.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20502.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20845.tsv \\\n",
    "        --contig-ploidy-calls $WORKSPACE_LOCAL/sandbox/ploidy-cohort7-calls \\\n",
    "        --annotated-intervals $TUTORIAL_DATA_PATH/twelveregions.annotated.tsv \\\n",
    "        --interval-merging-rule OVERLAPPING_ONLY \\\n",
    "        --output $WORKSPACE_LOCAL/sandbox/gcnv-cohort7-twelve \\\n",
    "        --output-prefix gcnv-cohort7-twelve-1of2 \\\n",
    "        --verbosity DEBUG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Second shard (24 samples)**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! gatk GermlineCNVCaller \\\n",
    "        --run-mode COHORT \\\n",
    "        -L $TUTORIAL_DATA_PATH/scatter-sm/twelve_2of2.interval_list \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18525.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18939.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19017.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19625.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19648.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20502.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20845.tsv \\\n",
    "        --contig-ploidy-calls $WORKSPACE_LOCAL/sandbox/ploidy-cohort7-calls \\\n",
    "        --annotated-intervals $TUTORIAL_DATA_PATH/twelveregions.annotated.tsv \\\n",
    "        --interval-merging-rule OVERLAPPING_ONLY \\\n",
    "        --output $WORKSPACE_LOCAL/sandbox/gcnv-cohort7-twelve \\\n",
    "        --output-prefix gcnv-cohort7-twelve-2of2 \\\n",
    "        --verbosity DEBUG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# let's get acquainted with the output\n",
    "! ls $WORKSPACE_LOCAL/sandbox/gcnv-cohort7-twelve"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls $WORKSPACE_LOCAL/sandbox/gcnv-cohort7-twelve/gcnv-cohort7-twelve-1of2-model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**(advanced)** The model directory contains aggregated model data for the cohort and the analyzed interval shard. This is the model to provide to a case-mode _GermlineCNVCaller_. The model files are as follows:\n",
    "\n",
    "- `calling_config.json` is the configuration of the copy-number caller sub-model (the two-level HMM)\n",
    "- `denoising_config.json` is the configuration of the read-count noise model\n",
    "- `gcnvkernel_version.json` notes the version of the gcnvkernel\n",
    "- `interval_list.tsv` recapitulates the intervals used in this shard\n",
    "- `log_q_tau_tk.tsv` is the posterior probability of region class (rare vs. common) in log-scale\n",
    "- `mu_ard_u_log__.tsv` is ARD coefficient of the bias factors in log-scale (mean)\n",
    "- `mu_log_mean_bias_t.tsv` is the mean read-count capture bias of each of interval (mean) \n",
    "- `mu_psi_t_log__.tsv` is the read-count unexplained variance (overdispersion) of each of interval in log-scale (mean)\n",
    "- `mu_W_tu.tsv` is the interval x number of bias factors matrix of learned bias factors (mean)\n",
    "- `std_ard_u_log__.tsv` is ARD coefficient of the bias factors in log-scale (std)\n",
    "- `std_log_mean_bias_t.tsv` is the mean read-count capture bias of each of interval (std) \n",
    "- `std_psi_t_log__.tsv` is the read-count unexplained variance (overdispersion) of each of interval in log-scale (std)\n",
    "- `std_W_tu.tsv` is the interval x number of bias factors matrix of learned bias factors (std)   \n",
    "Note: The PyMC3/Theano model automatically generates mu_ and std_ prefixed files and may append transformations it performs to the file name (e.g. log) as we see above. These are likely of interest only to advanced users who may want to inspect the model parameters."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls $WORKSPACE_LOCAL/sandbox/gcnv-cohort7-twelve/gcnv-cohort7-twelve-1of2-tracking"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**(advanced)** Loss function trajectory during model training:\n",
    "\n",
    "- `warm_up_elbo_history.tsv` is evidence lower bound (ELBO) history during the initial model warm-up period\n",
    "- `main_elbo_history.tsv` is evidence lower bound (ELBO) history during the main model training period"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls $WORKSPACE_LOCAL/sandbox/gcnv-cohort7-twelve/gcnv-cohort7-twelve-1of2-calls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls $WORKSPACE_LOCAL/sandbox/gcnv-cohort7-twelve/gcnv-cohort7-twelve-1of2-calls/SAMPLE_0"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**(advanced)** The calls directory contains various sample-specific estimated quantities, including per-interval estimated copy-number states. These intermediate results are later post-processed by `PostprocessGermlineCNVCalls` to produce VCF files (see the next section). The per-sample call files are as follows:\n",
    "\n",
    "- `baseline_copy_number_t.tsv` is the per-interval baseline copy-number of the sample (recapitulated from DetermineGermlineContigPloidy)\n",
    "- `log_c_emission_tc.tsv` is the emission probability of each interval to different copy-number states in log-scale\n",
    "- `log_q_c_tc.tsv` is the per-interval posterior probability of each copy-number state in log-scale\n",
    "- `mu_denoised_copy_ratio_t.tsv` is the per-interval denoised copy-ratio\n",
    "- `mu_psi_s_log__.tsv` is the global unexplained variance (overdispersion) of this sample in log-scale (mean)\n",
    "- `mu_read_depth_s_log__.tsv` is the global read-depth of this sample in log-scale (mean)\n",
    "- `mu_z_sg.tsv` is loading of each GC bin (related to GC bias correction) (mean) \n",
    "- `mu_z_su.tsv` is the loading of each bias factor (mean)\n",
    "- `sample_name.txt` is the name of the sample\n",
    "- `std_denoised_copy_ratio_t.tsv` is the per-interval denoised copy-ratio\n",
    "- `std_psi_s_log__.tsv` is the global unexplained variance (overdispersion) of this sample in log-scale (std)\n",
    "- `std_read_depth_s_log__.tsv` is the global read-depth of this sample in log-scale (std)\n",
    "- `std_z_sg.tsv` is loading of each GC bin (related to GC bias correction) (std) \n",
    "- `std_z_su.tsv` is the loading of each bias factor (std)  \n",
    "Note: The PyMC3/Theano model automatically generates mu_ and std_ prefixed files and may append transformations it performs to the file name (e.g. log__) as we see above. These are likely of interest only to advanced users who may want to inspect the model parameters."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To test _GermlineCNVCaller_ in case mode, let us assume we didn't have sample `NA19017` in the initial cohort and let us build a cohort of 23 samples. We will then run _GermlineCNVCaller_ in case mode using the model learned from the 23-sample cohort."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**First shard (23 samples)** "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! gatk GermlineCNVCaller \\\n",
    "        --run-mode COHORT \\\n",
    "        -L $TUTORIAL_DATA_PATH/scatter-sm/twelve_1of2.interval_list \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18525.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18939.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19625.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19648.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20502.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20845.tsv \\\n",
    "        --contig-ploidy-calls $WORKSPACE_LOCAL/sandbox/ploidy-cohort6-calls \\\n",
    "        --annotated-intervals $TUTORIAL_DATA_PATH/twelveregions.annotated.tsv \\\n",
    "        --interval-merging-rule OVERLAPPING_ONLY \\\n",
    "        --output $WORKSPACE_LOCAL/sandbox/gcnv-cohort6-twelve \\\n",
    "        --output-prefix gcnv-cohort6-twelve-1of2 \\\n",
    "        --verbosity DEBUG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Second shard (23 samples)** "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! gatk GermlineCNVCaller \\\n",
    "        --run-mode COHORT \\\n",
    "        -L $TUTORIAL_DATA_PATH/scatter-sm/twelve_2of2.interval_list \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18525.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA18939.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19625.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19648.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20502.tsv \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA20845.tsv \\\n",
    "        --contig-ploidy-calls $WORKSPACE_LOCAL/sandbox/ploidy-cohort6-calls \\\n",
    "        --annotated-intervals $TUTORIAL_DATA_PATH/twelveregions.annotated.tsv \\\n",
    "        --interval-merging-rule OVERLAPPING_ONLY \\\n",
    "        --output $WORKSPACE_LOCAL/sandbox/gcnv-cohort6-twelve \\\n",
    "        --output-prefix gcnv-cohort6-twelve-2of2 \\\n",
    "        --verbosity DEBUG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## _GermlineCNVCaller_ in case mode\n",
    "\n",
    "Call gCNVs on a sample against a cohort model. The case analysis must use the same scatter approach as the model generation. So, as above, we run two shard analyses. Here, `--model` and `--output-prefix` differ between the scatter the commands."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**First shard (case-calling sample `NA19017` against the model built from the 23-sample cohort)** "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! gatk GermlineCNVCaller \\\n",
    "        --run-mode CASE \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19017.tsv \\\n",
    "        --contig-ploidy-calls $WORKSPACE_LOCAL/sandbox/ploidy-cohort6-case-calls \\\n",
    "        --model $WORKSPACE_LOCAL/sandbox/gcnv-cohort6-twelve/gcnv-cohort6-twelve-1of2-model \\\n",
    "        --output $WORKSPACE_LOCAL/sandbox/gcnv-case-twelve-vs-cohort6 \\\n",
    "        --output-prefix gcnv-case-twelve-vs-cohort6-1of2 \\\n",
    "        --verbosity DEBUG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Second shard (case-calling sample `NA19017` against the model built from the 6-sample cohort)** "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! gatk GermlineCNVCaller \\\n",
    "        --run-mode CASE \\\n",
    "        -I $TUTORIAL_DATA_PATH/cvg/NA19017.tsv \\\n",
    "        --contig-ploidy-calls $WORKSPACE_LOCAL/sandbox/ploidy-cohort6-case-calls \\\n",
    "        --model $WORKSPACE_LOCAL/sandbox/gcnv-cohort6-twelve/gcnv-cohort6-twelve-2of2-model \\\n",
    "        --output $WORKSPACE_LOCAL/sandbox/gcnv-case-twelve-vs-cohort6 \\\n",
    "        --output-prefix gcnv-case-twelve-vs-cohort6-2of2 \\\n",
    "        --verbosity DEBUG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# let's get acquainted with the output\n",
    "! ls $WORKSPACE_LOCAL/sandbox/gcnv-case-twelve-vs-cohort6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls $WORKSPACE_LOCAL/sandbox/gcnv-case-twelve-vs-cohort6/gcnv-case-twelve-vs-cohort6-1of2-calls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls $WORKSPACE_LOCAL/sandbox/gcnv-case-twelve-vs-cohort6/gcnv-case-twelve-vs-cohort6-1of2-calls/SAMPLE_0"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The call output in the case mode is similar to the cohort mode."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "At this point, the workflow has done its most heavy lifting to produce data towards copy number calling. In Section 5, we consolidate the data from the scattered GermlineCNVCaller runs, perform segmentation and call copy number states.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Comments on select parameters of _GermlineCNVCaller_\n",
    "\n",
    "- The `-O` output directory must be extant before running the command. Future releases (v4.1.1.0) will create the directory.\n",
    "- The default `--max-copy-number` is capped at 5. This means the tool reports any events with more copies as CN5.\n",
    "- For the cohort mode, optionally provide `--annotated-intervals` to include the annotations as covariates. These must contain all of the `-L` intervals. The `-L` intervals is an exact match or a subset of the annotated intervals.\n",
    "- For the case mode, the tool accepts only a single `--model` directory at a time. So the case must be analyzed with the same number of scatters as the cohort model run. The case mode parameters appear fewer than the cohort mode because the `--model` directory provides the seemingly missing requirements, i.e. the scatter intervals and the annotated intervals.\n",
    "- For both modes, provide the `--contig-ploidy-calls` results from _DetermineGermlineContigPloidy_ (Section 4).\n",
    "- `--verbosity DEBUG` allows tracking the Python `gcnvkernel` model fitting in the stdout, e.g. with information on denoising epochs and whether the model converged. The default INFO level verbosity is the next most verbose and emits only GATK Engine level messages."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### How do I increase the sensitivity of CNV detection?\n",
    "\n",
    "The tutorial uses default _GermlineCNVCaller_ modeling parameters. However, researchers should expect to tune parameters for data, e.g. from different sequencing technologies. For tuning, first consider the coherence length parameters, `p-alt`, `p-active` and the `psi-scale` parameters. These hyperparameters are just a few of the plethora of adjustable parameters _GermlineCNVCaller_ offers. Refer to the GermlineCNVCaller tool documentation for detailed explanations, and ask on the GATK Forum for further guidance.\n",
    "\n",
    "One set of parameter changes for WGS data that dramatically increase the sensitivity of calling on the tutorial data:\n",
    "```\n",
    "    --class-coherence-length 1000.0 \\\n",
    "    --cnv-coherence-length 1000.0 \\\n",
    "    --enable-bias-factors false \\\n",
    "    --interval-psi-scale 1.0E-6 \\\n",
    "    --log-mean-bias-standard-deviation 0.01 \\\n",
    "    --sample-psi-scale 1.0E-6 \\\n",
    "```\n",
    "\n",
    "Article #11687 and Notebook #11686 compare the results of using default vs. the increased-sensitivity parameters. Given the absence of off-the-shelf filtering solutions for CNV calls, when tuning parameters to increase sensitivity, researchers should expect to perform additional due diligence, especially for analyses requiring high precision calls.\n",
    "\n",
    "**Comments on select sensitivity parameters**\n",
    "\n",
    "- Decreasing `--class-coherence-length` from its default of `10,000bp` to `1000bp` decreases the expected length of contiguous segments of genomic region classes (silent vs. active). Factor for bin size when tuning.\n",
    "- Decreasing `--cnv-coherence-length` from its default `10,000bp` to `1000bp` decreases the expected length of per-sample CNV events. Factor for bin size when tuning.\n",
    "- Turning off `--enable-bias-factors` from the default true state to false turns off active discovery of learnable bias factors. This should always be on for targeted exome data and in general can be turned off for WGS data.\n",
    "- Decreasing `--interval-psi-scale` from its default of `0.001` to `1.0E-6` reduces the scale the tool considers normal in per-interval noise.\n",
    "- Decreasing `--log-mean-bias-standard-deviation` from its default of `0.1` to `0.01` reduces what is considered normal noise in capture bias.\n",
    "- Decreasing `--sample-psi-scale` from its default of `0.0001` to `1.0E-6` reduces the scale that is considered normal in sample-specific global read-count overdispersion.\n",
    "\n",
    "\n",
    "Additional parameters to consider include `--depth-correction-tau`, `--p-active` and `--p-alt`:\n",
    "\n",
    "- `--depth-correction-tau` has a default of `10000.0` (10K) and defines the precision of read-depth concordance with the prior estimate.\n",
    "- `--p-active` has a default of `1e-2` (0.01) and defines the prior probability of designating common CNV genomic regions.\n",
    "- `--p-alt` has a default of `1e-6` (0.000001) and defines the expected probability of CNV events (in silent genomic regions)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### How do I make interval lists for scattering (genomic sharding)?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This step applies to the cohort mode. It is unnecessary for case mode analyses as the model implies the scatter intervals.\n",
    "\n",
    "The v4.1.0.0 `cnv_germline_cohort_workflow.wdl` pipeline workflow scatters the _GermlineCNVCaller_ step. Each scattered analysis is on genomic intervals subset from intervals produced either from _PreprocessIntervals_ (section 2) or from _FilterIntervals_ (section 3). The workflow uses Picard _IntervalListTools_ to break up the intervals list into roughly balanced lists:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir -p $WORKSPACE_LOCAL/sandbox/scatter\n",
    "! gatk IntervalListTools \\\n",
    "        --INPUT $TUTORIAL_DATA_PATH/chr20sub.cohort.gc.filtered.interval_list \\\n",
    "        --SUBDIVISION_MODE INTERVAL_COUNT \\\n",
    "        --SCATTER_CONTENT 5000 \\\n",
    "        --OUTPUT $WORKSPACE_LOCAL/sandbox/scatter"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This produces three intervals lists with ~5K intervals each. For the tutorial's 1Kbp bins, this gives ~5Mbp genomic coverage per scatter. Each list is identically named scattered.interval_list within its own folder within the scatter directory. _IntervalListTools_ systematically names the intermediate folders, e.g. temp_0001_of_3, temp_0002_of_3 and temp_0002_of_3."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Comments on select parameters of _IntervalListTools_\n",
    "\n",
    "- The `--SUBDIVISION_MODE INTERVAL_COUNT` mode scatters intervals into similarly sized lists according to the count of intervals regardless of the base count. The tool intelligently breaks up the chr20sub.cohort.gc.filtered.interval_list's ~15K intervals into lists of 5031, 5031 and 5033 intervals. This is preferable to having a fourth interval list with just 95 intervals.\n",
    "- The tool has another useful feature in the context of the gCNV workflow. To subset `-I` binned intervals, provide the regions of interest with `-SI` (`--SECOND_INPUT`) and use the `--ACTION OVERLAPS` mode to create a new intervals list of the overlapping bins. Adding `--SUBDIVISION_MODE INTERVAL_COUNT --SCATTER_CONTENT 5000` will produce scatter intervals concurrently with the subsetting."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Call copy number segments and consolidate sample results with _PostprocessGermlineCNVCalls_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "_PostprocessGermlineCNVCalls_ consolidates the scattered _GermlineCNVCaller_ results, performs segmentation and calls copy number states. The tool generates per-interval and per-segment CNV calls in VCF format and runs on a single sample at a time. It also produces consolidated denoised copy-ratio estimates from all interval shards."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## _PostprocessGermlineCNVCalls_ in cohort mode\n",
    "\n",
    "Process a single sample from the 7-sample cohort using the sample index. For `NA19017`, the sample index is 2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! gatk PostprocessGermlineCNVCalls \\\n",
    "        --model-shard-path $WORKSPACE_LOCAL/sandbox/gcnv-cohort7-twelve/gcnv-cohort7-twelve-1of2-model \\\n",
    "        --model-shard-path $WORKSPACE_LOCAL/sandbox/gcnv-cohort7-twelve/gcnv-cohort7-twelve-2of2-model \\\n",
    "        --calls-shard-path $WORKSPACE_LOCAL/sandbox/gcnv-cohort7-twelve/gcnv-cohort7-twelve-1of2-calls \\\n",
    "        --calls-shard-path $WORKSPACE_LOCAL/sandbox/gcnv-cohort7-twelve/gcnv-cohort7-twelve-2of2-calls \\\n",
    "        --allosomal-contig chrX --allosomal-contig chrY \\\n",
    "        --contig-ploidy-calls $WORKSPACE_LOCAL/sandbox/ploidy-cohort7-calls \\\n",
    "        --sample-index 2 \\\n",
    "        --output-genotyped-intervals $WORKSPACE_LOCAL/sandbox/genotyped-intervals-cohort7-twelve-NA19017.vcf.gz \\\n",
    "        --output-genotyped-segments $WORKSPACE_LOCAL/sandbox/genotyped-segments-cohort7-twelve-NA19017.vcf.gz \\\n",
    "        --output-denoised-copy-ratios $WORKSPACE_LOCAL/sandbox/denoised-copy-ratios-cohort7-twelve-NA19017.tsv \\\n",
    "        --sequence-dictionary $REFERENCE_BUCKET/Homo_sapiens_assembly38.dict"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## _PostprocessGermlineCNVCalls_ in case mode\n",
    "\n",
    "`NA19017` is the singular sample with index 0 in the case mode run from section 5."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! gatk PostprocessGermlineCNVCalls \\\n",
    "        --model-shard-path $WORKSPACE_LOCAL/sandbox/gcnv-cohort6-twelve/gcnv-cohort6-twelve-1of2-model \\\n",
    "        --model-shard-path $WORKSPACE_LOCAL/sandbox/gcnv-cohort6-twelve/gcnv-cohort6-twelve-2of2-model \\\n",
    "        --calls-shard-path $WORKSPACE_LOCAL/sandbox/gcnv-case-twelve-vs-cohort6/gcnv-case-twelve-vs-cohort6-1of2-calls \\\n",
    "        --calls-shard-path $WORKSPACE_LOCAL/sandbox/gcnv-case-twelve-vs-cohort6/gcnv-case-twelve-vs-cohort6-2of2-calls \\\n",
    "        --allosomal-contig chrX --allosomal-contig chrY \\\n",
    "        --contig-ploidy-calls $WORKSPACE_LOCAL/sandbox/ploidy-cohort6-case-calls \\\n",
    "        --sample-index 0 \\\n",
    "        --output-genotyped-intervals $WORKSPACE_LOCAL/sandbox/genotyped-intervals-case-twelve-vs-cohort6.vcf.gz \\\n",
    "        --output-genotyped-segments $WORKSPACE_LOCAL/sandbox/genotyped-segments-case-twelve-vs-cohort6.vcf.gz \\\n",
    "        --output-denoised-copy-ratios $WORKSPACE_LOCAL/sandbox/denoised-copy-ratios-cohort6-twelve-NA19017.tsv \\\n",
    "        --sequence-dictionary $REFERENCE_BUCKET/Homo_sapiens_assembly38.dict"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Each command generates two VCFs with indices, and a TSV file. The `genotyped-intervals` VCF contains variant records for each genomic interval and therefore data covers only the interval regions. For the tutorial's small data, this gives 1400 records. The `genotyped-segments` VCF contains records for each contiguous copy number state segment. For the tutorial's small data, this is 30 and 31 records for cohort and case mode analyses, respectively. Finally, the `denoised-copy-ratios` TSV contains the denoised copy-ratio estimates for each genomic interval."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The CNV calls for sample `NA19017` are expected to be highly concordant between cohort and case modes. The slight difference is due to the contribution of the sample itself to model training, which is absent in the 23-sample model used for case-calling the sample. Let's take a look."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# let's exact a few things from the VCF file to a nice table\n",
    "! gatk VariantsToTable \\\n",
    "        -V $WORKSPACE_LOCAL/sandbox/genotyped-intervals-case-twelve-vs-cohort6.vcf.gz \\\n",
    "        -F CHROM -F POS -F END -GF CN \\\n",
    "        -O $WORKSPACE_LOCAL/sandbox/genotyped-intervals-case-twelve-vs-cohort6.table.txt\n",
    "! gatk VariantsToTable \\\n",
    "        -V $WORKSPACE_LOCAL/sandbox/genotyped-intervals-cohort7-twelve-NA19017.vcf.gz \\\n",
    "        -F CHROM -F POS -F END -GF CN \\\n",
    "        -O $WORKSPACE_LOCAL/sandbox/genotyped-intervals-cohort7-twelve-NA19017.table.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_copy_number_table(table_file_path):\n",
    "    table_data = []\n",
    "    with open(table_file_path, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.split()[0] == 'CHROM':\n",
    "                pass\n",
    "            else:\n",
    "                line_tokens = line.strip().split()\n",
    "                start_pos = int(line_tokens[1])\n",
    "                end_pos = int(line_tokens[2])\n",
    "                cn = int(line_tokens[3])\n",
    "                table_data.append((start_pos, end_pos, cn))\n",
    "    return table_data\n",
    "\n",
    "\n",
    "cohort_cn_calls_data = load_copy_number_table(\n",
    "    f'{WORKSPACE_LOCAL}/sandbox/genotyped-intervals-cohort7-twelve-NA19017.table.txt')\n",
    "\n",
    "case_cn_calls_data = load_copy_number_table(\n",
    "    f'{WORKSPACE_LOCAL}/sandbox/genotyped-intervals-case-twelve-vs-cohort6.table.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(cohort_cn_calls_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(case_cn_calls_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "n_agreeing_intervals = sum(\n",
    "    cohort_cn_call[2] == case_cn_call[2]\n",
    "    for cohort_cn_call, case_cn_call in zip(cohort_cn_calls_data, case_cn_calls_data))\n",
    "\n",
    "print(n_agreeing_intervals)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Only copy-number calls over 13 intervals differ between cohort and case mode for sample `NA19017`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Where to go from here?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[Article #11687](https://gatk.zendesk.com/hc/en-us/articles/360035531452) visualizes the results in IGV and provides follow-up discussion.\n",
    "\n",
    "Towards data exploration, here are two illustrative Jupyter Notebook reports that dissect the results:\n",
    "- [Notebook #11685](https://gatk.zendesk.com/hc/en-us/articles/360035890031/) shows an approach to measuring concordance of sample NA19017 gCNV calls to 1000 Genomes Project truth set calls using tutorial chr20sub small data.\n",
    "- [Notebook #11686](https://gatk.zendesk.com/hc/en-us/articles/360035889891) examines gCNV callset annotations using larger data, namely chr20 gCNV results from the tutorial's 24-sample cohort.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "colab": {
   "name": "Copy of 2-somatic-cna-tutorial.ipynb",
   "provenance": [
    {
     "file_id": "1Mq1TdSLhOasgxxz-CIDgeZlA_sKdotTs",
     "timestamp": 1560256061207
    }
   ],
   "toc_visible": true,
   "version": "0.3.2"
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {
    "height": "150px",
    "width": "161px"
   },
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "246.055px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
